{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": "true"
   },
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Introduction-to-Pandas\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Introduction to Pandas</a></div><div class=\"lev2\"><a href=\"#Pandas-Data-Structures\"><span class=\"toc-item-num\">1.1&nbsp;&nbsp;</span>Pandas Data Structures</a></div><div class=\"lev3\"><a href=\"#Series\"><span class=\"toc-item-num\">1.1.1&nbsp;&nbsp;</span>Series</a></div><div class=\"lev3\"><a href=\"#DataFrame\"><span class=\"toc-item-num\">1.1.2&nbsp;&nbsp;</span>DataFrame</a></div><div class=\"lev3\"><a href=\"#Exercise-1\"><span class=\"toc-item-num\">1.1.3&nbsp;&nbsp;</span>Exercise 1</a></div><div class=\"lev3\"><a href=\"#Exercise-2\"><span class=\"toc-item-num\">1.1.4&nbsp;&nbsp;</span>Exercise 2</a></div><div class=\"lev2\"><a href=\"#Importing-data\"><span class=\"toc-item-num\">1.2&nbsp;&nbsp;</span>Importing data</a></div><div class=\"lev3\"><a href=\"#Microsoft-Excel\"><span class=\"toc-item-num\">1.2.1&nbsp;&nbsp;</span>Microsoft Excel</a></div><div class=\"lev2\"><a href=\"#Pandas-Fundamentals\"><span class=\"toc-item-num\">1.3&nbsp;&nbsp;</span>Pandas Fundamentals</a></div><div class=\"lev3\"><a href=\"#Manipulating-indices\"><span class=\"toc-item-num\">1.3.1&nbsp;&nbsp;</span>Manipulating indices</a></div><div class=\"lev2\"><a href=\"#Indexing-and-Selection\"><span class=\"toc-item-num\">1.4&nbsp;&nbsp;</span>Indexing and Selection</a></div><div class=\"lev3\"><a href=\"#Exercise-3\"><span class=\"toc-item-num\">1.4.1&nbsp;&nbsp;</span>Exercise 3</a></div><div class=\"lev2\"><a href=\"#Operations\"><span class=\"toc-item-num\">1.5&nbsp;&nbsp;</span>Operations</a></div><div class=\"lev2\"><a href=\"#Sorting-and-Ranking\"><span class=\"toc-item-num\">1.6&nbsp;&nbsp;</span>Sorting and Ranking</a></div><div class=\"lev3\"><a href=\"#Exercise-4\"><span class=\"toc-item-num\">1.6.1&nbsp;&nbsp;</span>Exercise 4</a></div><div class=\"lev2\"><a href=\"#Hierarchical-indexing\"><span class=\"toc-item-num\">1.7&nbsp;&nbsp;</span>Hierarchical indexing</a></div><div class=\"lev2\"><a href=\"#Missing-data\"><span class=\"toc-item-num\">1.8&nbsp;&nbsp;</span>Missing data</a></div><div class=\"lev3\"><a href=\"#Exercise-5\"><span class=\"toc-item-num\">1.8.1&nbsp;&nbsp;</span>Exercise 5</a></div><div class=\"lev2\"><a href=\"#Data-summarization\"><span class=\"toc-item-num\">1.9&nbsp;&nbsp;</span>Data summarization</a></div><div class=\"lev2\"><a href=\"#Writing-Data-to-Files\"><span class=\"toc-item-num\">1.10&nbsp;&nbsp;</span>Writing Data to Files</a></div><div class=\"lev3\"><a href=\"#Advanced-Exercise:-Compiling-Ebola-Data\"><span class=\"toc-item-num\">1.10.1&nbsp;&nbsp;</span>Advanced Exercise: Compiling Ebola Data</a></div><div class=\"lev2\"><a href=\"#References\"><span class=\"toc-item-num\">1.11&nbsp;&nbsp;</span>References</a></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction to Pandas\n",
    "\n",
    "**pandas** is a Python package providing fast, flexible, and expressive data structures designed to work with *relational* or *labeled* data both. It is a fundamental high-level building block for doing practical, real world data analysis in Python. \n",
    "\n",
    "pandas is well suited for:\n",
    "\n",
    "- Tabular data with heterogeneously-typed columns, as in an SQL table or Excel spreadsheet\n",
    "- Ordered and unordered (not necessarily fixed-frequency) time series data.\n",
    "- Arbitrary matrix data (homogeneously typed or heterogeneous) with row and column labels\n",
    "- Any other form of observational / statistical data sets. The data actually need not be labeled at all to be placed into a pandas data structure\n",
    "\n",
    "\n",
    "Key features:\n",
    "    \n",
    "- Easy handling of **missing data**\n",
    "- **Size mutability**: columns can be inserted and deleted from DataFrame and higher dimensional objects\n",
    "- Automatic and explicit **data alignment**: objects can be explicitly aligned to a set of labels, or the data can be aligned automatically\n",
    "- Powerful, flexible **group by functionality** to perform split-apply-combine operations on data sets\n",
    "- Intelligent label-based **slicing, fancy indexing, and subsetting** of large data sets\n",
    "- Intuitive **merging and joining** data sets\n",
    "- Flexible **reshaping and pivoting** of data sets\n",
    "- **Hierarchical labeling** of axes\n",
    "- Robust **IO tools** for loading data from flat files, Excel files, databases, and HDF5\n",
    "- **Time series functionality**: date range generation and frequency conversion, moving window statistics, moving window linear regressions, date shifting and lagging, etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "pd.options.mode.chained_assignment = None  # default='warn'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pandas Data Structures"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Series\n",
    "\n",
    "A **Series** is a single vector of data (like a NumPy array) with an *index* that labels each element in the vector."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     632\n",
       "1    1638\n",
       "2     569\n",
       "3     115\n",
       "dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts = pd.Series([632, 1638, 569, 115])\n",
    "counts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If an index is not specified, a default sequence of integers is assigned as the index. A NumPy array comprises the values of the `Series`, while the index is a pandas `Index` object."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 632, 1638,  569,  115])"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RangeIndex(start=0, stop=4, step=1)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts.index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can assign meaningful labels to the index, if they are available:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Firmicutes         632\n",
       "Proteobacteria    1638\n",
       "Actinobacteria     569\n",
       "Bacteroidetes      115\n",
       "dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria = pd.Series([632, 1638, 569, 115], \n",
    "    index=['Firmicutes', 'Proteobacteria', 'Actinobacteria', 'Bacteroidetes'])\n",
    "\n",
    "bacteria"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These labels can be used to refer to the values in the `Series`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "569"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria['Actinobacteria']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Proteobacteria    1638\n",
       "Actinobacteria     569\n",
       "dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria[[name.endswith('bacteria') for name in bacteria.index]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[False, True, True, False]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[name.endswith('bacteria') for name in bacteria.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that the indexing operation preserved the association between the values and the corresponding indices.\n",
    "\n",
    "We can still use positional indexing if we wish."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "632"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can give both the array of values and the index meaningful labels themselves:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phylum\n",
       "Firmicutes         632\n",
       "Proteobacteria    1638\n",
       "Actinobacteria     569\n",
       "Bacteroidetes      115\n",
       "Name: counts, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria.name = 'counts'\n",
    "bacteria.index.name = 'phylum'\n",
    "bacteria"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "NumPy's math functions and other operations can be applied to Series without losing the data structure."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phylum\n",
       "Firmicutes        2.800717\n",
       "Proteobacteria    3.214314\n",
       "Actinobacteria    2.755112\n",
       "Bacteroidetes     2.060698\n",
       "Name: counts, dtype: float64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# natural logarithm\n",
    "np.log(bacteria)\n",
    "# log base 10\n",
    "np.log10(bacteria)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also filter according to the values in the `Series`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phylum\n",
       "Proteobacteria    1638\n",
       "Name: counts, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria[bacteria>1000]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A `Series` can be thought of as an ordered key-value store. In fact, we can create one from a `dict`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Actinobacteria     569\n",
       "Bacteroidetes      115\n",
       "Firmicutes         632\n",
       "Proteobacteria    1638\n",
       "dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria_dict = {'Firmicutes': 632, 'Proteobacteria': 1638, 'Actinobacteria': 569,\n",
    "                 'Bacteroidetes': 115}\n",
    "pd.Series(bacteria_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that the `Series` is created in key-sorted order.\n",
    "\n",
    "If we pass a custom index to `Series`, it will select the corresponding values from the dict, and treat indices without corrsponding values as missing. Pandas uses the `NaN` (not a number) type for missing values."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cyanobacteria        NaN\n",
       "Firmicutes         632.0\n",
       "Proteobacteria    1638.0\n",
       "Actinobacteria     569.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria2 = pd.Series(bacteria_dict, \n",
    "                      index=['Cyanobacteria','Firmicutes',\n",
    "                             'Proteobacteria','Actinobacteria'])\n",
    "bacteria2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cyanobacteria      True\n",
       "Firmicutes        False\n",
       "Proteobacteria    False\n",
       "Actinobacteria    False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria2.isnull()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Critically, the labels are used to **align data** when used in operations with other Series objects:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Actinobacteria    1138.0\n",
       "Bacteroidetes        NaN\n",
       "Cyanobacteria        NaN\n",
       "Firmicutes        1264.0\n",
       "Proteobacteria    3276.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria + bacteria2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Contrast this with NumPy arrays, where arrays of the same length will combine values element-wise; adding Series combined values with the same label in the resulting series. Notice also that the missing values were propogated by addition."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DataFrame\n",
    "\n",
    "Inevitably, we want to be able to store, view and manipulate data that is *multivariate*, where for every index there are multiple fields or columns of data, often of varying data type.\n",
    "\n",
    "A `DataFrame` is a tabular data structure, encapsulating multiple series like columns in a spreadsheet. Data are stored internally as a 2-dimensional object, but the `DataFrame` allows us to represent and manipulate higher-dimensional data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value\n",
       "0        1      Firmicutes    632\n",
       "1        1  Proteobacteria   1638\n",
       "2        1  Actinobacteria    569\n",
       "3        1   Bacteroidetes    115\n",
       "4        2      Firmicutes    433\n",
       "5        2  Proteobacteria   1130\n",
       "6        2  Actinobacteria    754\n",
       "7        2   Bacteroidetes    555"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.DataFrame({'value':[632, 1638, 569, 115, 433, 1130, 754, 555],\n",
    "                     'patient':[1, 1, 1, 1, 2, 2, 2, 2],\n",
    "                     'phylum':['Firmicutes', 'Proteobacteria', 'Actinobacteria', \n",
    "    'Bacteroidetes', 'Firmicutes', 'Proteobacteria', 'Actinobacteria', 'Bacteroidetes']})\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice the `DataFrame` is sorted by column name. We can change the order by indexing them in the order we desire:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>115</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>433</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1130</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>754</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           phylum  value  patient\n",
       "0      Firmicutes    632        1\n",
       "1  Proteobacteria   1638        1\n",
       "2  Actinobacteria    569        1\n",
       "3   Bacteroidetes    115        1\n",
       "4      Firmicutes    433        2\n",
       "5  Proteobacteria   1130        2\n",
       "6  Actinobacteria    754        2\n",
       "7   Bacteroidetes    555        2"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[['phylum','value','patient']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A `DataFrame` has a second index, representing the columns:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['patient', 'phylum', 'value'], dtype='object')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `dtypes` attribute reveals the data type for each column in our DataFrame. \n",
    "\n",
    "- `int64` is numeric integer values \n",
    "- `object` strings (letters and numbers)\n",
    "- `float64` floating-point values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "patient     int64\n",
       "phylum     object\n",
       "value       int64\n",
       "dtype: object"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If we wish to access columns, we can do so either by dict-like indexing or by attribute:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    1\n",
       "1    1\n",
       "2    1\n",
       "3    1\n",
       "4    2\n",
       "5    2\n",
       "6    2\n",
       "7    2\n",
       "Name: patient, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['patient']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    1\n",
       "1    1\n",
       "2    1\n",
       "3    1\n",
       "4    2\n",
       "5    2\n",
       "6    2\n",
       "7    2\n",
       "Name: patient, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.patient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.series.Series"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(data.value)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   value\n",
       "0    632\n",
       "1   1638\n",
       "2    569\n",
       "3    115\n",
       "4    433\n",
       "5   1130\n",
       "6    754\n",
       "7    555"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[['value']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice this is different than with `Series`, where dict-like indexing retrieved a particular element (row). \n",
    "\n",
    "If we want access to a row in a `DataFrame`, we index its `loc` attribute."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "patient                1\n",
       "phylum     Bacteroidetes\n",
       "value                115\n",
       "Name: 3, dtype: object"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.loc[3]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exercise 1\n",
    "\n",
    "Try out these commands to see what they return:\n",
    "\n",
    "- `data.head()`\n",
    "- `data.tail(3)`\n",
    "- `data.shape`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>433</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value\n",
       "0        1      Firmicutes    632\n",
       "1        1  Proteobacteria   1638\n",
       "2        1  Actinobacteria    569\n",
       "3        1   Bacteroidetes    115\n",
       "4        2      Firmicutes    433"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head() # returns the first (5 by default) rows of data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value\n",
       "5        2  Proteobacteria   1130\n",
       "6        2  Actinobacteria    754\n",
       "7        2   Bacteroidetes    555"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.tail(3) # returns the 3 last rows of data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8, 3)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.shape # returns the dimension of data (nbr rows, nbr cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An alternative way of initializing a `DataFrame` is with a list of dicts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data = pd.DataFrame([{'patient': 1, 'phylum': 'Firmicutes', 'value': 632},\n",
    "                    {'patient': 1, 'phylum': 'Proteobacteria', 'value': 1638},\n",
    "                    {'patient': 1, 'phylum': 'Actinobacteria', 'value': 569},\n",
    "                    {'patient': 1, 'phylum': 'Bacteroidetes', 'value': 115},\n",
    "                    {'patient': 2, 'phylum': 'Firmicutes', 'value': 433},\n",
    "                    {'patient': 2, 'phylum': 'Proteobacteria', 'value': 1130},\n",
    "                    {'patient': 2, 'phylum': 'Actinobacteria', 'value': 754},\n",
    "                    {'patient': 2, 'phylum': 'Bacteroidetes', 'value': 555}])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value\n",
       "0        1      Firmicutes    632\n",
       "1        1  Proteobacteria   1638\n",
       "2        1  Actinobacteria    569\n",
       "3        1   Bacteroidetes    115\n",
       "4        2      Firmicutes    433\n",
       "5        2  Proteobacteria   1130\n",
       "6        2  Actinobacteria    754\n",
       "7        2   Bacteroidetes    555"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Its important to note that the Series returned when a DataFrame is indexted is merely a **view** on the DataFrame, and not a copy of the data itself. So you must be cautious when manipulating this data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     632\n",
       "1    1638\n",
       "2     569\n",
       "3     115\n",
       "4     433\n",
       "5    1130\n",
       "6     754\n",
       "7     555\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vals = data.value\n",
    "vals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     632\n",
       "1    1638\n",
       "2     569\n",
       "3     115\n",
       "4     433\n",
       "5       0\n",
       "6     754\n",
       "7     555\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vals[5] = 0\n",
    "vals"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If we plan on modifying an extracted Series, its a good idea to make a copy."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     632\n",
       "1    1638\n",
       "2     569\n",
       "3     115\n",
       "4     433\n",
       "5    1000\n",
       "6     754\n",
       "7     555\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vals = data.value.copy()\n",
    "vals[5] = 1000\n",
    "vals"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can create or modify columns by assignment:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value\n",
       "0        1      Firmicutes    632\n",
       "1        1  Proteobacteria   1638\n",
       "2        1  Actinobacteria    569\n",
       "3        1   Bacteroidetes     14\n",
       "4        2      Firmicutes     21\n",
       "5        2  Proteobacteria      0\n",
       "6        2  Actinobacteria      5\n",
       "7        2   Bacteroidetes    555"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.value[[3,4,6]] = [14, 21, 5]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>21</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year\n",
       "0        1      Firmicutes    632  2013\n",
       "1        1  Proteobacteria   1638  2013\n",
       "2        1  Actinobacteria    569  2013\n",
       "3        1   Bacteroidetes     14  2013\n",
       "4        2      Firmicutes     21  2013\n",
       "5        2  Proteobacteria      0  2013\n",
       "6        2  Actinobacteria      5  2013\n",
       "7        2   Bacteroidetes    555  2013"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['year'] = 2013\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "But note, we cannot use the attribute indexing method to add a new column:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>21</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year\n",
       "0        1      Firmicutes    632  2013\n",
       "1        1  Proteobacteria   1638  2013\n",
       "2        1  Actinobacteria    569  2013\n",
       "3        1   Bacteroidetes     14  2013\n",
       "4        2      Firmicutes     21  2013\n",
       "5        2  Proteobacteria      0  2013\n",
       "6        2  Actinobacteria      5  2013\n",
       "7        2   Bacteroidetes    555  2013"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.treatment = 1\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.treatment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exercise 2\n",
    "\n",
    "From the `data` table above, create an index to return all rows for which the phylum name ends in \"bacteria\" and the value is greater than 1000."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----------------------------\n",
    "Find the values of 'phylum' ending in 'bacteria'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Proteobacteria', 'Actinobacteria', 'Proteobacteria', 'Actinobacteria']"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colwitbacteria = [col for col in data['phylum'] if col.endswith('bacteria')]\n",
    "colwitbacteria"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "then filter the rows having one of the 'bacteria' values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year\n",
       "1        1  Proteobacteria   1638  2013\n",
       "2        1  Actinobacteria    569  2013\n",
       "5        2  Proteobacteria      0  2013\n",
       "6        2  Actinobacteria      5  2013"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rowswithbacteria = data[data['phylum'].isin(colwitbacteria)]\n",
    "rowswithbacteria"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "then take the values bigger than 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year\n",
       "1        1  Proteobacteria   1638  2013"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rowswithbacteria[rowswithbacteria.value > 1000]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note that it is probably faster to first filter the values bigger than 1000 as it filters more values out.\n",
    "----------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Specifying a `Series` as a new columns cause its values to be added according to the `DataFrame`'s index:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    0\n",
       "1    0\n",
       "2    0\n",
       "3    0\n",
       "4    1\n",
       "5    1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "treatment = pd.Series([0]*4 + [1]*2)\n",
    "treatment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "      <th>treatment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>21</td>\n",
       "      <td>2013</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "      <td>2013</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year  treatment\n",
       "0        1      Firmicutes    632  2013        0.0\n",
       "1        1  Proteobacteria   1638  2013        0.0\n",
       "2        1  Actinobacteria    569  2013        0.0\n",
       "3        1   Bacteroidetes     14  2013        0.0\n",
       "4        2      Firmicutes     21  2013        1.0\n",
       "5        2  Proteobacteria      0  2013        1.0\n",
       "6        2  Actinobacteria      5  2013        NaN\n",
       "7        2   Bacteroidetes    555  2013        NaN"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['treatment'] = treatment\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Other Python data structures (ones without an index) need to be the same length as the `DataFrame`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "month = ['Jan', 'Feb', 'Mar', 'Apr']\n",
    "# data['month'] = month # throws error (done on puropse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "      <th>treatment</th>\n",
       "      <th>month</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>21</td>\n",
       "      <td>2013</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "      <td>2013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Jan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year  treatment month\n",
       "0        1      Firmicutes    632  2013        0.0   Jan\n",
       "1        1  Proteobacteria   1638  2013        0.0   Jan\n",
       "2        1  Actinobacteria    569  2013        0.0   Jan\n",
       "3        1   Bacteroidetes     14  2013        0.0   Jan\n",
       "4        2      Firmicutes     21  2013        1.0   Jan\n",
       "5        2  Proteobacteria      0  2013        1.0   Jan\n",
       "6        2  Actinobacteria      5  2013        NaN   Jan\n",
       "7        2   Bacteroidetes    555  2013        NaN   Jan"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['month'] = ['Jan']*len(data)\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can use the `drop` method to remove rows or columns, which by default drops rows. We can be explicit by using the `axis` argument:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>phylum</th>\n",
       "      <th>value</th>\n",
       "      <th>year</th>\n",
       "      <th>treatment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>632</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1638</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>569</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>2013</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>21</td>\n",
       "      <td>2013</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>555</td>\n",
       "      <td>2013</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient          phylum  value  year  treatment\n",
       "0        1      Firmicutes    632  2013        0.0\n",
       "1        1  Proteobacteria   1638  2013        0.0\n",
       "2        1  Actinobacteria    569  2013        0.0\n",
       "3        1   Bacteroidetes     14  2013        0.0\n",
       "4        2      Firmicutes     21  2013        1.0\n",
       "5        2  Proteobacteria      0  2013        1.0\n",
       "6        2  Actinobacteria      5  2013        NaN\n",
       "7        2   Bacteroidetes    555  2013        NaN"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_nomonth = data.drop('month', axis=1)\n",
    "data_nomonth"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can extract the underlying data as a simple `ndarray` by accessing the `values` attribute:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1, 'Firmicutes', 632, 2013, 0.0, 'Jan'],\n",
       "       [1, 'Proteobacteria', 1638, 2013, 0.0, 'Jan'],\n",
       "       [1, 'Actinobacteria', 569, 2013, 0.0, 'Jan'],\n",
       "       [1, 'Bacteroidetes', 14, 2013, 0.0, 'Jan'],\n",
       "       [2, 'Firmicutes', 21, 2013, 1.0, 'Jan'],\n",
       "       [2, 'Proteobacteria', 0, 2013, 1.0, 'Jan'],\n",
       "       [2, 'Actinobacteria', 5, 2013, nan, 'Jan'],\n",
       "       [2, 'Bacteroidetes', 555, 2013, nan, 'Jan']], dtype=object)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that because of the mix of string and integer (and `NaN`) values, the dtype of the array is `object`. The dtype will automatically be chosen to be as general as needed to accomodate all the columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 0.4,  1. ],\n",
       "       [-1. ,  2. ],\n",
       "       [ 4.5,  3. ]])"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame({'foo': [1,2,3], 'bar':[0.4, -1.0, 4.5]})\n",
    "df.values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pandas uses a custom data structure to represent the indices of Series and DataFrames."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RangeIndex(start=0, stop=8, step=1)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Index objects are immutable:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# data.index[0] = 15 # throws error "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is so that Index objects can be shared between data structures without fear that they will be changed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.index = bacteria.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phylum\n",
       "Firmicutes           NaN\n",
       "Proteobacteria     632.0\n",
       "Actinobacteria    1638.0\n",
       "Bacteroidetes      569.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Importing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A key, but often under-appreciated, step in data analysis is importing the data that we wish to analyze. Though it is easy to load basic data structures into Python using built-in tools or those provided by packages like NumPy, it is non-trivial to import structured data well, and to easily convert this input into a robust data structure:\n",
    "\n",
    "    genes = np.loadtxt(\"genes.csv\", delimiter=\",\", dtype=[('gene', '|S10'), ('value', '<f4')])\n",
    "\n",
    "Pandas provides a convenient set of functions for importing tabular data in a number of formats directly into a `DataFrame` object. These functions include a slew of options to perform type inference, indexing, parsing, iterating and cleaning automatically as data are imported."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's start with some more bacteria data, stored in csv format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Taxon,Patient,Group,Tissue,Stool\r\n",
      "Firmicutes,1,0,136,4182\r\n",
      "Firmicutes,2,1,1174,703\r\n",
      "Firmicutes,3,0,408,3946\r\n",
      "Firmicutes,4,1,831,8605\r\n",
      "Firmicutes,5,0,693,50\r\n",
      "Firmicutes,6,1,718,717\r\n",
      "Firmicutes,7,0,173,33\r\n",
      "Firmicutes,8,1,228,80\r\n",
      "Firmicutes,9,0,162,3196\r\n",
      "Firmicutes,10,1,372,32\r\n",
      "Firmicutes,11,0,4255,4361\r\n",
      "Firmicutes,12,1,107,1667\r\n",
      "Firmicutes,13,0,96,223\r\n",
      "Firmicutes,14,1,281,2377\r\n",
      "Proteobacteria,1,0,2469,1821\r\n",
      "Proteobacteria,2,1,839,661\r\n",
      "Proteobacteria,3,0,4414,18\r\n",
      "Proteobacteria,4,1,12044,83\r\n",
      "Proteobacteria,5,0,2310,12\r\n",
      "Proteobacteria,6,1,3053,547\r\n",
      "Proteobacteria,7,0,395,2174\r\n",
      "Proteobacteria,8,1,2651,767\r\n",
      "Proteobacteria,9,0,1195,76\r\n",
      "Proteobacteria,10,1,6857,795\r\n",
      "Proteobacteria,11,0,483,666\r\n",
      "Proteobacteria,12,1,2950,3994\r\n",
      "Proteobacteria,13,0,1541,816\r\n",
      "Proteobacteria,14,1,1307,53\r\n",
      "Actinobacteria,1,0,1590,4\r\n",
      "Actinobacteria,2,1,25,2\r\n",
      "Actinobacteria,3,0,259,300\r\n",
      "Actinobacteria,4,1,568,7\r\n",
      "Actinobacteria,5,0,1102,9\r\n",
      "Actinobacteria,6,1,678,377\r\n",
      "Actinobacteria,7,0,260,58\r\n",
      "Actinobacteria,8,1,424,233\r\n",
      "Actinobacteria,9,0,548,21\r\n",
      "Actinobacteria,10,1,201,83\r\n",
      "Actinobacteria,11,0,42,75\r\n",
      "Actinobacteria,12,1,109,59\r\n",
      "Actinobacteria,13,0,51,183\r\n",
      "Actinobacteria,14,1,310,204\r\n",
      "Bacteroidetes,1,0,67,0\r\n",
      "Bacteroidetes,2,1,0,0\r\n",
      "Bacteroidetes,3,0,85,5\r\n",
      "Bacteroidetes,4,1,143,7\r\n",
      "Bacteroidetes,5,0,678,2\r\n",
      "Bacteroidetes,6,1,4829,209\r\n",
      "Bacteroidetes,7,0,74,651\r\n",
      "Bacteroidetes,8,1,169,254\r\n",
      "Bacteroidetes,9,0,106,10\r\n",
      "Bacteroidetes,10,1,73,381\r\n",
      "Bacteroidetes,11,0,30,359\r\n",
      "Bacteroidetes,12,1,51,51\r\n",
      "Bacteroidetes,13,0,2473,2314\r\n",
      "Bacteroidetes,14,1,102,33\r\n",
      "Other,1,0,195,18\r\n",
      "Other,2,1,42,2\r\n",
      "Other,3,0,316,43\r\n",
      "Other,4,1,202,40\r\n",
      "Other,5,0,116,0\r\n",
      "Other,6,1,527,12\r\n",
      "Other,7,0,357,11\r\n",
      "Other,8,1,106,11\r\n",
      "Other,9,0,67,14\r\n",
      "Other,10,1,203,6\r\n",
      "Other,11,0,392,6\r\n",
      "Other,12,1,28,25\r\n",
      "Other,13,0,12,22\r\n",
      "Other,14,1,305,32"
     ]
    }
   ],
   "source": [
    "!cat Data/microbiome.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This table can be read into a DataFrame using `read_csv`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Group</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "      <td>4182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1174</td>\n",
       "      <td>703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>3946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>831</td>\n",
       "      <td>8605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>693</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>718</td>\n",
       "      <td>717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>173</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>228</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>162</td>\n",
       "      <td>3196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>372</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>4255</td>\n",
       "      <td>4361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>107</td>\n",
       "      <td>1667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>96</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>281</td>\n",
       "      <td>2377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2469</td>\n",
       "      <td>1821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>839</td>\n",
       "      <td>661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>4414</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>12044</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2310</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>3053</td>\n",
       "      <td>547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>395</td>\n",
       "      <td>2174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>2651</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1195</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>6857</td>\n",
       "      <td>795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>483</td>\n",
       "      <td>666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>2950</td>\n",
       "      <td>3994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1541</td>\n",
       "      <td>816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>1307</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1590</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>310</td>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>143</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>678</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>4829</td>\n",
       "      <td>209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>74</td>\n",
       "      <td>651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>169</td>\n",
       "      <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>106</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>73</td>\n",
       "      <td>381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>2473</td>\n",
       "      <td>2314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Other</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>195</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Other</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>42</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Other</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>316</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Other</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>202</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Other</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Other</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>527</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Other</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>357</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Other</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Other</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Other</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Other</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>392</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Other</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Other</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Other</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>305</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Taxon  Patient  Group  Tissue  Stool\n",
       "0       Firmicutes        1      0     136   4182\n",
       "1       Firmicutes        2      1    1174    703\n",
       "2       Firmicutes        3      0     408   3946\n",
       "3       Firmicutes        4      1     831   8605\n",
       "4       Firmicutes        5      0     693     50\n",
       "5       Firmicutes        6      1     718    717\n",
       "6       Firmicutes        7      0     173     33\n",
       "7       Firmicutes        8      1     228     80\n",
       "8       Firmicutes        9      0     162   3196\n",
       "9       Firmicutes       10      1     372     32\n",
       "10      Firmicutes       11      0    4255   4361\n",
       "11      Firmicutes       12      1     107   1667\n",
       "12      Firmicutes       13      0      96    223\n",
       "13      Firmicutes       14      1     281   2377\n",
       "14  Proteobacteria        1      0    2469   1821\n",
       "15  Proteobacteria        2      1     839    661\n",
       "16  Proteobacteria        3      0    4414     18\n",
       "17  Proteobacteria        4      1   12044     83\n",
       "18  Proteobacteria        5      0    2310     12\n",
       "19  Proteobacteria        6      1    3053    547\n",
       "20  Proteobacteria        7      0     395   2174\n",
       "21  Proteobacteria        8      1    2651    767\n",
       "22  Proteobacteria        9      0    1195     76\n",
       "23  Proteobacteria       10      1    6857    795\n",
       "24  Proteobacteria       11      0     483    666\n",
       "25  Proteobacteria       12      1    2950   3994\n",
       "26  Proteobacteria       13      0    1541    816\n",
       "27  Proteobacteria       14      1    1307     53\n",
       "28  Actinobacteria        1      0    1590      4\n",
       "29  Actinobacteria        2      1      25      2\n",
       "..             ...      ...    ...     ...    ...\n",
       "40  Actinobacteria       13      0      51    183\n",
       "41  Actinobacteria       14      1     310    204\n",
       "42   Bacteroidetes        1      0      67      0\n",
       "43   Bacteroidetes        2      1       0      0\n",
       "44   Bacteroidetes        3      0      85      5\n",
       "45   Bacteroidetes        4      1     143      7\n",
       "46   Bacteroidetes        5      0     678      2\n",
       "47   Bacteroidetes        6      1    4829    209\n",
       "48   Bacteroidetes        7      0      74    651\n",
       "49   Bacteroidetes        8      1     169    254\n",
       "50   Bacteroidetes        9      0     106     10\n",
       "51   Bacteroidetes       10      1      73    381\n",
       "52   Bacteroidetes       11      0      30    359\n",
       "53   Bacteroidetes       12      1      51     51\n",
       "54   Bacteroidetes       13      0    2473   2314\n",
       "55   Bacteroidetes       14      1     102     33\n",
       "56           Other        1      0     195     18\n",
       "57           Other        2      1      42      2\n",
       "58           Other        3      0     316     43\n",
       "59           Other        4      1     202     40\n",
       "60           Other        5      0     116      0\n",
       "61           Other        6      1     527     12\n",
       "62           Other        7      0     357     11\n",
       "63           Other        8      1     106     11\n",
       "64           Other        9      0      67     14\n",
       "65           Other       10      1     203      6\n",
       "66           Other       11      0     392      6\n",
       "67           Other       12      1      28     25\n",
       "68           Other       13      0      12     22\n",
       "69           Other       14      1     305     32\n",
       "\n",
       "[70 rows x 5 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mb = pd.read_csv(\"Data/microbiome.csv\")\n",
    "mb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that `read_csv` automatically considered the first row in the file to be a header row.\n",
    "\n",
    "We can override default behavior by customizing some the arguments, like `header`, `names` or `index_col`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Taxon</td>\n",
       "      <td>Patient</td>\n",
       "      <td>Group</td>\n",
       "      <td>Tissue</td>\n",
       "      <td>Stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "      <td>4182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1174</td>\n",
       "      <td>703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>3946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>831</td>\n",
       "      <td>8605</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            0        1      2       3      4\n",
       "0       Taxon  Patient  Group  Tissue  Stool\n",
       "1  Firmicutes        1      0     136   4182\n",
       "2  Firmicutes        2      1    1174    703\n",
       "3  Firmicutes        3      0     408   3946\n",
       "4  Firmicutes        4      1     831   8605"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/microbiome.csv\", header=None).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`read_csv` is just a convenience function for `read_table`, since csv is such a common format:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb = pd.read_table(\"Data/microbiome.csv\", sep=',')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `sep` argument can be customized as needed to accomodate arbitrary separators. For example, we can use a regular expression to define a variable amount of whitespace, which is unfortunately very common in some data formats: \n",
    "    \n",
    "    sep='\\s+'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For a more useful index, we can specify the first two columns, which together provide a unique index to the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient</th>\n",
       "      <th>Taxon</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "      <td>4182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <td>1</td>\n",
       "      <td>1174</td>\n",
       "      <td>703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>3946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <td>1</td>\n",
       "      <td>831</td>\n",
       "      <td>8605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <td>0</td>\n",
       "      <td>693</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Group  Tissue  Stool\n",
       "Patient Taxon                           \n",
       "1       Firmicutes      0     136   4182\n",
       "2       Firmicutes      1    1174    703\n",
       "3       Firmicutes      0     408   3946\n",
       "4       Firmicutes      1     831   8605\n",
       "5       Firmicutes      0     693     50"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mb = pd.read_csv(\"Data/microbiome.csv\", index_col=['Patient','Taxon'])\n",
    "mb.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is called a *hierarchical* index, which we will revisit later in the section."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If we have sections of data that we do not wish to import (for example, known bad data), we can populate the `skiprows` argument:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Group</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "      <td>4182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1174</td>\n",
       "      <td>703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>693</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>173</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>228</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Taxon  Patient  Group  Tissue  Stool\n",
       "0  Firmicutes        1      0     136   4182\n",
       "1  Firmicutes        2      1    1174    703\n",
       "2  Firmicutes        5      0     693     50\n",
       "3  Firmicutes        7      0     173     33\n",
       "4  Firmicutes        8      1     228     80"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/microbiome.csv\", skiprows=[3,4,6]).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If we only want to import a small number of rows from, say, a very large data file we can use `nrows`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Group</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "      <td>4182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1174</td>\n",
       "      <td>703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>3946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>831</td>\n",
       "      <td>8605</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Taxon  Patient  Group  Tissue  Stool\n",
       "0  Firmicutes        1      0     136   4182\n",
       "1  Firmicutes        2      1    1174    703\n",
       "2  Firmicutes        3      0     408   3946\n",
       "3  Firmicutes        4      1     831   8605"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/microbiome.csv\", nrows=4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Alternately, if we want to process our data in reasonable chunks, the `chunksize` argument will return an iterable object that can be employed in a data processing loop. For example, our microbiome data are organized by bacterial phylum, with 14 patients represented in each:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pandas.io.parsers.TextFileReader at 0x7f8484a90d68>"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/microbiome.csv\", chunksize=14)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Actinobacteria     440.500000\n",
       "Bacteroidetes      634.285714\n",
       "Firmicutes         688.142857\n",
       "Other              204.857143\n",
       "Proteobacteria    3036.285714\n",
       "dtype: float64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_chunks = pd.read_csv(\"Data/microbiome.csv\", chunksize=14)\n",
    "\n",
    "mean_tissue = pd.Series({chunk.Taxon[0]: chunk.Tissue.mean() for chunk in data_chunks})\n",
    "    \n",
    "mean_tissue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Most real-world data is incomplete, with values missing due to incomplete observation, data entry or transcription error, or other reasons. Pandas will automatically recognize and parse common missing data indicators, including `NA` and `NULL`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Taxon,Patient,Tissue,Stool\r\n",
      "Firmicutes,1,632,305\r\n",
      "Firmicutes,2,136,4182\r\n",
      "Firmicutes,3,,703\r\n",
      "Firmicutes,4,408,3946\r\n",
      "Firmicutes,5,831,8605\r\n",
      "Firmicutes,6,693,50\r\n",
      "Firmicutes,7,718,717\r\n",
      "Firmicutes,8,173,33\r\n",
      "Firmicutes,9,228,NA\r\n",
      "Firmicutes,10,162,3196\r\n",
      "Firmicutes,11,372,-99999\r\n",
      "Firmicutes,12,4255,4361\r\n",
      "Firmicutes,13,107,1667\r\n",
      "Firmicutes,14,?,223\r\n",
      "Firmicutes,15,281,2377\r\n",
      "Proteobacteria,1,1638,3886\r\n",
      "Proteobacteria,2,2469,1821\r\n",
      "Proteobacteria,3,839,661\r\n",
      "Proteobacteria,4,4414,18\r\n",
      "Proteobacteria,5,12044,83\r\n",
      "Proteobacteria,6,2310,12\r\n",
      "Proteobacteria,7,3053,547\r\n",
      "Proteobacteria,8,395,2174\r\n",
      "Proteobacteria,9,2651,767\r\n",
      "Proteobacteria,10,1195,76\r\n",
      "Proteobacteria,11,6857,795\r\n",
      "Proteobacteria,12,483,666\r\n",
      "Proteobacteria,13,2950,3994\r\n",
      "Proteobacteria,14,1541,816\r\n",
      "Proteobacteria,15,1307,53\r\n",
      "Actinobacteria,1,569,648\r\n",
      "Actinobacteria,2,1590,4\r\n",
      "Actinobacteria,3,25,2\r\n",
      "Actinobacteria,4,259,300\r\n",
      "Actinobacteria,5,568,7\r\n",
      "Actinobacteria,6,1102,9\r\n",
      "Actinobacteria,7,678,377\r\n",
      "Actinobacteria,8,260,58\r\n",
      "Actinobacteria,9,424,233\r\n",
      "Actinobacteria,10,548,21\r\n",
      "Actinobacteria,11,201,83\r\n",
      "Actinobacteria,12,42,75\r\n",
      "Actinobacteria,13,109,59\r\n",
      "Actinobacteria,14,51,183\r\n",
      "Actinobacteria,15,310,204\r\n",
      "Bacteroidetes,1,115,380\r\n",
      "Bacteroidetes,2,67,0\r\n",
      "Bacteroidetes,3,0,0\r\n",
      "Bacteroidetes,4,85,5\r\n",
      "Bacteroidetes,5,143,7\r\n",
      "Bacteroidetes,6,678,2\r\n",
      "Bacteroidetes,7,4829,209\r\n",
      "Bacteroidetes,8,74,651\r\n",
      "Bacteroidetes,9,169,254\r\n",
      "Bacteroidetes,10,106,10\r\n",
      "Bacteroidetes,11,73,381\r\n",
      "Bacteroidetes,12,30,359\r\n",
      "Bacteroidetes,13,51,51\r\n",
      "Bacteroidetes,14,2473,2314\r\n",
      "Bacteroidetes,15,102,33\r\n",
      "Other,1,114,277\r\n",
      "Other,2,195,18\r\n",
      "Other,3,42,2\r\n",
      "Other,4,316,43\r\n",
      "Other,5,202,40\r\n",
      "Other,6,116,0\r\n",
      "Other,7,527,12\r\n",
      "Other,8,357,11\r\n",
      "Other,9,106,11\r\n",
      "Other,10,67,14\r\n",
      "Other,11,203,6\r\n",
      "Other,12,392,6\r\n",
      "Other,13,28,25\r\n",
      "Other,14,12,22\r\n",
      "Other,15,305,32"
     ]
    }
   ],
   "source": [
    "!cat Data/microbiome_missing.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>632</td>\n",
       "      <td>305.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2</td>\n",
       "      <td>136</td>\n",
       "      <td>4182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>703.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>4</td>\n",
       "      <td>408</td>\n",
       "      <td>3946.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>5</td>\n",
       "      <td>831</td>\n",
       "      <td>8605.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>6</td>\n",
       "      <td>693</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>7</td>\n",
       "      <td>718</td>\n",
       "      <td>717.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>8</td>\n",
       "      <td>173</td>\n",
       "      <td>33.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>9</td>\n",
       "      <td>228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>10</td>\n",
       "      <td>162</td>\n",
       "      <td>3196.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>11</td>\n",
       "      <td>372</td>\n",
       "      <td>-99999.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>12</td>\n",
       "      <td>4255</td>\n",
       "      <td>4361.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>13</td>\n",
       "      <td>107</td>\n",
       "      <td>1667.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "      <td>?</td>\n",
       "      <td>223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>15</td>\n",
       "      <td>281</td>\n",
       "      <td>2377.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>1638</td>\n",
       "      <td>3886.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>2</td>\n",
       "      <td>2469</td>\n",
       "      <td>1821.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3</td>\n",
       "      <td>839</td>\n",
       "      <td>661.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>4</td>\n",
       "      <td>4414</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>12044</td>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Taxon  Patient Tissue    Stool\n",
       "0       Firmicutes        1    632    305.0\n",
       "1       Firmicutes        2    136   4182.0\n",
       "2       Firmicutes        3    NaN    703.0\n",
       "3       Firmicutes        4    408   3946.0\n",
       "4       Firmicutes        5    831   8605.0\n",
       "5       Firmicutes        6    693     50.0\n",
       "6       Firmicutes        7    718    717.0\n",
       "7       Firmicutes        8    173     33.0\n",
       "8       Firmicutes        9    228      NaN\n",
       "9       Firmicutes       10    162   3196.0\n",
       "10      Firmicutes       11    372 -99999.0\n",
       "11      Firmicutes       12   4255   4361.0\n",
       "12      Firmicutes       13    107   1667.0\n",
       "13      Firmicutes       14      ?    223.0\n",
       "14      Firmicutes       15    281   2377.0\n",
       "15  Proteobacteria        1   1638   3886.0\n",
       "16  Proteobacteria        2   2469   1821.0\n",
       "17  Proteobacteria        3    839    661.0\n",
       "18  Proteobacteria        4   4414     18.0\n",
       "19  Proteobacteria        5  12044     83.0"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/microbiome_missing.csv\").head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Above, Pandas recognized `NA` and an empty field as missing data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Taxon Patient Tissue  Stool\n",
       "0   False   False  False  False\n",
       "1   False   False  False  False\n",
       "2   False   False   True  False\n",
       "3   False   False  False  False\n",
       "4   False   False  False  False\n",
       "5   False   False  False  False\n",
       "6   False   False  False  False\n",
       "7   False   False  False  False\n",
       "8   False   False  False   True\n",
       "9   False   False  False  False\n",
       "10  False   False  False  False\n",
       "11  False   False  False  False\n",
       "12  False   False  False  False\n",
       "13  False   False  False  False\n",
       "14  False   False  False  False\n",
       "15  False   False  False  False\n",
       "16  False   False  False  False\n",
       "17  False   False  False  False\n",
       "18  False   False  False  False\n",
       "19  False   False  False  False"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.isnull(pd.read_csv(\"Data/microbiome_missing.csv\")).head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unfortunately, there will sometimes be inconsistency with the conventions for missing data. In this example, there is a question mark \"?\" and a large negative number where there should have been a positive integer. We can specify additional symbols with the `na_values` argument:\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>632.0</td>\n",
       "      <td>305.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2</td>\n",
       "      <td>136.0</td>\n",
       "      <td>4182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>703.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>4</td>\n",
       "      <td>408.0</td>\n",
       "      <td>3946.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>5</td>\n",
       "      <td>831.0</td>\n",
       "      <td>8605.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>6</td>\n",
       "      <td>693.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>7</td>\n",
       "      <td>718.0</td>\n",
       "      <td>717.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>8</td>\n",
       "      <td>173.0</td>\n",
       "      <td>33.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>9</td>\n",
       "      <td>228.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>10</td>\n",
       "      <td>162.0</td>\n",
       "      <td>3196.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>11</td>\n",
       "      <td>372.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>12</td>\n",
       "      <td>4255.0</td>\n",
       "      <td>4361.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>13</td>\n",
       "      <td>107.0</td>\n",
       "      <td>1667.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>15</td>\n",
       "      <td>281.0</td>\n",
       "      <td>2377.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>1638.0</td>\n",
       "      <td>3886.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>2</td>\n",
       "      <td>2469.0</td>\n",
       "      <td>1821.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3</td>\n",
       "      <td>839.0</td>\n",
       "      <td>661.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>4</td>\n",
       "      <td>4414.0</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>12044.0</td>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Taxon  Patient   Tissue   Stool\n",
       "0       Firmicutes        1    632.0   305.0\n",
       "1       Firmicutes        2    136.0  4182.0\n",
       "2       Firmicutes        3      NaN   703.0\n",
       "3       Firmicutes        4    408.0  3946.0\n",
       "4       Firmicutes        5    831.0  8605.0\n",
       "5       Firmicutes        6    693.0    50.0\n",
       "6       Firmicutes        7    718.0   717.0\n",
       "7       Firmicutes        8    173.0    33.0\n",
       "8       Firmicutes        9    228.0     NaN\n",
       "9       Firmicutes       10    162.0  3196.0\n",
       "10      Firmicutes       11    372.0     NaN\n",
       "11      Firmicutes       12   4255.0  4361.0\n",
       "12      Firmicutes       13    107.0  1667.0\n",
       "13      Firmicutes       14      NaN   223.0\n",
       "14      Firmicutes       15    281.0  2377.0\n",
       "15  Proteobacteria        1   1638.0  3886.0\n",
       "16  Proteobacteria        2   2469.0  1821.0\n",
       "17  Proteobacteria        3    839.0   661.0\n",
       "18  Proteobacteria        4   4414.0    18.0\n",
       "19  Proteobacteria        5  12044.0    83.0"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/microbiome_missing.csv\", na_values=['?', -99999]).head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These can be specified on a column-wise basis using an appropriate dict as the argument for `na_values`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Microsoft Excel\n",
    "\n",
    "Since so much financial and scientific data ends up in Excel spreadsheets (regrettably), Pandas' ability to directly import Excel spreadsheets is valuable. This support is contingent on having one or two dependencies (depending on what version of Excel file is being imported) installed: `xlrd` and `openpyxl` (these may be installed with either `pip` or `easy_install`).\n",
    "\n",
    "The read_excel convenience function in pandas imports a specific sheet from an Excel file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Acidiloba...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Acidiloba...</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   0   1\n",
       "0  Archaea \"Crenarchaeota\" Thermoprotei Acidiloba...   2\n",
       "1  Archaea \"Crenarchaeota\" Thermoprotei Acidiloba...  14\n",
       "2  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...  23\n",
       "3  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...   1\n",
       "4  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...   2"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mb = pd.read_excel('Data/microbiome/MID2.xls', sheetname='Sheet 1', header=None)\n",
    "mb.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are several other data formats that can be imported into Python and converted into DataFrames, with the help of buitl-in or third-party libraries. These include JSON, XML, HDF5, relational and non-relational databases, and various web APIs. These are beyond the scope of this tutorial, but are covered in [Python for Data Analysis](http://shop.oreilly.com/product/0636920023784.do)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pandas Fundamentals"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This section introduces the new user to the key functionality of Pandas that is required to use the software effectively.\n",
    "\n",
    "For some variety, we will leave our digestive tract bacteria behind and employ some baseball data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "      <th>stint</th>\n",
       "      <th>team</th>\n",
       "      <th>lg</th>\n",
       "      <th>g</th>\n",
       "      <th>ab</th>\n",
       "      <th>r</th>\n",
       "      <th>h</th>\n",
       "      <th>X2b</th>\n",
       "      <th>...</th>\n",
       "      <th>rbi</th>\n",
       "      <th>sb</th>\n",
       "      <th>cs</th>\n",
       "      <th>bb</th>\n",
       "      <th>so</th>\n",
       "      <th>ibb</th>\n",
       "      <th>hbp</th>\n",
       "      <th>sh</th>\n",
       "      <th>sf</th>\n",
       "      <th>gidp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88641</th>\n",
       "      <td>womacto01</td>\n",
       "      <td>2006</td>\n",
       "      <td>2</td>\n",
       "      <td>CHN</td>\n",
       "      <td>NL</td>\n",
       "      <td>19</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88643</th>\n",
       "      <td>schilcu01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>BOS</td>\n",
       "      <td>AL</td>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88645</th>\n",
       "      <td>myersmi01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88649</th>\n",
       "      <td>helliri01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>MIL</td>\n",
       "      <td>NL</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88650</th>\n",
       "      <td>johnsra05</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>33</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          player  year  stint team  lg   g  ab  r   h  X2b  ...   rbi   sb  \\\n",
       "id                                                          ...              \n",
       "88641  womacto01  2006      2  CHN  NL  19  50  6  14    1  ...   2.0  1.0   \n",
       "88643  schilcu01  2006      1  BOS  AL  31   2  0   1    0  ...   0.0  0.0   \n",
       "88645  myersmi01  2006      1  NYA  AL  62   0  0   0    0  ...   0.0  0.0   \n",
       "88649  helliri01  2006      1  MIL  NL  20   3  0   0    0  ...   0.0  0.0   \n",
       "88650  johnsra05  2006      1  NYA  AL  33   6  0   1    0  ...   0.0  0.0   \n",
       "\n",
       "        cs  bb   so  ibb  hbp   sh   sf  gidp  \n",
       "id                                             \n",
       "88641  1.0   4  4.0  0.0  0.0  3.0  0.0   0.0  \n",
       "88643  0.0   0  1.0  0.0  0.0  0.0  0.0   0.0  \n",
       "88645  0.0   0  0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "88649  0.0   0  2.0  0.0  0.0  0.0  0.0   0.0  \n",
       "88650  0.0   0  4.0  0.0  0.0  0.0  0.0   0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball = pd.read_csv(\"Data/baseball.csv\", index_col='id')\n",
    "baseball.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that we specified the `id` column as the index, since it appears to be a unique identifier. We could try to create a unique index ourselves by combining `player` and `year`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "      <th>stint</th>\n",
       "      <th>team</th>\n",
       "      <th>lg</th>\n",
       "      <th>g</th>\n",
       "      <th>ab</th>\n",
       "      <th>r</th>\n",
       "      <th>h</th>\n",
       "      <th>X2b</th>\n",
       "      <th>...</th>\n",
       "      <th>rbi</th>\n",
       "      <th>sb</th>\n",
       "      <th>cs</th>\n",
       "      <th>bb</th>\n",
       "      <th>so</th>\n",
       "      <th>ibb</th>\n",
       "      <th>hbp</th>\n",
       "      <th>sh</th>\n",
       "      <th>sf</th>\n",
       "      <th>gidp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>womacto012006</th>\n",
       "      <td>womacto01</td>\n",
       "      <td>2006</td>\n",
       "      <td>2</td>\n",
       "      <td>CHN</td>\n",
       "      <td>NL</td>\n",
       "      <td>19</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>schilcu012006</th>\n",
       "      <td>schilcu01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>BOS</td>\n",
       "      <td>AL</td>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>myersmi012006</th>\n",
       "      <td>myersmi01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>helliri012006</th>\n",
       "      <td>helliri01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>MIL</td>\n",
       "      <td>NL</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>johnsra052006</th>\n",
       "      <td>johnsra05</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>33</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  player  year  stint team  lg   g  ab  r   h  X2b  ...   rbi  \\\n",
       "womacto012006  womacto01  2006      2  CHN  NL  19  50  6  14    1  ...   2.0   \n",
       "schilcu012006  schilcu01  2006      1  BOS  AL  31   2  0   1    0  ...   0.0   \n",
       "myersmi012006  myersmi01  2006      1  NYA  AL  62   0  0   0    0  ...   0.0   \n",
       "helliri012006  helliri01  2006      1  MIL  NL  20   3  0   0    0  ...   0.0   \n",
       "johnsra052006  johnsra05  2006      1  NYA  AL  33   6  0   1    0  ...   0.0   \n",
       "\n",
       "                sb   cs  bb   so  ibb  hbp   sh   sf  gidp  \n",
       "womacto012006  1.0  1.0   4  4.0  0.0  0.0  3.0  0.0   0.0  \n",
       "schilcu012006  0.0  0.0   0  1.0  0.0  0.0  0.0  0.0   0.0  \n",
       "myersmi012006  0.0  0.0   0  0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "helliri012006  0.0  0.0   0  2.0  0.0  0.0  0.0  0.0   0.0  \n",
       "johnsra052006  0.0  0.0   0  4.0  0.0  0.0  0.0  0.0   0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "player_id = baseball.player + baseball.year.astype(str)\n",
    "baseball_newind = baseball.copy()\n",
    "baseball_newind.index = player_id\n",
    "baseball_newind.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This looks okay, but let's check:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball_newind.index.is_unique"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So, indices need not be unique. Our choice is not unique because some players change teams within years."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "hernaro012007    2\n",
       "sweenma012007    2\n",
       "wellsda012007    2\n",
       "claytro012007    2\n",
       "loftoke012007    2\n",
       "coninje012007    2\n",
       "gomezch022007    2\n",
       "wickmbo012007    2\n",
       "benitar012007    2\n",
       "francju012007    2\n",
       "cirilje012007    2\n",
       "trachst012007    2\n",
       "sheffga012007    1\n",
       "ramirma022007    1\n",
       "vizquom012007    1\n",
       "griffke022007    1\n",
       "thomafr042007    1\n",
       "wakefti012007    1\n",
       "williwo022007    1\n",
       "gonzalu012006    1\n",
       "kleskry012007    1\n",
       "graffto012007    1\n",
       "stantmi022007    1\n",
       "ausmubr012007    1\n",
       "witasja012007    1\n",
       "jonesto022007    1\n",
       "cormirh012007    1\n",
       "parkch012007     1\n",
       "guarded012007    1\n",
       "schilcu012006    1\n",
       "                ..\n",
       "embreal012007    1\n",
       "johnsra052007    1\n",
       "bondsba012007    1\n",
       "tavarju012007    1\n",
       "delgaca012007    1\n",
       "clemero022007    1\n",
       "schmija012007    1\n",
       "gonzalu012007    1\n",
       "womacto012006    1\n",
       "smoltjo012007    1\n",
       "zaungr012007     1\n",
       "seleaa012007     1\n",
       "mabryjo012007    1\n",
       "maddugr012007    1\n",
       "finlest012007    1\n",
       "edmonji012007    1\n",
       "gordoto012007    1\n",
       "suppaje012007    1\n",
       "seaneru012007    1\n",
       "alomasa022007    1\n",
       "loaizes012007    1\n",
       "easleda012007    1\n",
       "aloumo012007     1\n",
       "walketo042007    1\n",
       "villoro012007    1\n",
       "whitero022007    1\n",
       "stairma012007    1\n",
       "timlimi012007    1\n",
       "martipe022007    1\n",
       "greensh012007    1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.Series(baseball_newind.index).value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The most important consequence of a non-unique index is that indexing by label will return multiple values for some labels:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "      <th>stint</th>\n",
       "      <th>team</th>\n",
       "      <th>lg</th>\n",
       "      <th>g</th>\n",
       "      <th>ab</th>\n",
       "      <th>r</th>\n",
       "      <th>h</th>\n",
       "      <th>X2b</th>\n",
       "      <th>...</th>\n",
       "      <th>rbi</th>\n",
       "      <th>sb</th>\n",
       "      <th>cs</th>\n",
       "      <th>bb</th>\n",
       "      <th>so</th>\n",
       "      <th>ibb</th>\n",
       "      <th>hbp</th>\n",
       "      <th>sh</th>\n",
       "      <th>sf</th>\n",
       "      <th>gidp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>wickmbo012007</th>\n",
       "      <td>wickmbo01</td>\n",
       "      <td>2007</td>\n",
       "      <td>2</td>\n",
       "      <td>ARI</td>\n",
       "      <td>NL</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wickmbo012007</th>\n",
       "      <td>wickmbo01</td>\n",
       "      <td>2007</td>\n",
       "      <td>1</td>\n",
       "      <td>ATL</td>\n",
       "      <td>NL</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  player  year  stint team  lg   g  ab  r  h  X2b  ...   rbi  \\\n",
       "wickmbo012007  wickmbo01  2007      2  ARI  NL   8   0  0  0    0  ...   0.0   \n",
       "wickmbo012007  wickmbo01  2007      1  ATL  NL  47   0  0  0    0  ...   0.0   \n",
       "\n",
       "                sb   cs  bb   so  ibb  hbp   sh   sf  gidp  \n",
       "wickmbo012007  0.0  0.0   0  0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "wickmbo012007  0.0  0.0   0  0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "\n",
       "[2 rows x 22 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball_newind.loc['wickmbo012007']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will learn more about indexing below."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can create a truly unique index by combining `player`, `team` and `year`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "      <th>stint</th>\n",
       "      <th>team</th>\n",
       "      <th>lg</th>\n",
       "      <th>g</th>\n",
       "      <th>ab</th>\n",
       "      <th>r</th>\n",
       "      <th>h</th>\n",
       "      <th>X2b</th>\n",
       "      <th>...</th>\n",
       "      <th>rbi</th>\n",
       "      <th>sb</th>\n",
       "      <th>cs</th>\n",
       "      <th>bb</th>\n",
       "      <th>so</th>\n",
       "      <th>ibb</th>\n",
       "      <th>hbp</th>\n",
       "      <th>sh</th>\n",
       "      <th>sf</th>\n",
       "      <th>gidp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>womacto01CHN2006</th>\n",
       "      <td>womacto01</td>\n",
       "      <td>2006</td>\n",
       "      <td>2</td>\n",
       "      <td>CHN</td>\n",
       "      <td>NL</td>\n",
       "      <td>19</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>schilcu01BOS2006</th>\n",
       "      <td>schilcu01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>BOS</td>\n",
       "      <td>AL</td>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>myersmi01NYA2006</th>\n",
       "      <td>myersmi01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>helliri01MIL2006</th>\n",
       "      <td>helliri01</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>MIL</td>\n",
       "      <td>NL</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>johnsra05NYA2006</th>\n",
       "      <td>johnsra05</td>\n",
       "      <td>2006</td>\n",
       "      <td>1</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>33</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     player  year  stint team  lg   g  ab  r   h  X2b  ...   \\\n",
       "womacto01CHN2006  womacto01  2006      2  CHN  NL  19  50  6  14    1  ...    \n",
       "schilcu01BOS2006  schilcu01  2006      1  BOS  AL  31   2  0   1    0  ...    \n",
       "myersmi01NYA2006  myersmi01  2006      1  NYA  AL  62   0  0   0    0  ...    \n",
       "helliri01MIL2006  helliri01  2006      1  MIL  NL  20   3  0   0    0  ...    \n",
       "johnsra05NYA2006  johnsra05  2006      1  NYA  AL  33   6  0   1    0  ...    \n",
       "\n",
       "                  rbi   sb   cs  bb   so  ibb  hbp   sh   sf  gidp  \n",
       "womacto01CHN2006  2.0  1.0  1.0   4  4.0  0.0  0.0  3.0  0.0   0.0  \n",
       "schilcu01BOS2006  0.0  0.0  0.0   0  1.0  0.0  0.0  0.0  0.0   0.0  \n",
       "myersmi01NYA2006  0.0  0.0  0.0   0  0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "helliri01MIL2006  0.0  0.0  0.0   0  2.0  0.0  0.0  0.0  0.0   0.0  \n",
       "johnsra05NYA2006  0.0  0.0  0.0   0  4.0  0.0  0.0  0.0  0.0   0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "player_unique = baseball.player + baseball.team + baseball.year.astype(str)\n",
    "baseball_newind = baseball.copy()\n",
    "baseball_newind.index = player_unique\n",
    "baseball_newind.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball_newind.index.is_unique"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can create meaningful indices more easily using a hierarchical index; for now, we will stick with the numeric `id` field as our index."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulating indices\n",
    "\n",
    "**Reindexing** allows users to manipulate the data labels in a DataFrame. It forces a DataFrame to conform to the new index, and optionally, fill in missing data if requested.\n",
    "\n",
    "A simple use of `reindex` is to alter the order of the rows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "      <th>stint</th>\n",
       "      <th>team</th>\n",
       "      <th>lg</th>\n",
       "      <th>g</th>\n",
       "      <th>ab</th>\n",
       "      <th>r</th>\n",
       "      <th>h</th>\n",
       "      <th>X2b</th>\n",
       "      <th>...</th>\n",
       "      <th>rbi</th>\n",
       "      <th>sb</th>\n",
       "      <th>cs</th>\n",
       "      <th>bb</th>\n",
       "      <th>so</th>\n",
       "      <th>ibb</th>\n",
       "      <th>hbp</th>\n",
       "      <th>sh</th>\n",
       "      <th>sf</th>\n",
       "      <th>gidp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89534</th>\n",
       "      <td>alomasa02</td>\n",
       "      <td>2007</td>\n",
       "      <td>1</td>\n",
       "      <td>NYN</td>\n",
       "      <td>NL</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89533</th>\n",
       "      <td>aloumo01</td>\n",
       "      <td>2007</td>\n",
       "      <td>1</td>\n",
       "      <td>NYN</td>\n",
       "      <td>NL</td>\n",
       "      <td>87</td>\n",
       "      <td>328</td>\n",
       "      <td>51</td>\n",
       "      <td>112</td>\n",
       "      <td>19</td>\n",
       "      <td>...</td>\n",
       "      <td>49.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>30.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89530</th>\n",
       "      <td>ausmubr01</td>\n",
       "      <td>2007</td>\n",
       "      <td>1</td>\n",
       "      <td>HOU</td>\n",
       "      <td>NL</td>\n",
       "      <td>117</td>\n",
       "      <td>349</td>\n",
       "      <td>38</td>\n",
       "      <td>82</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>37</td>\n",
       "      <td>74.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89526</th>\n",
       "      <td>benitar01</td>\n",
       "      <td>2007</td>\n",
       "      <td>1</td>\n",
       "      <td>SFN</td>\n",
       "      <td>NL</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89525</th>\n",
       "      <td>benitar01</td>\n",
       "      <td>2007</td>\n",
       "      <td>2</td>\n",
       "      <td>FLO</td>\n",
       "      <td>NL</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          player  year  stint team  lg    g   ab   r    h  X2b  ...    rbi  \\\n",
       "id                                                              ...          \n",
       "89534  alomasa02  2007      1  NYN  NL    8   22   1    3    1  ...    0.0   \n",
       "89533   aloumo01  2007      1  NYN  NL   87  328  51  112   19  ...   49.0   \n",
       "89530  ausmubr01  2007      1  HOU  NL  117  349  38   82   16  ...   25.0   \n",
       "89526  benitar01  2007      1  SFN  NL   19    0   0    0    0  ...    0.0   \n",
       "89525  benitar01  2007      2  FLO  NL   34    0   0    0    0  ...    0.0   \n",
       "\n",
       "        sb   cs  bb    so  ibb  hbp   sh   sf  gidp  \n",
       "id                                                   \n",
       "89534  0.0  0.0   0   3.0  0.0  0.0  0.0  0.0   0.0  \n",
       "89533  3.0  0.0  27  30.0  5.0  2.0  0.0  3.0  13.0  \n",
       "89530  6.0  1.0  37  74.0  3.0  6.0  4.0  1.0  11.0  \n",
       "89526  0.0  0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "89525  0.0  0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball.reindex(baseball.index[::-1]).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that the `id` index is not sequential. Say we wanted to populate the table with every `id` value. We could specify and index that is a sequence from the first to the last `id` numbers in the database, and Pandas would fill in the missing data with `NaN` values:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "      <th>stint</th>\n",
       "      <th>team</th>\n",
       "      <th>lg</th>\n",
       "      <th>g</th>\n",
       "      <th>ab</th>\n",
       "      <th>r</th>\n",
       "      <th>h</th>\n",
       "      <th>X2b</th>\n",
       "      <th>...</th>\n",
       "      <th>rbi</th>\n",
       "      <th>sb</th>\n",
       "      <th>cs</th>\n",
       "      <th>bb</th>\n",
       "      <th>so</th>\n",
       "      <th>ibb</th>\n",
       "      <th>hbp</th>\n",
       "      <th>sh</th>\n",
       "      <th>sf</th>\n",
       "      <th>gidp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88641</th>\n",
       "      <td>womacto01</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CHN</td>\n",
       "      <td>NL</td>\n",
       "      <td>19.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88642</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88643</th>\n",
       "      <td>schilcu01</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BOS</td>\n",
       "      <td>AL</td>\n",
       "      <td>31.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88644</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88645</th>\n",
       "      <td>myersmi01</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NYA</td>\n",
       "      <td>AL</td>\n",
       "      <td>62.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          player    year  stint team   lg     g    ab    r     h  X2b  ...   \\\n",
       "id                                                                     ...    \n",
       "88641  womacto01  2006.0    2.0  CHN   NL  19.0  50.0  6.0  14.0  1.0  ...    \n",
       "88642        NaN     NaN    NaN  NaN  NaN   NaN   NaN  NaN   NaN  NaN  ...    \n",
       "88643  schilcu01  2006.0    1.0  BOS   AL  31.0   2.0  0.0   1.0  0.0  ...    \n",
       "88644        NaN     NaN    NaN  NaN  NaN   NaN   NaN  NaN   NaN  NaN  ...    \n",
       "88645  myersmi01  2006.0    1.0  NYA   AL  62.0   0.0  0.0   0.0  0.0  ...    \n",
       "\n",
       "       rbi   sb   cs   bb   so  ibb  hbp   sh   sf  gidp  \n",
       "id                                                        \n",
       "88641  2.0  1.0  1.0  4.0  4.0  0.0  0.0  3.0  0.0   0.0  \n",
       "88642  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  \n",
       "88643  0.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0   0.0  \n",
       "88644  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  \n",
       "88645  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_range = range(baseball.index.values.min(), baseball.index.values.max())\n",
    "baseball.reindex(id_range).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Missing values can be filled as desired, either with selected values, or by rule:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88641</th>\n",
       "      <td>womacto01</td>\n",
       "      <td>2006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88642</th>\n",
       "      <td>womacto01</td>\n",
       "      <td>2006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88643</th>\n",
       "      <td>schilcu01</td>\n",
       "      <td>2006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88644</th>\n",
       "      <td>schilcu01</td>\n",
       "      <td>2006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88645</th>\n",
       "      <td>myersmi01</td>\n",
       "      <td>2006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          player  year\n",
       "id                    \n",
       "88641  womacto01  2006\n",
       "88642  womacto01  2006\n",
       "88643  schilcu01  2006\n",
       "88644  schilcu01  2006\n",
       "88645  myersmi01  2006"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball.reindex(id_range, method='ffill', columns=['player','year']).head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>player</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88641</th>\n",
       "      <td>womacto01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88642</th>\n",
       "      <td>charliebrown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88643</th>\n",
       "      <td>schilcu01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88644</th>\n",
       "      <td>charliebrown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88645</th>\n",
       "      <td>myersmi01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             player\n",
       "id                 \n",
       "88641     womacto01\n",
       "88642  charliebrown\n",
       "88643     schilcu01\n",
       "88644  charliebrown\n",
       "88645     myersmi01"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball.reindex(id_range, fill_value='charliebrown', columns=['player']).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Keep in mind that `reindex` does not work if we pass a non-unique index series."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can remove rows or columns via the `drop` method:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(100, 22)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseball.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-79-906241f84dc5>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mbaseball\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdrop\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m89525\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m89526\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/core/displayhook.py\u001b[0m in \u001b[0;36m__call__\u001b[1;34m(self, result)\u001b[0m\n\u001b[0;32m    244\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstart_displayhook\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    245\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwrite_output_prompt\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 246\u001b[1;33m             \u001b[0mformat_dict\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmd_dict\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcompute_format_data\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    247\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate_user_ns\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    248\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfill_exec_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/core/displayhook.py\u001b[0m in \u001b[0;36mcompute_format_data\u001b[1;34m(self, result)\u001b[0m\n\u001b[0;32m    150\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    151\u001b[0m         \"\"\"\n\u001b[1;32m--> 152\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshell\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdisplay_formatter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    153\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    154\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mwrite_format_data\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mformat_dict\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmd_dict\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/core/formatters.py\u001b[0m in \u001b[0;36mformat\u001b[1;34m(self, obj, include, exclude)\u001b[0m\n\u001b[0;32m    175\u001b[0m             \u001b[0mmd\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    176\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 177\u001b[1;33m                 \u001b[0mdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mformatter\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    178\u001b[0m             \u001b[1;32mexcept\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    179\u001b[0m                 \u001b[1;31m# FIXME: log the exception\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<decorator-gen-10>\u001b[0m in \u001b[0;36m__call__\u001b[1;34m(self, obj)\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/core/formatters.py\u001b[0m in \u001b[0;36mcatch_format_error\u001b[1;34m(method, self, *args, **kwargs)\u001b[0m\n\u001b[0;32m    220\u001b[0m     \u001b[1;34m\"\"\"show traceback on failed format call\"\"\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    221\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 222\u001b[1;33m         \u001b[0mr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmethod\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    223\u001b[0m     \u001b[1;32mexcept\u001b[0m \u001b[0mNotImplementedError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    224\u001b[0m         \u001b[1;31m# don't warn on NotImplementedErrors\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/core/formatters.py\u001b[0m in \u001b[0;36m__call__\u001b[1;34m(self, obj)\u001b[0m\n\u001b[0;32m    697\u001b[0m                 \u001b[0mtype_pprinters\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtype_printers\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    698\u001b[0m                 deferred_pprinters=self.deferred_printers)\n\u001b[1;32m--> 699\u001b[1;33m             \u001b[0mprinter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpretty\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    700\u001b[0m             \u001b[0mprinter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    701\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mstream\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgetvalue\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/lib/pretty.py\u001b[0m in \u001b[0;36mpretty\u001b[1;34m(self, obj)\u001b[0m\n\u001b[0;32m    381\u001b[0m                             \u001b[1;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmeth\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    382\u001b[0m                                 \u001b[1;32mreturn\u001b[0m \u001b[0mmeth\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcycle\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 383\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0m_default_pprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcycle\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    384\u001b[0m         \u001b[1;32mfinally\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    385\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mend_group\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/lib/pretty.py\u001b[0m in \u001b[0;36m_default_pprint\u001b[1;34m(obj, p, cycle)\u001b[0m\n\u001b[0;32m    501\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0m_safe_getattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mklass\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'__repr__'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32min\u001b[0m \u001b[0m_baseclass_reprs\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    502\u001b[0m         \u001b[1;31m# A user-provided repr. Find newlines and replace them with p.break_()\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 503\u001b[1;33m         \u001b[0m_repr_pprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mp\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcycle\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    504\u001b[0m         \u001b[1;32mreturn\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    505\u001b[0m     \u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbegin_group\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'<'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/IPython/lib/pretty.py\u001b[0m in \u001b[0;36m_repr_pprint\u001b[1;34m(obj, p, cycle)\u001b[0m\n\u001b[0;32m    692\u001b[0m     \u001b[1;34m\"\"\"A pprint that just redirects to the normal repr function.\"\"\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    693\u001b[0m     \u001b[1;31m# Find newlines and replace them with p.break_()\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 694\u001b[1;33m     \u001b[0moutput\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mrepr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    695\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0midx\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0moutput_line\u001b[0m \u001b[1;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0moutput\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplitlines\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    696\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0midx\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/core/base.py\u001b[0m in \u001b[0;36m__repr__\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m     65\u001b[0m         \u001b[0mYields\u001b[0m \u001b[0mBytestring\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mPy2\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mUnicode\u001b[0m \u001b[0mString\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mpy3\u001b[0m\u001b[1;33m.\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     66\u001b[0m         \"\"\"\n\u001b[1;32m---> 67\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     68\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     69\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/core/base.py\u001b[0m in \u001b[0;36m__str__\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m     44\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     45\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mcompat\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mPY3\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 46\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__unicode__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     47\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__bytes__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     48\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__unicode__\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    533\u001b[0m             \u001b[0mwidth\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    534\u001b[0m         self.to_string(buf=buf, max_rows=max_rows, max_cols=max_cols,\n\u001b[1;32m--> 535\u001b[1;33m                        line_width=width, show_dimensions=show_dimensions)\n\u001b[0m\u001b[0;32m    536\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    537\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mbuf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgetvalue\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mto_string\u001b[1;34m(self, buf, columns, col_space, header, index, na_rep, formatters, float_format, sparsify, index_names, justify, line_width, max_rows, max_cols, show_dimensions)\u001b[0m\n\u001b[0;32m   1486\u001b[0m                                            \u001b[0mmax_cols\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmax_cols\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1487\u001b[0m                                            show_dimensions=show_dimensions)\n\u001b[1;32m-> 1488\u001b[1;33m         \u001b[0mformatter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mto_string\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1489\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1490\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mbuf\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py\u001b[0m in \u001b[0;36mto_string\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    547\u001b[0m             \u001b[0mtext\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0minfo_line\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    548\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 549\u001b[1;33m             \u001b[0mstrcols\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_to_str_columns\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    550\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mline_width\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m  \u001b[1;31m# no need to wrap around just print\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    551\u001b[0m                 \u001b[1;31m# the whole frame\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py\u001b[0m in \u001b[0;36m_to_str_columns\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    477\u001b[0m                 fmt_values = _make_fixed_width(fmt_values, self.justify,\n\u001b[0;32m    478\u001b[0m                                                \u001b[0mminimum\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmax_colwidth\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 479\u001b[1;33m                                                adj=self.adj)\n\u001b[0m\u001b[0;32m    480\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    481\u001b[0m                 max_len = max(np.max([self.adj.len(x) for x in fmt_values]),\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py\u001b[0m in \u001b[0;36m_make_fixed_width\u001b[1;34m(strings, justify, minimum, adj)\u001b[0m\n\u001b[0;32m   2407\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mx\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2408\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2409\u001b[1;33m     \u001b[0mstrings\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mjust\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mstrings\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2410\u001b[0m     \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0madj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjustify\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mstrings\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_len\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mjustify\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2411\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/lukas/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[1;34m(.0)\u001b[0m\n\u001b[0;32m   2407\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mx\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2408\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2409\u001b[1;33m     \u001b[0mstrings\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mjust\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mstrings\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2410\u001b[0m     \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0madj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjustify\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mstrings\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_len\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mjustify\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2411\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "baseball.drop([89525, 89526])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.drop(['ibb','hbp'], axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Indexing and Selection\n",
    "\n",
    "Indexing works analogously to indexing in NumPy arrays, except we can use the labels in the `Index` object to extract values in addition to arrays of integers."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Sample Series object\n",
    "hits = baseball_newind.h\n",
    "hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Numpy-style indexing\n",
    "hits[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Indexing by label\n",
    "hits[['womacto01CHN2006','schilcu01BOS2006']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also slice with data labels, since they have an intrinsic order within the Index:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hits['womacto01CHN2006':'gonzalu01ARI2006']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hits['womacto01CHN2006':'gonzalu01ARI2006'] = 5\n",
    "hits"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In a `DataFrame` we can slice along either or both axes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind[['h','ab']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind[baseball_newind.ab>500]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For a more concise (and readable) syntax, we can use the new `query` method to perform selection on a `DataFrame`. Instead of having to type the fully-specified column, we can simply pass a string that describes what to select. The query above is then simply:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.query('ab > 500')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `DataFrame.index` and `DataFrame.columns` are placed in the query namespace by default. If you want to refer to a variable in the current namespace, you can prefix the variable with `@`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "min_ab = 450"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.query('ab > @min_ab')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The indexing field `loc` allows us to select subsets of rows and columns in an intuitive way:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.loc['gonzalu01ARI2006', ['h','X2b', 'X3b', 'hr']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.loc[:'myersmi01NYA2006', 'hr']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In addition to using `loc` to select rows and columns by **label**, pandas also allows indexing by **position** using the `iloc` attribute.\n",
    "\n",
    "So, we can query rows and columns by absolute position, rather than by name:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.iloc[:5, 5:8]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exercise 3\n",
    "\n",
    "You can use the `isin` method query a DataFrame based upon a list of values as follows: \n",
    "\n",
    "    data['phylum'].isin(['Firmacutes', 'Bacteroidetes'])\n",
    "\n",
    "Use `isin` to find all players that played for the Los Angeles Dodgers (LAN) or the San Francisco Giants (SFN). How many records contain these values?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "---------------------\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball[baseball['team'].isin(['LAN', 'SFN'])]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Operations\n",
    "\n",
    "`DataFrame` and `Series` objects allow for several operations to take place either on a single object, or between two or more objects.\n",
    "\n",
    "For example, we can perform arithmetic on the elements of two objects, such as combining baseball statistics across years. First, let's (artificially) construct two Series, consisting of home runs hit in years 2006 and 2007, respectively:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hr2006 = baseball.loc[baseball.year==2006, 'hr']\n",
    "hr2006.index = baseball.player[baseball.year==2006]\n",
    "\n",
    "hr2007 = baseball.loc[baseball.year==2007, 'hr']\n",
    "hr2007.index = baseball.player[baseball.year==2007]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hr2007"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, let's add them together, in hopes of getting 2-year home run totals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hr_total = hr2006 + hr2007\n",
    "hr_total"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pandas' data alignment places `NaN` values for labels that do not overlap in the two Series. In fact, there are only 6 players that occur in both years."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hr_total[hr_total.notnull()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "While we do want the operation to honor the data labels in this way, we probably do not want the missing values to be filled with `NaN`. We can use the `add` method to calculate player home run totals by using the `fill_value` argument to insert a zero for home runs where labels do not overlap:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hr2007.add(hr2006, fill_value=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Operations can also be **broadcast** between rows or columns.\n",
    "\n",
    "For example, if we subtract the maximum number of home runs hit from the `hr` column, we get how many fewer than the maximum were hit by each player:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.hr - baseball.hr.max()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Or, looking at things row-wise, we can see how a particular player compares with the rest of the group with respect to important statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.loc[89521, \"player\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stats = baseball[['h','X2b', 'X3b', 'hr']]\n",
    "diff = stats - stats.loc[88641]\n",
    "diff[:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also apply functions to each column or row of a `DataFrame`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stats.apply(np.median)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def range_calc(x):\n",
    "    return x.max() - x.min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stat_range = lambda x: x.max() - x.min()\n",
    "stats.apply(stat_range)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lets use apply to calculate a meaningful baseball statistics, [slugging percentage](https://en.wikipedia.org/wiki/Slugging_percentage):\n",
    "\n",
    "$$SLG = \\frac{1B + (2 \\times 2B) + (3 \\times 3B) + (4 \\times HR)}{AB}$$\n",
    "\n",
    "And just for fun, we will format the resulting estimate."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def slugging(x): \n",
    "    bases = x['h']-x['X2b']-x['X3b']-x['hr'] + 2*x['X2b'] + 3*x['X3b'] + 4*x['hr']\n",
    "    ab = x['ab']+1e-6\n",
    "    \n",
    "    return bases/ab\n",
    "\n",
    "baseball.apply(slugging, axis=1).round(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sorting and Ranking\n",
    "\n",
    "Pandas objects include methods for re-ordering data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.sort_index().head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.sort_index(ascending=False).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Try sorting the **columns** instead of the rows, in ascending order:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_newind.sort_index(axis=1).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also use `sort_values` to sort a `Series` by value, rather than by label."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.hr.sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For a `DataFrame`, we can sort according to the values of one or more columns using the `by` argument of `sort_values`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball[['player','sb','cs']].sort_values(ascending=[False,True], \n",
    "                                           by=['sb', 'cs']).head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Ranking** does not re-arrange data, but instead returns an index that ranks each value relative to others in the Series."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.hr.rank()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ties are assigned the mean value of the tied ranks, which may result in decimal values."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "pd.Series([100,100]).rank()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Alternatively, you can break ties via one of several methods, such as by the order in which they occur in the dataset:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.hr.rank(method='first')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calling the `DataFrame`'s `rank` method results in the ranks of all columns:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.rank(ascending=False).head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball[['r','h','hr']].rank(ascending=False).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exercise 4\n",
    "\n",
    "Calculate **on base percentage** for each player, and return the ordered series of estimates.\n",
    "\n",
    "$$OBP = \\frac{H + BB + HBP}{AB + BB + HBP + SF}$$"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "-----------------------\n",
    "define the function and appy it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def on_base_perc(pl):\n",
    "    nominator = pl['h'] + pl['bb'] + pl['hbp'] #H+BB+HBP\n",
    "    denom = pl['ab'] + pl['bb'] +pl['hbp'] +pl['sf'] +1e-6\n",
    "    return nominator / denom\n",
    "baseball.apply(on_base_perc, axis=1).round(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "and again but ordered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.apply(on_base_perc, axis=1).round(3).sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Hierarchical indexing\n",
    "\n",
    "In the baseball example, I was forced to combine 3 fields to obtain a unique index that was not simply an integer value. A more elegant way to have done this would be to create a hierarchical index from the three fields."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_h = baseball.set_index(['year', 'team', 'player'])\n",
    "baseball_h.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This index is a `MultiIndex` object that consists of a sequence of tuples, the elements of which is some combination of the three columns used to create the index. Where there are multiple repeated values, Pandas does not print the repeats, making it easy to identify groups of values."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_h.index[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_h.index.is_unique"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Try using this hierarchical index to retrieve Julio Franco (`francju01`), who played for the Atlanta Braves (`ATL`) in 2007:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball_h.loc[(2007, 'ATL', 'francju01')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Recall earlier we imported some microbiome data using two index columns. This created a 2-level hierarchical index:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb = pd.read_csv(\"Data/microbiome.csv\", index_col=['Taxon','Patient'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With a hierachical index, we can select subsets of the data based on a *partial* index:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.loc['Proteobacteria']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Hierarchical indices can be created on either or both axes. Here is a trivial example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "frame = pd.DataFrame(np.arange(12).reshape(( 4, 3)), \n",
    "                  index =[['a', 'a', 'b', 'b'], [1, 2, 1, 2]], \n",
    "                  columns =[['Ohio', 'Ohio', 'Colorado'], ['Green', 'Red', 'Green']])\n",
    "\n",
    "frame"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you want to get fancy, both the row and column indices themselves can be given names:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "frame.index.names = ['key1', 'key2']\n",
    "frame.columns.names = ['state', 'color']\n",
    "frame"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With this, we can do all sorts of custom indexing:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "frame.loc['a', 'Ohio']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Try retrieving the value corresponding to `b2` in `Colorado`:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------------------------\n",
    "fetch b2 and then Colorado"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "frame.loc['b', 2]['Colorado']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Additionally, the order of the set of indices in a hierarchical `MultiIndex` can be changed by swapping them pairwise:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.swaplevel('Patient', 'Taxon').head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data can also be sorted by any index level, using `sortlevel`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.sortlevel('Patient', ascending=False).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Missing data\n",
    "\n",
    "The occurence of missing data is so prevalent that it pays to use tools like Pandas, which seamlessly integrates missing data handling so that it can be dealt with easily, and in the manner required by the analysis at hand.\n",
    "\n",
    "Missing data are represented in `Series` and `DataFrame` objects by the `NaN` floating point value. However, `None` is also treated as missing, since it is commonly used as such in other contexts (*e.g.* NumPy)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "foo = pd.Series([np.nan, -3, None, 'foobar'])\n",
    "foo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "foo.isnull()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Missing values may be dropped or indexed out:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.isnull()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2[bacteria2.notnull()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "By default, `dropna` drops entire rows in which one or more values are missing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.dropna()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This can be overridden by passing the `how='all'` argument, which only drops a row when every field is a missing value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.dropna(how='all')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This can be customized further by specifying how many values need to be present before a row is dropped via the `thresh` argument."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.loc[7, 'year'] = np.nan\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.dropna(thresh=5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is typically used in time series applications, where there are repeated measurements that are incomplete for some subjects."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exercise 5\n",
    "\n",
    "Try using the `axis` argument to drop columns with missing values:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.dropna(axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rather than omitting missing data from an analysis, in some cases it may be suitable to fill the missing value in, either with a default value (such as zero) or a value that is either imputed or carried forward/backward from similar data points. We can do this programmatically in Pandas with the `fillna` argument."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.fillna({'year': 2013, 'treatment':2})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that `fillna` by default returns a new object with the desired filling behavior, rather than changing the `Series` or  `DataFrame` in place (**in general, we like to do this, by the way!**)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can alter values in-place using `inplace=True`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.year.fillna(2013, inplace=True)\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Missing values can also be interpolated, using any one of a variety of methods:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.fillna(method='bfill')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data summarization\n",
    "\n",
    "We often wish to summarize data in `Series` or `DataFrame` objects, so that they can more easily be understood or compared with similar data. The NumPy package contains several functions that are useful here, but several summarization or reduction methods are built into Pandas data structures."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clearly, `sum` is more meaningful for some columns than others. For methods like `mean` for which application to string variables is not just meaningless, but impossible, these columns are automatically exculded:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The important difference between NumPy's functions and Pandas' methods is that the latter have built-in support for handling missing data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sometimes we may not want to ignore missing values, and allow the `nan` to propagate."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bacteria2.mean(skipna=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Passing `axis=1` will summarize over rows instead of columns, which only makes sense in certain situations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "extra_bases = baseball[['X2b','X3b','hr']].sum(axis=1)\n",
    "extra_bases.sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A useful summarization that gives a quick snapshot of multiple statistics for a `Series` or `DataFrame` is `describe`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`describe` can detect non-numeric data and sometimes yield useful information about it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.player.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also calculate summary statistics *across* multiple columns, for example, correlation and covariance.\n",
    "\n",
    "$$cov(x,y) = \\sum_i (x_i - \\bar{x})(y_i - \\bar{y})$$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.hr.cov(baseball.X2b)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "$$corr(x,y) = \\frac{cov(x,y)}{(n-1)s_x s_y} = \\frac{\\sum_i (x_i - \\bar{x})(y_i - \\bar{y})}{\\sqrt{\\sum_i (x_i - \\bar{x})^2 \\sum_i (y_i - \\bar{y})^2}}$$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.hr.corr(baseball.X2b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.ab.corr(baseball.h)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Try running `corr` on the entire `baseball` DataFrame to see what is returned:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.corr()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "it returns the correlation matrix for all features"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If we have a `DataFrame` with a hierarchical index (or indices), summary statistics can be applied with respect to any of the index levels:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.sum(level='Taxon')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Writing Data to Files\n",
    "\n",
    "As well as being able to read several data input formats, Pandas can also export data to a variety of storage formats. We will bring your attention to just a couple of these."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mb.to_csv(\"mb.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `to_csv` method writes a `DataFrame` to a comma-separated values (csv) file. You can specify custom delimiters (via `sep` argument), how missing values are written (via `na_rep` argument), whether the index is writen (via `index` argument), whether the header is included (via `header` argument), among other options."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An efficient way of storing data to disk is in binary format. Pandas supports this using Python’s built-in pickle serialization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "baseball.to_pickle(\"baseball_pickle\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The complement to `to_pickle` is the `read_pickle` function, which restores the pickle to a `DataFrame` or `Series`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "pd.read_pickle(\"baseball_pickle\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As Wes warns in his book, it is recommended that binary storage of data via pickle only be used as a temporary storage format, in situations where speed is relevant. This is because there is no guarantee that the pickle format will not change with future versions of Python."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Advanced Exercise: Compiling Ebola Data\n",
    "\n",
    "The `Data/ebola` folder contains summarized reports of Ebola cases from three countries during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "From these data files, use pandas to import them and create a single data frame that includes the daily totals of new cases and deaths for each country."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### we have:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob # to find all files in folder\n",
    "pd.options.mode.chained_assignment = None  # default='warn'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "guinea_data  liberia_data  LICENSE  sl_data\r\n"
     ]
    }
   ],
   "source": [
    "!ls Data/ebola"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2014-08-04.csv\t2014-09-02.csv\t2014-09-11.csv\t2014-09-21.csv\t2014-09-30.csv\r\n",
      "2014-08-26.csv\t2014-09-04.csv\t2014-09-14.csv\t2014-09-22.csv\t2014-10-01.csv\r\n",
      "2014-08-27.csv\t2014-09-07.csv\t2014-09-16.csv\t2014-09-23.csv\r\n",
      "2014-08-30.csv\t2014-09-08.csv\t2014-09-17.csv\t2014-09-24.csv\r\n",
      "2014-08-31.csv\t2014-09-09.csv\t2014-09-19.csv\t2014-09-26.csv\r\n"
     ]
    }
   ],
   "source": [
    "!ls Data/ebola/guinea_data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2014-06-16.csv\t2014-08-25.csv\t     2014-09-30-v138.csv  2014-10-31-v169.csv\r\n",
      "2014-06-17.csv\t2014-08-28.csv\t     2014-10-01-v139.csv  2014-11-02-v171.csv\r\n",
      "2014-06-22.csv\t2014-09-01.csv\t     2014-10-03-v141.csv  2014-11-04-v173.csv\r\n",
      "2014-06-24.csv\t2014-09-02.csv\t     2014-10-04-v142.csv  2014-11-08-v177.csv\r\n",
      "2014-06-25.csv\t2014-09-03.csv\t     2014-10-05-v143.csv  2014-11-14-v183.csv\r\n",
      "2014-06-28.csv\t2014-09-04.csv\t     2014-10-07-v145.csv  2014-11-15-v184.csv\r\n",
      "2014-06-29.csv\t2014-09-05.csv\t     2014-10-08-v146.csv  2014-11-19-v188.csv\r\n",
      "2014-07-01.csv\t2014-09-06.csv\t     2014-10-09-v147.csv  2014-11-20-v189.csv\r\n",
      "2014-07-02.csv\t2014-09-07.csv\t     2014-10-10-v148.csv  2014-11-21-v190.csv\r\n",
      "2014-07-03.csv\t2014-09-08.csv\t     2014-10-11-v149.csv  2014-11-23-v192.csv\r\n",
      "2014-07-07.csv\t2014-09-10.csv\t     2014-10-12-v150.csv  2014-11-24-v193.csv\r\n",
      "2014-07-08.csv\t2014-09-11.csv\t     2014-10-13-v151.csv  2014-11-26.csv\r\n",
      "2014-07-10.csv\t2014-09-12.csv\t     2014-10-16-v154.csv  2014-11-27.csv\r\n",
      "2014-07-13.csv\t2014-09-13.csv\t     2014-10-17-v155.csv  2014-11-28.csv\r\n",
      "2014-07-17.csv\t2014-09-14.csv\t     2014-10-18-v156.csv  2014-11-29.csv\r\n",
      "2014-07-20.csv\t2014-09-15.csv\t     2014-10-19-v157.csv  2014-11-30.csv\r\n",
      "2014-07-24.csv\t2014-09-16.csv\t     2014-10-20-v158.csv  2014-12-01.csv\r\n",
      "2014-07-26.csv\t2014-09-17-v125.csv  2014-10-21-v159.csv  2014-12-02.csv\r\n",
      "2014-08-02.csv\t2014-09-20-v128.csv  2014-10-22-v160.csv  2014-12-03.csv\r\n",
      "2014-08-04.csv\t2014-09-21-v129.csv  2014-10-23-v161.csv  2014-12-04.csv\r\n",
      "2014-08-12.csv\t2014-09-23-v131.csv  2014-10-24-v162.csv  2014-12-05.csv\r\n",
      "2014-08-15.csv\t2014-09-25-v133.csv  2014-10-25-v163.csv  2014-12-06.csv\r\n",
      "2014-08-17.csv\t2014-09-26-v134.csv  2014-10-28-v166.csv  2014-12-07.csv\r\n",
      "2014-08-18.csv\t2014-09-27-v135.csv  2014-10-29-v167.csv  2014-12-08.csv\r\n",
      "2014-08-20.csv\t2014-09-28-v136.csv  2014-10-30-v168.csv  2014-12-09.csv\r\n"
     ]
    }
   ],
   "source": [
    "!ls Data/ebola/liberia_data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2014-08-12-v77.csv   2014-09-17-v113.csv  2014-10-24-v150.csv\r\n",
      "2014-08-13-v78.csv   2014-09-18-v114.csv  2014-10-25-v151.csv\r\n",
      "2014-08-14-v79.csv   2014-09-19-v115.csv  2014-10-26-v152.csv\r\n",
      "2014-08-15-v80.csv   2014-09-20-v116.csv  2014-10-27-v153.csv\r\n",
      "2014-08-16-v81.csv   2014-09-21-v117.csv  2014-10-28-v154.csv\r\n",
      "2014-08-17-v82.csv   2014-09-22-v118.csv  2014-10-30-v156.csv\r\n",
      "2014-08-18-v83.csv   2014-09-23-v119.csv  2014-10-31-v157.csv\r\n",
      "2014-08-19-v84.csv   2014-09-24-v120.csv  2014-11-01-v158.csv\r\n",
      "2014-08-20-v85.csv   2014-09-25-v121.csv  2014-11-02-v159.csv\r\n",
      "2014-08-21-v86.csv   2014-09-26-v122.csv  2014-11-06-v163.csv\r\n",
      "2014-08-22-v87.csv   2014-09-27-v123.csv  2014-11-07-v164.csv\r\n",
      "2014-08-23-v88.csv   2014-09-28-v124.csv  2014-11-08-v165.csv\r\n",
      "2014-08-24-v89.csv   2014-09-29-v125.csv  2014-11-10-v167.csv\r\n",
      "2014-08-25-v90.csv   2014-09-30-v126.csv  2014-11-12-v169.csv\r\n",
      "2014-08-26-v91.csv   2014-10-01-v127.csv  2014-11-13.csv\r\n",
      "2014-08-27-v92.csv   2014-10-03-v129.csv  2014-11-14.csv\r\n",
      "2014-08-28-v93.csv   2014-10-04-v130.csv  2014-11-15.csv\r\n",
      "2014-08-29-v94.csv   2014-10-05-v131.csv  2014-11-16.csv\r\n",
      "2014-08-30-v95.csv   2014-10-06-v132.csv  2014-11-17.csv\r\n",
      "2014-08-31-v96.csv   2014-10-07-v133.csv  2014-11-18-v175.csv\r\n",
      "2014-09-01-v97.csv   2014-10-08-v134.csv  2014-11-19-v176.csv\r\n",
      "2014-09-02-v98.csv   2014-10-10-v136.csv  2014-11-20-v177.csv\r\n",
      "2014-09-03-v99.csv   2014-10-11-v137.csv  2014-11-21-v178.csv\r\n",
      "2014-09-05-v101.csv  2014-10-12-v138.csv  2014-11-22-v179.csv\r\n",
      "2014-09-06-v102.csv  2014-10-13-v139.csv  2014-11-23-v180.csv\r\n",
      "2014-09-07-v103.csv  2014-10-14-v140.csv  2014-11-24-v181.csv\r\n",
      "2014-09-08-v104.csv  2014-10-16-v141.csv  2014-11-28.csv\r\n",
      "2014-09-09-v105.csv  2014-10-16-v142.csv  2014-11-29.csv\r\n",
      "2014-09-10-v106.csv  2014-10-17-v143.csv  2014-12-01.csv\r\n",
      "2014-09-11-v107.csv  2014-10-18-v144.csv  2014-12-04.csv\r\n",
      "2014-09-12-v108.csv  2014-10-19-v145.csv  2014-12-05.csv\r\n",
      "2014-09-13-v109.csv  2014-10-20-v146.csv  2014-12-06.csv\r\n",
      "2014-09-14-v110.csv  2014-10-21-v147.csv  2014-12-13.csv\r\n",
      "2014-09-15-v111.csv  2014-10-22-v148.csv\r\n",
      "2014-09-16-v112.csv  2014-10-23-v149.csv\r\n"
     ]
    }
   ],
   "source": [
    "!ls Data/ebola/sl_data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th>Kailahun</th>\n",
       "      <th>Kenema</th>\n",
       "      <th>Kono</th>\n",
       "      <th>Kambia</th>\n",
       "      <th>Koinadugu</th>\n",
       "      <th>Bombali</th>\n",
       "      <th>Tonkolili</th>\n",
       "      <th>Port Loko</th>\n",
       "      <th>Pujehun</th>\n",
       "      <th>Bo</th>\n",
       "      <th>Moyamba</th>\n",
       "      <th>Bonthe</th>\n",
       "      <th>Western area urban</th>\n",
       "      <th>Western area rural</th>\n",
       "      <th>National</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>population</td>\n",
       "      <td>465048</td>\n",
       "      <td>653013</td>\n",
       "      <td>325003.0</td>\n",
       "      <td>341690.0</td>\n",
       "      <td>335471.0</td>\n",
       "      <td>494139</td>\n",
       "      <td>434937</td>\n",
       "      <td>557978</td>\n",
       "      <td>335574</td>\n",
       "      <td>654142</td>\n",
       "      <td>278119</td>\n",
       "      <td>168729.0</td>\n",
       "      <td>1040888</td>\n",
       "      <td>263619</td>\n",
       "      <td>6348350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_noncase</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_suspected</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_probable</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_confirmed</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cum_noncase</td>\n",
       "      <td>201</td>\n",
       "      <td>269</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>54</td>\n",
       "      <td>12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>56</td>\n",
       "      <td>5</td>\n",
       "      <td>657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cum_suspected</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cum_probable</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cum_confirmed</td>\n",
       "      <td>378</td>\n",
       "      <td>259</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>3</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>death_suspected</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>death_probable</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>151</td>\n",
       "      <td>104</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cfr</td>\n",
       "      <td>39.9</td>\n",
       "      <td>40.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.3</td>\n",
       "      <td>0</td>\n",
       "      <td>4.2</td>\n",
       "      <td>0</td>\n",
       "      <td>9.1</td>\n",
       "      <td>60</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.4</td>\n",
       "      <td>0</td>\n",
       "      <td>36.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_new_admission</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_currently_admitted</td>\n",
       "      <td>53</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_cum_admission</td>\n",
       "      <td>341</td>\n",
       "      <td>303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_new_deaths</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_cum_deaths</td>\n",
       "      <td>104</td>\n",
       "      <td>154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_new_discharges</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>etc_cum_discharges</td>\n",
       "      <td>134</td>\n",
       "      <td>59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cum_contacts</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1,356</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>161</td>\n",
       "      <td>16</td>\n",
       "      <td>471</td>\n",
       "      <td>57</td>\n",
       "      <td>227</td>\n",
       "      <td>73</td>\n",
       "      <td>26.0</td>\n",
       "      <td>390</td>\n",
       "      <td>23</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>cum_completed_contacts</td>\n",
       "      <td>NaN</td>\n",
       "      <td>701</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>67</td>\n",
       "      <td>0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>contacts_followed</td>\n",
       "      <td>321</td>\n",
       "      <td>497</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>161</td>\n",
       "      <td>11</td>\n",
       "      <td>466</td>\n",
       "      <td>47</td>\n",
       "      <td>160</td>\n",
       "      <td>73</td>\n",
       "      <td>0.0</td>\n",
       "      <td>310</td>\n",
       "      <td>23</td>\n",
       "      <td>2255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_contacts</td>\n",
       "      <td>321</td>\n",
       "      <td>73</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>308</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>contacts_healthy</td>\n",
       "      <td>305</td>\n",
       "      <td>427</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>141</td>\n",
       "      <td>11</td>\n",
       "      <td>443</td>\n",
       "      <td>47</td>\n",
       "      <td>157</td>\n",
       "      <td>73</td>\n",
       "      <td>0.0</td>\n",
       "      <td>103</td>\n",
       "      <td>23</td>\n",
       "      <td>1937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>contacts_ill</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>contacts_not_seen</td>\n",
       "      <td>16</td>\n",
       "      <td>64</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_completed_contacts</td>\n",
       "      <td>12</td>\n",
       "      <td>38</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>percent_seen</td>\n",
       "      <td>98%</td>\n",
       "      <td>61%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>97%</td>\n",
       "      <td>100%</td>\n",
       "      <td>95%</td>\n",
       "      <td>100%</td>\n",
       "      <td>98%</td>\n",
       "      <td>100%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>101%</td>\n",
       "      <td>100%</td>\n",
       "      <td>87%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          date                variable Kailahun  Kenema      Kono    Kambia  \\\n",
       "0   2014-08-12              population   465048  653013  325003.0  341690.0   \n",
       "1   2014-08-12             new_noncase        0       3       0.0       0.0   \n",
       "2   2014-08-12           new_suspected        0       9       0.0       0.0   \n",
       "3   2014-08-12            new_probable        0       0       0.0       0.0   \n",
       "4   2014-08-12           new_confirmed        0       9       0.0       0.0   \n",
       "5   2014-08-12             cum_noncase      201     269      15.0       1.0   \n",
       "6   2014-08-12           cum_suspected        0      10      33.0       0.0   \n",
       "7   2014-08-12            cum_probable       32       0       1.0       0.0   \n",
       "8   2014-08-12           cum_confirmed      378     259       1.0       1.0   \n",
       "9   2014-08-12         death_suspected        2       1       0.0       0.0   \n",
       "10  2014-08-12          death_probable       32       0       0.0       0.0   \n",
       "11  2014-08-12         death_confirmed      151     104       0.0       0.0   \n",
       "12  2014-08-12                     cfr     39.9    40.2       0.0       0.0   \n",
       "13  2014-08-12       etc_new_admission       16       7       NaN       NaN   \n",
       "14  2014-08-12  etc_currently_admitted       53      41       NaN       NaN   \n",
       "15  2014-08-12       etc_cum_admission      341     303       NaN       NaN   \n",
       "16  2014-08-12          etc_new_deaths        5       3       NaN       NaN   \n",
       "17  2014-08-12          etc_cum_deaths      104     154       NaN       NaN   \n",
       "18  2014-08-12      etc_new_discharges        7       2       NaN       NaN   \n",
       "19  2014-08-12      etc_cum_discharges      134      59       NaN       NaN   \n",
       "20  2014-08-12            cum_contacts      NaN   1,356      38.0       0.0   \n",
       "21  2014-08-12  cum_completed_contacts      NaN     701      38.0       0.0   \n",
       "22  2014-08-12       contacts_followed      321     497       0.0       0.0   \n",
       "23  2014-08-12            new_contacts      321      73       0.0       0.0   \n",
       "24  2014-08-12        contacts_healthy      305     427       0.0       0.0   \n",
       "25  2014-08-12            contacts_ill       11       3       0.0       0.0   \n",
       "26  2014-08-12       contacts_not_seen       16      64       0.0       0.0   \n",
       "27  2014-08-12  new_completed_contacts       12      38       0.0       0.0   \n",
       "28  2014-08-12            percent_seen      98%     61%       NaN       NaN   \n",
       "\n",
       "    Koinadugu Bombali Tonkolili Port Loko Pujehun      Bo Moyamba    Bonthe  \\\n",
       "0    335471.0  494139    434937    557978  335574  654142  278119  168729.0   \n",
       "1         0.0       0         0         1       0       0       0       0.0   \n",
       "2         0.0       0         0         0       0       1       0       0.0   \n",
       "3         0.0       0         0         0       0       1       0       0.0   \n",
       "4         0.0       0         0         2       0       0       0       0.0   \n",
       "5        15.0      10        10         7       2      54      12       0.0   \n",
       "6         0.0       1         0         0       0       2       0       0.0   \n",
       "7         0.0       1         0         1       0       1       0       0.0   \n",
       "8         0.0       7         2        24       3      22       5       1.0   \n",
       "9         0.0       0         0         1       0       0       0       0.0   \n",
       "10        0.0       0         0         1       0       0       0       0.0   \n",
       "11        0.0       1         0         1       0       2       3       0.0   \n",
       "12        0.0    14.3         0       4.2       0     9.1      60       0.0   \n",
       "13        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "14        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "15        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "16        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "17        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "18        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "19        NaN     NaN       NaN       NaN     NaN     NaN     NaN       NaN   \n",
       "20        0.0     161        16       471      57     227      73      26.0   \n",
       "21        0.0      16         0         7      10      67       0      26.0   \n",
       "22        0.0     161        11       466      47     160      73       0.0   \n",
       "23        0.0       5         5       308       0       6       0       0.0   \n",
       "24        0.0     141        11       443      47     157      73       0.0   \n",
       "25        0.0       0         0         0       0       0       0       0.0   \n",
       "26        0.0      20         0        28       0       3       0       0.0   \n",
       "27        0.0      16         0         0       1       0       0       0.0   \n",
       "28        NaN     97%      100%       95%    100%     98%    100%       NaN   \n",
       "\n",
       "   Western area urban Western area rural National  \n",
       "0             1040888             263619  6348350  \n",
       "1                   0                  0        4  \n",
       "2                   0                  0       10  \n",
       "3                   0                  0        1  \n",
       "4                   0                  0       11  \n",
       "5                  56                  5      657  \n",
       "6                   0                  0       46  \n",
       "7                   1                  0       37  \n",
       "8                  13                  1      717  \n",
       "9                   1                  0        5  \n",
       "10                  1                  0       34  \n",
       "11                  2                  0      264  \n",
       "12               15.4                  0     36.8  \n",
       "13                NaN                NaN      NaN  \n",
       "14                NaN                NaN      NaN  \n",
       "15                NaN                NaN      NaN  \n",
       "16                NaN                NaN      NaN  \n",
       "17                NaN                NaN      NaN  \n",
       "18                NaN                NaN      NaN  \n",
       "19                NaN                NaN      NaN  \n",
       "20                390                 23     2838  \n",
       "21                 82                  0      901  \n",
       "22                310                 23     2255  \n",
       "23                 25                  0      743  \n",
       "24                103                 23     1937  \n",
       "25                  0                  0       14  \n",
       "26                  0                  0      131  \n",
       "27                  6                  0       73  \n",
       "28               101%               100%      87%  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# example file\n",
    "pd.read_csv('Data/ebola/sl_data/2014-08-12-v77.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "#### we want:\n",
    "following schema: but with key=(day, country) and feaures are new cases and deaths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Stool</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Taxon</th>\n",
       "      <th>Patient</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">Firmicutes</th>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "      <td>4182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1174</td>\n",
       "      <td>703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>3946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>831</td>\n",
       "      <td>8605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0</td>\n",
       "      <td>693</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Group  Tissue  Stool\n",
       "Taxon      Patient                      \n",
       "Firmicutes 1            0     136   4182\n",
       "           2            1    1174    703\n",
       "           3            0     408   3946\n",
       "           4            1     831   8605\n",
       "           5            0     693     50"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mb = pd.read_csv(\"Data/microbiome.csv\", index_col=['Taxon','Patient'])\n",
    "mb.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### lets start:\n",
    "##### reading data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "files=glob.glob('Data/ebola/sl_data/*.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sl_data_array=[]\n",
    "for file in files:\n",
    "    read = pd.read_csv(file)\n",
    "    sl_data_array.append(read)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>34 Military Hospital</th>\n",
       "      <th>Bo</th>\n",
       "      <th>Bo EMC</th>\n",
       "      <th>Bombali</th>\n",
       "      <th>Bonthe</th>\n",
       "      <th>Hastings-F/Town</th>\n",
       "      <th>Kailahun</th>\n",
       "      <th>Kambia</th>\n",
       "      <th>Kenema</th>\n",
       "      <th>Kenema (IFRC)</th>\n",
       "      <th>...</th>\n",
       "      <th>Port Loko</th>\n",
       "      <th>Pujehun</th>\n",
       "      <th>Tonkolili</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "      <th>Western area</th>\n",
       "      <th>Western area combined</th>\n",
       "      <th>Western area rural</th>\n",
       "      <th>Western area urban</th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>654142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>494139</td>\n",
       "      <td>168729</td>\n",
       "      <td>NaN</td>\n",
       "      <td>465048</td>\n",
       "      <td>341690</td>\n",
       "      <td>653013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>557978</td>\n",
       "      <td>335574</td>\n",
       "      <td>434937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>263619</td>\n",
       "      <td>1040888</td>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>population</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_noncase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_suspected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_probable</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_confirmed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   34 Military Hospital      Bo  Bo EMC Bombali  Bonthe  Hastings-F/Town  \\\n",
       "0                   NaN  654142     NaN  494139  168729              NaN   \n",
       "1                   NaN       0     NaN       0       0              NaN   \n",
       "2                   NaN       1     NaN       0       0              NaN   \n",
       "3                   NaN       1     NaN       0       0              NaN   \n",
       "4                   NaN       0     NaN       0       0              NaN   \n",
       "\n",
       "  Kailahun  Kambia  Kenema  Kenema (IFRC)      ...       Port Loko Pujehun  \\\n",
       "0   465048  341690  653013            NaN      ...          557978  335574   \n",
       "1        0       0       3            NaN      ...               1       0   \n",
       "2        0       0       9            NaN      ...               0       0   \n",
       "3        0       0       0            NaN      ...               0       0   \n",
       "4        0       0       9            NaN      ...               2       0   \n",
       "\n",
       "  Tonkolili Unnamed: 18 Western area  Western area combined  \\\n",
       "0    434937         NaN          NaN                    NaN   \n",
       "1         0         NaN          NaN                    NaN   \n",
       "2         0         NaN          NaN                    NaN   \n",
       "3         0         NaN          NaN                    NaN   \n",
       "4         0         NaN          NaN                    NaN   \n",
       "\n",
       "   Western area rural Western area urban        date       variable  \n",
       "0              263619            1040888  2014-08-12     population  \n",
       "1                   0                  0  2014-08-12    new_noncase  \n",
       "2                   0                  0  2014-08-12  new_suspected  \n",
       "3                   0                  0  2014-08-12   new_probable  \n",
       "4                   0                  0  2014-08-12  new_confirmed  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_data_array[0].append(sl_data_array[1], ignore_index=True)\n",
    "sl_data = pd.concat(sl_data_array, ignore_index=True)\n",
    "sl_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['34 Military Hospital', 'Bo', 'Bo EMC', 'Bombali', 'Bonthe',\n",
       "       'Hastings-F/Town', 'Kailahun', 'Kambia', 'Kenema', 'Kenema (IFRC)',\n",
       "       'Kenema (KGH)', 'Koinadugu', 'Kono', 'Moyamba', 'National',\n",
       "       'Police training School', 'Police traning School', 'Port Loko',\n",
       "       'Pujehun', 'Tonkolili', 'Unnamed: 18', 'Western area',\n",
       "       'Western area combined', 'Western area rural', 'Western area urban',\n",
       "       'date', 'variable'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['population', 'new_noncase', 'new_suspected', 'new_probable',\n",
       "       'new_confirmed', 'cum_noncase', 'cum_suspected', 'cum_probable',\n",
       "       'cum_confirmed', 'death_suspected', 'death_probable',\n",
       "       'death_confirmed', 'cfr', 'etc_new_admission',\n",
       "       'etc_currently_admitted', 'etc_cum_admission', 'etc_new_deaths',\n",
       "       'etc_cum_deaths', 'etc_new_discharges', 'etc_cum_discharges',\n",
       "       'cum_contacts', 'cum_completed_contacts', 'contacts_followed',\n",
       "       'new_contacts', 'contacts_healthy', 'contacts_ill',\n",
       "       'contacts_not_seen', 'new_completed_contacts', 'percent_seen',\n",
       "       'total_lab_samples', 'repeat_samples', 'new_samples',\n",
       "       'new_positive', 'new_negative', 'pending', 'positive_corpse',\n",
       "       'negative_corpse'], dtype=object)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_data['variable'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# we only need the variables 'new_confirmed' and 'death_confirmed' for now.\n",
    "sl_data_relevant = sl_data[sl_data['variable'].isin(['new_confirmed', 'death_confirmed'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make the date the index\n",
    "sl_data_relevant.set_index(['date', 'variable'])\n",
    "sl_data_relevant.index.is_unique"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### cleaning data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sl_data_relevant.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "206"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sl_data_relevant.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34 Military Hospital      206\n",
       "Bo EMC                    206\n",
       "Kenema (KGH)              206\n",
       "Hastings-F/Town           206\n",
       "Police training School    206\n",
       "Police traning School     206\n",
       "Kenema (IFRC)             206\n",
       "Unnamed: 18               206\n",
       "Western area combined     206\n",
       "Western area              184\n",
       "Western area rural         35\n",
       "Western area urban         35\n",
       "National                   15\n",
       "Bo                         14\n",
       "Kambia                     14\n",
       "Bonthe                     14\n",
       "Kailahun                   14\n",
       "Bombali                    14\n",
       "Moyamba                    14\n",
       "Kenema                     14\n",
       "Koinadugu                  14\n",
       "Kono                       14\n",
       "Port Loko                  14\n",
       "Pujehun                    14\n",
       "Tonkolili                  14\n",
       "date                        0\n",
       "variable                    0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how many NaNs are there\n",
    "# columns\n",
    "sl_data_relevant.isnull().sum().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3236    25\n",
       "3229    25\n",
       "2785    25\n",
       "2792    25\n",
       "1983    25\n",
       "1976    25\n",
       "3192    25\n",
       "2718    24\n",
       "2681    24\n",
       "1345    24\n",
       "1461    24\n",
       "2755    24\n",
       "2644    24\n",
       "2607    24\n",
       "1454    12\n",
       "1693    11\n",
       "1664    11\n",
       "1686    11\n",
       "1722    11\n",
       "1715    11\n",
       "1744    11\n",
       "1751    11\n",
       "1773    11\n",
       "1780    11\n",
       "1657    11\n",
       "475     11\n",
       "410     11\n",
       "642     11\n",
       "1635    11\n",
       "1628    11\n",
       "        ..\n",
       "1309    10\n",
       "1287    10\n",
       "1280    10\n",
       "1258    10\n",
       "1251    10\n",
       "1229    10\n",
       "1222    10\n",
       "1200    10\n",
       "3014    10\n",
       "2977    10\n",
       "2570    10\n",
       "2859    10\n",
       "2600    10\n",
       "2637    10\n",
       "2674    10\n",
       "2711    10\n",
       "2748    10\n",
       "2822    10\n",
       "2970    10\n",
       "2829    10\n",
       "2866    10\n",
       "2896    10\n",
       "2903    10\n",
       "2933    10\n",
       "2940    10\n",
       "1519    10\n",
       "1512    10\n",
       "4       10\n",
       "1809     9\n",
       "1802     9\n",
       "dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rows\n",
    "sl_data_relevant.isnull().sum(axis=1).sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# rename the data\n",
    "sl_data_clean = sl_data_relevant\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#remove cols that have only NaN\n",
    "sl_data_clean = sl_data_clean.dropna(axis=1, how='all')\n",
    "len(sl_data_clean.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "199"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#remove rows that have only NaN\n",
    "sl_data_clean = sl_data_clean.dropna(axis=0, thresh=3)\n",
    "len(sl_data_clean.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Western area          177\n",
       "Western area urban     28\n",
       "Western area rural     28\n",
       "National                8\n",
       "Moyamba                 7\n",
       "Koinadugu               7\n",
       "Bombali                 7\n",
       "Bonthe                  7\n",
       "Kailahun                7\n",
       "Kambia                  7\n",
       "Kenema                  7\n",
       "Bo                      7\n",
       "Kono                    7\n",
       "Port Loko               7\n",
       "Pujehun                 7\n",
       "Tonkolili               7\n",
       "date                    0\n",
       "variable                0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# count again\n",
    "# columns\n",
    "sl_data_clean.isnull().sum().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2718    15\n",
       "2644    15\n",
       "2755    15\n",
       "2607    15\n",
       "1345    15\n",
       "1461    15\n",
       "2681    15\n",
       "1454     3\n",
       "1606     2\n",
       "1628     2\n",
       "1577     2\n",
       "1599     2\n",
       "1773     2\n",
       "410      2\n",
       "1635     2\n",
       "417      2\n",
       "1657     2\n",
       "1664     2\n",
       "468      2\n",
       "475      2\n",
       "1686     2\n",
       "1693     2\n",
       "1715     2\n",
       "1722     2\n",
       "1570     2\n",
       "1541     2\n",
       "1548     2\n",
       "642      2\n",
       "845      2\n",
       "852      2\n",
       "        ..\n",
       "1367     1\n",
       "1338     1\n",
       "1316     1\n",
       "1309     1\n",
       "1287     1\n",
       "1280     1\n",
       "1258     1\n",
       "2234     1\n",
       "1867     1\n",
       "1896     1\n",
       "2079     1\n",
       "2227     1\n",
       "2197     1\n",
       "2190     1\n",
       "2160     1\n",
       "2153     1\n",
       "2123     1\n",
       "2116     1\n",
       "2086     1\n",
       "2049     1\n",
       "1918     1\n",
       "2042     1\n",
       "2020     1\n",
       "2013     1\n",
       "1954     1\n",
       "1947     1\n",
       "1925     1\n",
       "4        1\n",
       "1802     0\n",
       "1809     0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rows\n",
    "sl_data_clean.isnull().sum(axis=1).sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# there are still NaNs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# there is a 'national' feature. Maybe we can use it\n",
    "sl_data_clean_nat = sl_data_clean[['date', 'variable', 'National']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "date        0\n",
       "variable    0\n",
       "National    8\n",
       "dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_data_clean_nat.set_index(['date', 'variable'])\n",
    "sl_data_clean_nat.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bo</th>\n",
       "      <th>Bombali</th>\n",
       "      <th>Bonthe</th>\n",
       "      <th>Kailahun</th>\n",
       "      <th>Kambia</th>\n",
       "      <th>Kenema</th>\n",
       "      <th>Koinadugu</th>\n",
       "      <th>Kono</th>\n",
       "      <th>Moyamba</th>\n",
       "      <th>National</th>\n",
       "      <th>Port Loko</th>\n",
       "      <th>Pujehun</th>\n",
       "      <th>Tonkolili</th>\n",
       "      <th>Western area</th>\n",
       "      <th>Western area rural</th>\n",
       "      <th>Western area urban</th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2014-08-26</td>\n",
       "      <td>new_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>180</td>\n",
       "      <td>0</td>\n",
       "      <td>157</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2014-08-26</td>\n",
       "      <td>death_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>468</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2014-08-29</td>\n",
       "      <td>new_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>180</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2014-08-29</td>\n",
       "      <td>death_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2014-09-05</td>\n",
       "      <td>new_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>181</td>\n",
       "      <td>0</td>\n",
       "      <td>190</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2014-09-05</td>\n",
       "      <td>death_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>845</th>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>2014-09-12</td>\n",
       "      <td>new_confirmed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852</th>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>186</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2014-09-12</td>\n",
       "      <td>death_confirmed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Bo Bombali Bonthe Kailahun Kambia Kenema Koinadugu Kono Moyamba National  \\\n",
       "410   0       0      0        0      0      2         0    0       0      NaN   \n",
       "417  15       2      0      180      0    157         0    0       4      NaN   \n",
       "468   4       0      0        4      0      3         0    0       0      NaN   \n",
       "475  15       6      1      180      0    168         0    0       4      NaN   \n",
       "642   0       0      0        6      0      1         0    0       0      NaN   \n",
       "649  16       6      1      181      0    190         0    2       6      NaN   \n",
       "845   1       9      0        0      0      1         0    0       0      NaN   \n",
       "852  16       6      1      186      1    217         0    2       6      NaN   \n",
       "\n",
       "    Port Loko Pujehun Tonkolili  Western area Western area rural  \\\n",
       "410         0       0         0           NaN                  0   \n",
       "417         2       0         1           NaN                  0   \n",
       "468         0       0         0           NaN                  1   \n",
       "475         2       1         2           NaN                  2   \n",
       "642        11       0         0           NaN                  3   \n",
       "649         3       1         2           NaN                  2   \n",
       "845        10       0         1           NaN                  1   \n",
       "852        16       1         2           NaN                  2   \n",
       "\n",
       "    Western area urban        date         variable  \n",
       "410                  4  2014-08-26    new_confirmed  \n",
       "417                  2  2014-08-26  death_confirmed  \n",
       "468                  3  2014-08-29    new_confirmed  \n",
       "475                  2  2014-08-29  death_confirmed  \n",
       "642                  2  2014-09-05    new_confirmed  \n",
       "649                  3  2014-09-05  death_confirmed  \n",
       "845                  8  2014-09-12    new_confirmed  \n",
       "852                  3  2014-09-12  death_confirmed  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are 8 NaNs left, lets check the rows\n",
    "sl_data_clean[sl_data_clean['National'].isnull()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### plan:\n",
    "- replace all NaNs with 0\n",
    "- make my own 'national' column\n",
    "- check if it is the same as the existing National col\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# replace all with 0\n",
    "sl_data_clean.fillna(value=0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Bo                    float64\n",
       "Bombali               float64\n",
       "Bonthe                float64\n",
       "Kailahun              float64\n",
       "Kambia                float64\n",
       "Kenema                float64\n",
       "Koinadugu             float64\n",
       "Kono                  float64\n",
       "Moyamba               float64\n",
       "National              float64\n",
       "Port Loko             float64\n",
       "Pujehun               float64\n",
       "Tonkolili             float64\n",
       "Western area          float64\n",
       "Western area rural    float64\n",
       "Western area urban    float64\n",
       "date                   object\n",
       "variable               object\n",
       "dtype: object"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# first make all types objects\n",
    "def change_to_numeric(data):\n",
    "    col_list = list(data.columns)\n",
    "    col_list.remove('date')\n",
    "    col_list.remove('variable')\n",
    "    data[col_list] = data[col_list].apply(pd.to_numeric)\n",
    "    \n",
    "change_to_numeric(sl_data_clean)\n",
    "sl_data_clean.dtypes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Bo</th>\n",
       "      <th>Bombali</th>\n",
       "      <th>Bonthe</th>\n",
       "      <th>Kailahun</th>\n",
       "      <th>Kambia</th>\n",
       "      <th>Kenema</th>\n",
       "      <th>Koinadugu</th>\n",
       "      <th>Kono</th>\n",
       "      <th>Moyamba</th>\n",
       "      <th>National</th>\n",
       "      <th>Port Loko</th>\n",
       "      <th>Pujehun</th>\n",
       "      <th>Tonkolili</th>\n",
       "      <th>Western area</th>\n",
       "      <th>Western area rural</th>\n",
       "      <th>Western area urban</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-12</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>264.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-13</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>273.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-14</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-15</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>287.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-16</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>297.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-17</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>172.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-18</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>312.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-19</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>320.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-20</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>322.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-21</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>329.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-22</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>333.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-23</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-24</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>341.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-25</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>355.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-26</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014-11-13</th>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-15</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-16</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-17</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-18</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1240.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-20</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>68.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1282.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>133.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-21</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>6.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>68.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1303.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-22</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>68.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1333.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>153.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-23</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>69.0</td>\n",
       "      <td>185.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>1364.0</td>\n",
       "      <td>134.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>162.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-24</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>70.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1397.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-28</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>4.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>76.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1481.0</td>\n",
       "      <td>1645.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>147.0</td>\n",
       "      <td>193.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-29</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>77.0</td>\n",
       "      <td>195.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1496.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>152.0</td>\n",
       "      <td>198.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-12-01</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>78.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>1549.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>218.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-12-04</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>81.0</td>\n",
       "      <td>209.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>1648.0</td>\n",
       "      <td>201.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-12-05</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>1.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>82.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1669.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>255.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014-12-06</th>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>83.0</td>\n",
       "      <td>223.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>1708.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>193.0</td>\n",
       "      <td>267.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>199 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Bo  Bombali  Bonthe  Kailahun  Kambia  Kenema  \\\n",
       "date       variable                                                           \n",
       "2014-08-12 new_confirmed     0.0      0.0     0.0       0.0     0.0     9.0   \n",
       "           death_confirmed   2.0      1.0     0.0     151.0     0.0   104.0   \n",
       "2014-08-13 new_confirmed     1.0      0.0     0.0      11.0     0.0     1.0   \n",
       "           death_confirmed   2.0      1.0     0.0     157.0     0.0   107.0   \n",
       "2014-08-14 new_confirmed     0.0      0.0     0.0       0.0     0.0     8.0   \n",
       "           death_confirmed   2.0      1.0     0.0     160.0     0.0   111.0   \n",
       "2014-08-15 new_confirmed     1.0      0.0     0.0       0.0     0.0     3.0   \n",
       "           death_confirmed   2.0      1.0     0.0     162.0     0.0   116.0   \n",
       "2014-08-16 new_confirmed     0.0      0.0     0.0      12.0     0.0     4.0   \n",
       "           death_confirmed   2.0      1.0     0.0     167.0     0.0   121.0   \n",
       "2014-08-17 new_confirmed     0.0      0.0     0.0       0.0     0.0     2.0   \n",
       "           death_confirmed   2.0      1.0     0.0     172.0     0.0   124.0   \n",
       "2014-08-18 new_confirmed     0.0      0.0     0.0       5.0     0.0     0.0   \n",
       "           death_confirmed   2.0      1.0     0.0     173.0     0.0   130.0   \n",
       "2014-08-19 new_confirmed     2.0      0.0     0.0       3.0     0.0     2.0   \n",
       "           death_confirmed   2.0      1.0     0.0     176.0     0.0   133.0   \n",
       "2014-08-20 new_confirmed     0.0      0.0     0.0       1.0     0.0     2.0   \n",
       "           death_confirmed   2.0      1.0     0.0     176.0     0.0   135.0   \n",
       "2014-08-21 new_confirmed     1.0      0.0     0.0       1.0     0.0     0.0   \n",
       "           death_confirmed   2.0      1.0     0.0     180.0     0.0   138.0   \n",
       "2014-08-22 new_confirmed     6.0      6.0     0.0       6.0     0.0    12.0   \n",
       "           death_confirmed   2.0      1.0     0.0     180.0     0.0   142.0   \n",
       "2014-08-23 new_confirmed     0.0      0.0     0.0      11.0     0.0     5.0   \n",
       "           death_confirmed   2.0      1.0     0.0     180.0     0.0   145.0   \n",
       "2014-08-24 new_confirmed     2.0      0.0     0.0      12.0     0.0    12.0   \n",
       "           death_confirmed   2.0      1.0     0.0     180.0     0.0   150.0   \n",
       "2014-08-25 new_confirmed     0.0      3.0     0.0       0.0     0.0     6.0   \n",
       "           death_confirmed  10.0      2.0     0.0     180.0     0.0   154.0   \n",
       "2014-08-26 new_confirmed     0.0      0.0     0.0       0.0     0.0     2.0   \n",
       "           death_confirmed  15.0      2.0     0.0     180.0     0.0   157.0   \n",
       "...                          ...      ...     ...       ...     ...     ...   \n",
       "2014-11-13 death_confirmed   0.0      0.0     0.0       0.0     0.0     0.0   \n",
       "2014-11-15 new_confirmed     8.0      4.0     1.0       0.0     0.0     0.0   \n",
       "           death_confirmed   0.0      0.0     0.0       0.0     0.0     0.0   \n",
       "2014-11-16 new_confirmed     2.0     12.0     0.0       1.0     1.0     0.0   \n",
       "           death_confirmed   0.0      0.0     0.0       0.0     0.0     0.0   \n",
       "2014-11-17 new_confirmed     3.0      2.0     0.0       0.0     0.0     0.0   \n",
       "           death_confirmed   0.0      0.0     0.0       0.0     0.0     0.0   \n",
       "2014-11-18 new_confirmed     1.0      2.0     0.0       0.0     0.0     0.0   \n",
       "           death_confirmed   0.0      0.0     0.0       0.0     0.0     0.0   \n",
       "2014-11-20 new_confirmed     2.0     18.0     0.0       0.0     1.0     0.0   \n",
       "           death_confirmed  68.0    175.0     2.0     228.0    18.0   261.0   \n",
       "2014-11-21 new_confirmed     6.0     14.0     0.0       2.0     1.0     0.0   \n",
       "           death_confirmed  68.0    180.0     2.0     228.0    18.0   261.0   \n",
       "2014-11-22 new_confirmed     1.0      4.0     0.0       0.0     0.0     0.0   \n",
       "           death_confirmed  68.0    183.0     2.0     228.0    18.0   261.0   \n",
       "2014-11-23 new_confirmed     0.0      0.0     0.0       0.0     2.0     0.0   \n",
       "           death_confirmed  69.0    185.0     2.0     228.0    18.0   261.0   \n",
       "2014-11-24 new_confirmed     8.0     10.0     0.0       0.0     6.0     0.0   \n",
       "           death_confirmed  70.0    187.0     2.0     228.0    21.0   261.0   \n",
       "2014-11-28 new_confirmed     4.0     11.0     0.0       0.0     0.0     0.0   \n",
       "           death_confirmed  76.0    194.0     2.0     228.0    22.0   261.0   \n",
       "2014-11-29 new_confirmed     2.0     77.0     0.0       1.0     1.0     0.0   \n",
       "           death_confirmed  77.0    195.0     2.0     228.0    23.0   261.0   \n",
       "2014-12-01 new_confirmed     3.0     10.0     0.0       0.0     1.0     0.0   \n",
       "           death_confirmed  78.0    200.0     2.0     228.0    24.0   261.0   \n",
       "2014-12-04 new_confirmed     1.0      0.0     0.0       0.0     1.0     0.0   \n",
       "           death_confirmed  81.0    209.0     3.0     228.0    26.0   261.0   \n",
       "2014-12-05 new_confirmed     1.0     11.0     0.0       0.0     4.0     0.0   \n",
       "           death_confirmed  82.0    214.0     3.0     228.0    26.0   261.0   \n",
       "2014-12-06 death_confirmed  83.0    223.0     3.0     228.0    26.0   261.0   \n",
       "\n",
       "                            Koinadugu  Kono  Moyamba  National  Port Loko  \\\n",
       "date       variable                                                         \n",
       "2014-08-12 new_confirmed          0.0   0.0      0.0      11.0        2.0   \n",
       "           death_confirmed        0.0   0.0      3.0     264.0        1.0   \n",
       "2014-08-13 new_confirmed          0.0   0.0      0.0      15.0        0.0   \n",
       "           death_confirmed        0.0   0.0      3.0     273.0        1.0   \n",
       "2014-08-14 new_confirmed          0.0   0.0      1.0      13.0        2.0   \n",
       "           death_confirmed        0.0   0.0      3.0     280.0        1.0   \n",
       "2014-08-15 new_confirmed          0.0   0.0      0.0      10.0        2.0   \n",
       "           death_confirmed        0.0   0.0      3.0     287.0        1.0   \n",
       "2014-08-16 new_confirmed          0.0   0.0      0.0      18.0        0.0   \n",
       "           death_confirmed        0.0   0.0      3.0     297.0        1.0   \n",
       "2014-08-17 new_confirmed          0.0   0.0      0.0       2.0        0.0   \n",
       "           death_confirmed        0.0   0.0      3.0     305.0        1.0   \n",
       "2014-08-18 new_confirmed          0.0   0.0      0.0       5.0        0.0   \n",
       "           death_confirmed        0.0   0.0      3.0     312.0        1.0   \n",
       "2014-08-19 new_confirmed          0.0   0.0      0.0       9.0        0.0   \n",
       "           death_confirmed        0.0   0.0      4.0     320.0        1.0   \n",
       "2014-08-20 new_confirmed          0.0   0.0      0.0       4.0        1.0   \n",
       "           death_confirmed        0.0   0.0      4.0     322.0        1.0   \n",
       "2014-08-21 new_confirmed          0.0   0.0      0.0       9.0        0.0   \n",
       "           death_confirmed        0.0   0.0      4.0     329.0        1.0   \n",
       "2014-08-22 new_confirmed          0.0   0.0      1.0      56.0       13.0   \n",
       "           death_confirmed        0.0   0.0      4.0     333.0        1.0   \n",
       "2014-08-23 new_confirmed          0.0   0.0      0.0      23.0        7.0   \n",
       "           death_confirmed        0.0   0.0      4.0     336.0        1.0   \n",
       "2014-08-24 new_confirmed          0.0   0.0      0.0      31.0        0.0   \n",
       "           death_confirmed        0.0   0.0      4.0     341.0        1.0   \n",
       "2014-08-25 new_confirmed          0.0   0.0      0.0      20.0        3.0   \n",
       "           death_confirmed        0.0   0.0      4.0     355.0        2.0   \n",
       "2014-08-26 new_confirmed          0.0   0.0      0.0       0.0        0.0   \n",
       "           death_confirmed        0.0   0.0      4.0       0.0        2.0   \n",
       "...                               ...   ...      ...       ...        ...   \n",
       "2014-11-13 death_confirmed        0.0   0.0      0.0    1180.0        0.0   \n",
       "2014-11-15 new_confirmed          1.0   0.0      2.0      54.0       14.0   \n",
       "           death_confirmed        0.0   0.0      0.0    1206.0        0.0   \n",
       "2014-11-16 new_confirmed          0.0   0.0      3.0      89.0        9.0   \n",
       "           death_confirmed        0.0   0.0      0.0    1223.0        0.0   \n",
       "2014-11-17 new_confirmed          2.0   5.0      2.0      53.0       18.0   \n",
       "           death_confirmed        0.0   0.0      0.0    1223.0        0.0   \n",
       "2014-11-18 new_confirmed          1.0   0.0      2.0      43.0        0.0   \n",
       "           death_confirmed        0.0   0.0      0.0    1240.0        0.0   \n",
       "2014-11-20 new_confirmed          2.0  10.0      3.0      94.0       25.0   \n",
       "           death_confirmed       22.0  27.0     38.0    1282.0      118.0   \n",
       "2014-11-21 new_confirmed          0.0   0.0      1.0      51.0        5.0   \n",
       "           death_confirmed       22.0  27.0     38.0    1303.0      122.0   \n",
       "2014-11-22 new_confirmed          0.0   0.0      1.0      47.0        3.0   \n",
       "           death_confirmed       22.0  27.0     38.0    1333.0      124.0   \n",
       "2014-11-23 new_confirmed          0.0   1.0      1.0      39.0       13.0   \n",
       "           death_confirmed       23.0  29.0     39.0    1364.0      134.0   \n",
       "2014-11-24 new_confirmed          0.0   0.0      1.0      83.0       13.0   \n",
       "           death_confirmed       23.0  29.0     40.0    1397.0      141.0   \n",
       "2014-11-28 new_confirmed          2.0   0.0      0.0      84.0       14.0   \n",
       "           death_confirmed       26.0  29.0     40.0    1481.0     1645.0   \n",
       "2014-11-29 new_confirmed          0.0  15.0      0.0      64.0        0.0   \n",
       "           death_confirmed       26.0  31.0     40.0    1496.0      164.0   \n",
       "2014-12-01 new_confirmed          0.0   0.0      2.0      72.0       26.0   \n",
       "           death_confirmed       26.0  31.0     41.0    1549.0      174.0   \n",
       "2014-12-04 new_confirmed          0.0   0.0      1.0      37.0       20.0   \n",
       "           death_confirmed       27.0  39.0     42.0    1648.0      201.0   \n",
       "2014-12-05 new_confirmed          0.0   7.0      0.0      54.0       10.0   \n",
       "           death_confirmed       27.0  39.0     43.0    1669.0      205.0   \n",
       "2014-12-06 death_confirmed       27.0  41.0     44.0    1708.0      208.0   \n",
       "\n",
       "                            Pujehun  Tonkolili  Western area  \\\n",
       "date       variable                                            \n",
       "2014-08-12 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-13 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-14 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-15 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-16 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-17 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-18 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-08-19 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-20 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-21 new_confirmed        0.0        6.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-22 new_confirmed        1.0        2.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-23 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-24 new_confirmed        0.0        1.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-25 new_confirmed        2.0        2.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "2014-08-26 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        1.0           0.0   \n",
       "...                             ...        ...           ...   \n",
       "2014-11-13 death_confirmed      0.0        0.0           0.0   \n",
       "2014-11-15 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-11-16 new_confirmed        0.0        2.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-11-17 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-11-18 new_confirmed        2.0       11.0           0.0   \n",
       "           death_confirmed      0.0        0.0           0.0   \n",
       "2014-11-20 new_confirmed        0.0        2.0           0.0   \n",
       "           death_confirmed     16.0       70.0           0.0   \n",
       "2014-11-21 new_confirmed        0.0        5.0           0.0   \n",
       "           death_confirmed     16.0       70.0           0.0   \n",
       "2014-11-22 new_confirmed        0.0        5.0           0.0   \n",
       "           death_confirmed     16.0       71.0           0.0   \n",
       "2014-11-23 new_confirmed        0.0        3.0           0.0   \n",
       "           death_confirmed     16.0       73.0           0.0   \n",
       "2014-11-24 new_confirmed        0.0        4.0           0.0   \n",
       "           death_confirmed     16.0       74.0           0.0   \n",
       "2014-11-28 new_confirmed        1.0        7.0           0.0   \n",
       "           death_confirmed     16.0       83.0           0.0   \n",
       "2014-11-29 new_confirmed        0.0        1.0           0.0   \n",
       "           death_confirmed     16.0       83.0           0.0   \n",
       "2014-12-01 new_confirmed        0.0        0.0           0.0   \n",
       "           death_confirmed     16.0       84.0           0.0   \n",
       "2014-12-04 new_confirmed        0.0        1.0           0.0   \n",
       "           death_confirmed     16.0       86.0           0.0   \n",
       "2014-12-05 new_confirmed        0.0        1.0           0.0   \n",
       "           death_confirmed     16.0       86.0           0.0   \n",
       "2014-12-06 death_confirmed     16.0       88.0           0.0   \n",
       "\n",
       "                            Western area rural  Western area urban  \n",
       "date       variable                                                 \n",
       "2014-08-12 new_confirmed                   0.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-13 new_confirmed                   0.0                 2.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-14 new_confirmed                   2.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-15 new_confirmed                   0.0                 4.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-16 new_confirmed                   1.0                 1.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-17 new_confirmed                   0.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-18 new_confirmed                   0.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-19 new_confirmed                   0.0                 2.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-20 new_confirmed                   0.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-21 new_confirmed                   1.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-22 new_confirmed                   5.0                 4.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-23 new_confirmed                   0.0                 0.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-24 new_confirmed                   1.0                 3.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-25 new_confirmed                   3.0                 1.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "2014-08-26 new_confirmed                   0.0                 4.0  \n",
       "           death_confirmed                 0.0                 2.0  \n",
       "...                                        ...                 ...  \n",
       "2014-11-13 death_confirmed                 0.0                 0.0  \n",
       "2014-11-15 new_confirmed                   8.0                16.0  \n",
       "           death_confirmed                 0.0                 0.0  \n",
       "2014-11-16 new_confirmed                  24.0                35.0  \n",
       "           death_confirmed                 0.0                 0.0  \n",
       "2014-11-17 new_confirmed                   6.0                15.0  \n",
       "           death_confirmed                 0.0                 0.0  \n",
       "2014-11-18 new_confirmed                   7.0                17.0  \n",
       "           death_confirmed                 0.0                 0.0  \n",
       "2014-11-20 new_confirmed                   6.0                25.0  \n",
       "           death_confirmed               106.0               133.0  \n",
       "2014-11-21 new_confirmed                   8.0                 9.0  \n",
       "           death_confirmed               112.0               139.0  \n",
       "2014-11-22 new_confirmed                   7.0                26.0  \n",
       "           death_confirmed               122.0               153.0  \n",
       "2014-11-23 new_confirmed                   8.0                11.0  \n",
       "           death_confirmed               125.0               162.0  \n",
       "2014-11-24 new_confirmed                  10.0                31.0  \n",
       "           death_confirmed               131.0               174.0  \n",
       "2014-11-28 new_confirmed                   3.0                42.0  \n",
       "           death_confirmed               147.0               193.0  \n",
       "2014-11-29 new_confirmed                  11.0                26.0  \n",
       "           death_confirmed               152.0               198.0  \n",
       "2014-12-01 new_confirmed                   6.0                24.0  \n",
       "           death_confirmed               166.0               218.0  \n",
       "2014-12-04 new_confirmed                   5.0                 8.0  \n",
       "           death_confirmed               179.0               250.0  \n",
       "2014-12-05 new_confirmed                   5.0                15.0  \n",
       "           death_confirmed               184.0               255.0  \n",
       "2014-12-06 death_confirmed               193.0               267.0  \n",
       "\n",
       "[199 rows x 16 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# set index\n",
    "sl_data_clean.set_index(['date', 'variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4         11.0\n",
       "11       264.0\n",
       "33        15.0\n",
       "40       273.0\n",
       "62        13.0\n",
       "69       280.0\n",
       "91        10.0\n",
       "98       287.0\n",
       "120       18.0\n",
       "127      297.0\n",
       "149        2.0\n",
       "156      305.0\n",
       "178        5.0\n",
       "185      312.0\n",
       "207        9.0\n",
       "214      320.0\n",
       "236        4.0\n",
       "243      322.0\n",
       "265        9.0\n",
       "272      329.0\n",
       "294       56.0\n",
       "301      333.0\n",
       "323       23.0\n",
       "330      336.0\n",
       "352       31.0\n",
       "359      341.0\n",
       "381       20.0\n",
       "388      355.0\n",
       "410        6.0\n",
       "417      363.0\n",
       "         ...  \n",
       "2607       0.0\n",
       "2637      54.0\n",
       "2644       0.0\n",
       "2674      89.0\n",
       "2681       0.0\n",
       "2711      53.0\n",
       "2718       0.0\n",
       "2748      43.0\n",
       "2755       0.0\n",
       "2822      94.0\n",
       "2829    1282.0\n",
       "2859      51.0\n",
       "2866    1303.0\n",
       "2896      47.0\n",
       "2903    1333.0\n",
       "2933      39.0\n",
       "2940    1364.0\n",
       "2970      83.0\n",
       "2977    1397.0\n",
       "3007      84.0\n",
       "3014    2962.0\n",
       "3044     134.0\n",
       "3051    1496.0\n",
       "3081      72.0\n",
       "3088    1549.0\n",
       "3118      37.0\n",
       "3125    1648.0\n",
       "3155      54.0\n",
       "3162    1669.0\n",
       "3199    1708.0\n",
       "Name: total, dtype: float64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#  create my own national col\n",
    "col_list = list(sl_data_clean.columns)\n",
    "col_list.remove('date')\n",
    "col_list.remove('variable')\n",
    "col_list.remove('National')\n",
    "sl_data_clean['total'] = sl_data_clean[col_list].sum(axis=1)\n",
    "sl_data_clean['total']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4          0.0\n",
       "11         0.0\n",
       "33         0.0\n",
       "40         0.0\n",
       "62         0.0\n",
       "69         0.0\n",
       "91         0.0\n",
       "98         0.0\n",
       "120        0.0\n",
       "127        0.0\n",
       "149        0.0\n",
       "156        0.0\n",
       "178        0.0\n",
       "185        0.0\n",
       "207        0.0\n",
       "214        0.0\n",
       "236        0.0\n",
       "243        0.0\n",
       "265        0.0\n",
       "272        0.0\n",
       "294        0.0\n",
       "301        0.0\n",
       "323        0.0\n",
       "330        0.0\n",
       "352        0.0\n",
       "359        0.0\n",
       "381        0.0\n",
       "388        0.0\n",
       "410        6.0\n",
       "417      363.0\n",
       "         ...  \n",
       "2607   -1180.0\n",
       "2637       0.0\n",
       "2644   -1206.0\n",
       "2674       0.0\n",
       "2681   -1223.0\n",
       "2711       0.0\n",
       "2718   -1223.0\n",
       "2748       0.0\n",
       "2755   -1240.0\n",
       "2822       0.0\n",
       "2829       0.0\n",
       "2859       0.0\n",
       "2866       0.0\n",
       "2896       0.0\n",
       "2903       0.0\n",
       "2933       0.0\n",
       "2940       0.0\n",
       "2970       0.0\n",
       "2977       0.0\n",
       "3007       0.0\n",
       "3014    1481.0\n",
       "3044      70.0\n",
       "3051       0.0\n",
       "3081       0.0\n",
       "3088       0.0\n",
       "3118       0.0\n",
       "3125       0.0\n",
       "3155       0.0\n",
       "3162       0.0\n",
       "3199       0.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check my total with National\n",
    "sl_data_clean['total'] - sl_data_clean['National']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>total</th>\n",
       "      <th>National</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>363.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>468</th>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>383.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>23.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>413.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>845</th>\n",
       "      <td>31.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852</th>\n",
       "      <td>459.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1345</th>\n",
       "      <td>0.0</td>\n",
       "      <td>550.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454</th>\n",
       "      <td>22.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1461</th>\n",
       "      <td>0.0</td>\n",
       "      <td>703.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1512</th>\n",
       "      <td>16.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519</th>\n",
       "      <td>1426.0</td>\n",
       "      <td>713.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1722</th>\n",
       "      <td>946.0</td>\n",
       "      <td>947.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1802</th>\n",
       "      <td>102.0</td>\n",
       "      <td>69.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1809</th>\n",
       "      <td>1136.0</td>\n",
       "      <td>986.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1180.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1206.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2681</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2718</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2755</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1240.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3014</th>\n",
       "      <td>2962.0</td>\n",
       "      <td>1481.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3044</th>\n",
       "      <td>134.0</td>\n",
       "      <td>64.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       total  National\n",
       "410      6.0       0.0\n",
       "417    363.0       0.0\n",
       "468     15.0       0.0\n",
       "475    383.0       0.0\n",
       "642     23.0       0.0\n",
       "649    413.0       0.0\n",
       "845     31.0       0.0\n",
       "852    459.0       0.0\n",
       "1345     0.0     550.0\n",
       "1454    22.0      40.0\n",
       "1461     0.0     703.0\n",
       "1512    16.0       8.0\n",
       "1519  1426.0     713.0\n",
       "1722   946.0     947.0\n",
       "1802   102.0      69.0\n",
       "1809  1136.0     986.0\n",
       "2607     0.0    1180.0\n",
       "2644     0.0    1206.0\n",
       "2681     0.0    1223.0\n",
       "2718     0.0    1223.0\n",
       "2755     0.0    1240.0\n",
       "3014  2962.0    1481.0\n",
       "3044   134.0      64.0"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_data_clean[['total', 'National']][sl_data_clean['total'] != sl_data_clean['National']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lukas/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bo</th>\n",
       "      <th>Bombali</th>\n",
       "      <th>Bonthe</th>\n",
       "      <th>Kailahun</th>\n",
       "      <th>Kambia</th>\n",
       "      <th>Kenema</th>\n",
       "      <th>Koinadugu</th>\n",
       "      <th>Kono</th>\n",
       "      <th>Moyamba</th>\n",
       "      <th>National</th>\n",
       "      <th>Port Loko</th>\n",
       "      <th>Pujehun</th>\n",
       "      <th>Tonkolili</th>\n",
       "      <th>Western area</th>\n",
       "      <th>Western area rural</th>\n",
       "      <th>Western area urban</th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1345</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>550.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-09-30</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454</th>\n",
       "      <td>5.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-10-06</td>\n",
       "      <td>new_confirmed</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1461</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>703.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-10-06</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1512</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2014-10-08</td>\n",
       "      <td>new_confirmed</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519</th>\n",
       "      <td>49.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>218.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>237.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>713.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>713.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>2014-10-08</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>1426.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1722</th>\n",
       "      <td>56.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>240.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>947.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>140.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-10-16</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>946.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1802</th>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>2014-10-19</td>\n",
       "      <td>new_confirmed</td>\n",
       "      <td>102.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1809</th>\n",
       "      <td>58.0</td>\n",
       "      <td>129.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>220.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>986.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>150.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>2014-10-19</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>1136.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1180.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-11-13</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-11-15</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2681</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-11-16</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2718</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-11-17</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2755</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1240.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2014-11-18</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3014</th>\n",
       "      <td>76.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1481.0</td>\n",
       "      <td>1645.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>147.0</td>\n",
       "      <td>193.0</td>\n",
       "      <td>2014-11-28</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>2962.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3044</th>\n",
       "      <td>2.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>2014-11-29</td>\n",
       "      <td>new_confirmed</td>\n",
       "      <td>134.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Bo  Bombali  Bonthe  Kailahun  Kambia  Kenema  Koinadugu  Kono  \\\n",
       "1345   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "1454   5.0     14.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "1461   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "1512   1.0      0.0     0.0       0.0     0.0     0.0        0.0   3.0   \n",
       "1519  49.0     92.0     1.0     218.0     1.0   237.0        0.0  18.0   \n",
       "1722  56.0    124.0     1.0     219.0     9.0   240.0        0.0  18.0   \n",
       "1802   5.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "1809  58.0    129.0     1.0     220.0     9.0   246.0        0.0  18.0   \n",
       "2607   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "2644   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "2681   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "2718   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "2755   0.0      0.0     0.0       0.0     0.0     0.0        0.0   0.0   \n",
       "3014  76.0    194.0     2.0     228.0    22.0   261.0       26.0  29.0   \n",
       "3044   2.0     77.0     0.0       1.0     1.0     0.0        0.0  15.0   \n",
       "\n",
       "      Moyamba  National  Port Loko  Pujehun  Tonkolili  Western area  \\\n",
       "1345      0.0     550.0        0.0      0.0        0.0           0.0   \n",
       "1454      1.0      40.0        2.0      0.0        0.0           0.0   \n",
       "1461      0.0     703.0        0.0      0.0        0.0           0.0   \n",
       "1512      2.0       8.0        0.0      0.0        0.0           0.0   \n",
       "1519     27.0     713.0       18.0      9.0       29.0           0.0   \n",
       "1722     29.0     947.0       61.0     12.0       37.0         140.0   \n",
       "1802      2.0      69.0        9.0      0.0        4.0          33.0   \n",
       "1809     32.0     986.0       69.0     12.0       40.0         150.0   \n",
       "2607      0.0    1180.0        0.0      0.0        0.0           0.0   \n",
       "2644      0.0    1206.0        0.0      0.0        0.0           0.0   \n",
       "2681      0.0    1223.0        0.0      0.0        0.0           0.0   \n",
       "2718      0.0    1223.0        0.0      0.0        0.0           0.0   \n",
       "2755      0.0    1240.0        0.0      0.0        0.0           0.0   \n",
       "3014     40.0    1481.0     1645.0     16.0       83.0           0.0   \n",
       "3044      0.0      64.0        0.0      0.0        1.0           0.0   \n",
       "\n",
       "      Western area rural  Western area urban        date         variable  \\\n",
       "1345                 0.0                 0.0  2014-09-30  death_confirmed   \n",
       "1454                 0.0                 0.0  2014-10-06    new_confirmed   \n",
       "1461                 0.0                 0.0  2014-10-06  death_confirmed   \n",
       "1512                 8.0                 2.0  2014-10-08    new_confirmed   \n",
       "1519               713.0                14.0  2014-10-08  death_confirmed   \n",
       "1722                 0.0                 0.0  2014-10-16  death_confirmed   \n",
       "1802                19.0                30.0  2014-10-19    new_confirmed   \n",
       "1809                75.0                77.0  2014-10-19  death_confirmed   \n",
       "2607                 0.0                 0.0  2014-11-13  death_confirmed   \n",
       "2644                 0.0                 0.0  2014-11-15  death_confirmed   \n",
       "2681                 0.0                 0.0  2014-11-16  death_confirmed   \n",
       "2718                 0.0                 0.0  2014-11-17  death_confirmed   \n",
       "2755                 0.0                 0.0  2014-11-18  death_confirmed   \n",
       "3014               147.0               193.0  2014-11-28  death_confirmed   \n",
       "3044                11.0                26.0  2014-11-29    new_confirmed   \n",
       "\n",
       "       total  \n",
       "1345     0.0  \n",
       "1454    22.0  \n",
       "1461     0.0  \n",
       "1512    16.0  \n",
       "1519  1426.0  \n",
       "1722   946.0  \n",
       "1802   102.0  \n",
       "1809  1136.0  \n",
       "2607     0.0  \n",
       "2644     0.0  \n",
       "2681     0.0  \n",
       "2718     0.0  \n",
       "2755     0.0  \n",
       "3014  2962.0  \n",
       "3044   134.0  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are several not equal rows. where the National is 0 i can take the total:\n",
    "sl_data_clean[sl_data_clean['total'] != sl_data_clean['National']][sl_data_clean['National'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-12</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>264.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-13</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>273.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-14</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>280.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-15</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>287.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-16</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>297.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-17</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>305.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-18</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>312.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-19</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>320.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-20</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>322.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-21</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>329.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-22</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>333.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-23</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>336.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-24</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>341.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-25</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-26</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>363.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014-11-13</th>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-15</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-16</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>89.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-17</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>53.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-18</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-20</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>94.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1282.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-21</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>51.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1303.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-22</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>47.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1333.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-23</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1364.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-24</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1397.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-28</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>2962.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-11-29</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>134.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1496.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-12-01</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>72.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1549.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-12-04</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1648.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-12-05</th>\n",
       "      <th>new_confirmed</th>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1669.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014-12-06</th>\n",
       "      <th>death_confirmed</th>\n",
       "      <td>1708.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>199 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             total\n",
       "date       variable               \n",
       "2014-08-12 new_confirmed      11.0\n",
       "           death_confirmed   264.0\n",
       "2014-08-13 new_confirmed      15.0\n",
       "           death_confirmed   273.0\n",
       "2014-08-14 new_confirmed      13.0\n",
       "           death_confirmed   280.0\n",
       "2014-08-15 new_confirmed      10.0\n",
       "           death_confirmed   287.0\n",
       "2014-08-16 new_confirmed      18.0\n",
       "           death_confirmed   297.0\n",
       "2014-08-17 new_confirmed       2.0\n",
       "           death_confirmed   305.0\n",
       "2014-08-18 new_confirmed       5.0\n",
       "           death_confirmed   312.0\n",
       "2014-08-19 new_confirmed       9.0\n",
       "           death_confirmed   320.0\n",
       "2014-08-20 new_confirmed       4.0\n",
       "           death_confirmed   322.0\n",
       "2014-08-21 new_confirmed       9.0\n",
       "           death_confirmed   329.0\n",
       "2014-08-22 new_confirmed      56.0\n",
       "           death_confirmed   333.0\n",
       "2014-08-23 new_confirmed      23.0\n",
       "           death_confirmed   336.0\n",
       "2014-08-24 new_confirmed      31.0\n",
       "           death_confirmed   341.0\n",
       "2014-08-25 new_confirmed      20.0\n",
       "           death_confirmed   355.0\n",
       "2014-08-26 new_confirmed       6.0\n",
       "           death_confirmed   363.0\n",
       "...                            ...\n",
       "2014-11-13 death_confirmed     0.0\n",
       "2014-11-15 new_confirmed      54.0\n",
       "           death_confirmed     0.0\n",
       "2014-11-16 new_confirmed      89.0\n",
       "           death_confirmed     0.0\n",
       "2014-11-17 new_confirmed      53.0\n",
       "           death_confirmed     0.0\n",
       "2014-11-18 new_confirmed      43.0\n",
       "           death_confirmed     0.0\n",
       "2014-11-20 new_confirmed      94.0\n",
       "           death_confirmed  1282.0\n",
       "2014-11-21 new_confirmed      51.0\n",
       "           death_confirmed  1303.0\n",
       "2014-11-22 new_confirmed      47.0\n",
       "           death_confirmed  1333.0\n",
       "2014-11-23 new_confirmed      39.0\n",
       "           death_confirmed  1364.0\n",
       "2014-11-24 new_confirmed      83.0\n",
       "           death_confirmed  1397.0\n",
       "2014-11-28 new_confirmed      84.0\n",
       "           death_confirmed  2962.0\n",
       "2014-11-29 new_confirmed     134.0\n",
       "           death_confirmed  1496.0\n",
       "2014-12-01 new_confirmed      72.0\n",
       "           death_confirmed  1549.0\n",
       "2014-12-04 new_confirmed      37.0\n",
       "           death_confirmed  1648.0\n",
       "2014-12-05 new_confirmed      54.0\n",
       "           death_confirmed  1669.0\n",
       "2014-12-06 death_confirmed  1708.0\n",
       "\n",
       "[199 rows x 1 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# in fact I take always the 'total'. National is sometimes just wrong (eg 'Bo' has 77 but National only 64) \n",
    "# -> i dont trust the National\n",
    "sl_data_final = sl_data_clean[['date', 'variable', 'total']]\n",
    "sl_data_final.set_index(['date', 'variable'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Do the same for the other two countries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Beyla</th>\n",
       "      <th>Boffa</th>\n",
       "      <th>Conakry</th>\n",
       "      <th>Coyah</th>\n",
       "      <th>Dabola</th>\n",
       "      <th>Dalaba</th>\n",
       "      <th>Date</th>\n",
       "      <th>Description</th>\n",
       "      <th>Dinguiraye</th>\n",
       "      <th>Dubreka</th>\n",
       "      <th>...</th>\n",
       "      <th>Kouroussa</th>\n",
       "      <th>Lola</th>\n",
       "      <th>Macenta</th>\n",
       "      <th>Mzerekore</th>\n",
       "      <th>Nzerekore</th>\n",
       "      <th>Pita</th>\n",
       "      <th>Siguiri</th>\n",
       "      <th>Telimele</th>\n",
       "      <th>Totals</th>\n",
       "      <th>Yomou</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New cases of suspects</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New cases of probables</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New cases of confirmed</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total new cases registered so far</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total cases of suspects</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Beyla Boffa Conakry Coyah Dabola Dalaba        Date  \\\n",
       "0   NaN     0       5   NaN      0    NaN  2014-08-04   \n",
       "1   NaN     0       0   NaN      0    NaN  2014-08-04   \n",
       "2   NaN     0       1   NaN      0    NaN  2014-08-04   \n",
       "3   NaN     0       6   NaN      0    NaN  2014-08-04   \n",
       "4   NaN     0       9   NaN      0    NaN  2014-08-04   \n",
       "\n",
       "                         Description Dinguiraye Dubreka  ...  Kouroussa Lola  \\\n",
       "0              New cases of suspects          0       0  ...          0  NaN   \n",
       "1             New cases of probables          0       0  ...          0  NaN   \n",
       "2             New cases of confirmed          0       0  ...          0  NaN   \n",
       "3  Total new cases registered so far          0       0  ...          0  NaN   \n",
       "4            Total cases of suspects          0       0  ...          0  NaN   \n",
       "\n",
       "  Macenta Mzerekore Nzerekore Pita Siguiri Telimele Totals Yomou  \n",
       "0       0       NaN         0    0       0        0      5   NaN  \n",
       "1       0       NaN         0    0       0        0      0   NaN  \n",
       "2       0       NaN         0    0       0        0      4   NaN  \n",
       "3       0       NaN         0    0       0        0      9   NaN  \n",
       "4       0       NaN         0    0       0        0     11   NaN  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load\n",
    "files=glob.glob('Data/ebola/guinea_data/*.csv')\n",
    "\n",
    "#\n",
    "guinea_data_array=[]\n",
    "for file in files:\n",
    "    read = pd.read_csv(file)\n",
    "    guinea_data_array.append(read)\n",
    "\n",
    "#\n",
    "guinea_data_array[0].append(guinea_data_array[1], ignore_index=True)\n",
    "guinea = pd.concat(guinea_data_array, ignore_index=True)\n",
    "guinea.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Beyla', 'Boffa', 'Conakry', 'Coyah', 'Dabola', 'Dalaba', 'Date',\n",
       "       'Description', 'Dinguiraye', 'Dubreka', 'Forecariah', 'Gueckedou',\n",
       "       'Kerouane', 'Kindia', 'Kissidougou', 'Kouroussa', 'Lola', 'Macenta',\n",
       "       'Mzerekore', 'Nzerekore', 'Pita', 'Siguiri', 'Telimele', 'Totals',\n",
       "       'Yomou'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "guinea.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['New cases of suspects', 'New cases of probables',\n",
       "       'New cases of confirmed', 'Total new cases registered so far',\n",
       "       'Total cases of suspects', 'Total cases of probables',\n",
       "       'Total cases of confirmed',\n",
       "       'Cumulative (confirmed + probable + suspects)',\n",
       "       'New deaths registered today',\n",
       "       'New deaths registered today (confirmed)',\n",
       "       'New deaths registered today (probables)',\n",
       "       'New deaths registered today (suspects)',\n",
       "       'Total deaths of suspects', 'Total deaths of probables',\n",
       "       'Total deaths of confirmed',\n",
       "       'Total deaths (confirmed + probables + suspects)',\n",
       "       'Total PEC center today', 'Total PEC center today (confirmed)',\n",
       "       'Total PEC center today (probables)',\n",
       "       'Total PEC center today (suspects)',\n",
       "       'Total of deaths in confirmed cases in CTE',\n",
       "       'Total of cured in confirmed cases in CTE',\n",
       "       'Number of male confirmed cases',\n",
       "       'Number of female confirmed cases',\n",
       "       'Number of male probables cases',\n",
       "       'Number of female probables cases', 'Number of male suspects cases',\n",
       "       'Number of female suspects cases', 'Total number of male cases',\n",
       "       'Total number of female cases',\n",
       "       'Number of confirmed cases among health workers',\n",
       "       'Number of deaths of confirmed cases among health workers',\n",
       "       'Number of probables cases among health workers',\n",
       "       'Number of deaths of probables cases among health workers',\n",
       "       'Number of suspects cases among health workers',\n",
       "       'Number of death of confirmed cases among health workers',\n",
       "       'Total contacts registered from start date',\n",
       "       'Number of contacts to follow today',\n",
       "       'Number of contacts out of track',\n",
       "       'Number of contacts followed yesterday',\n",
       "       'Number of patients tested', 'Number of samples collected',\n",
       "       'Total suspected non-class cases', 'New deaths registered',\n",
       "       'Fatality rate for confirmed and probables',\n",
       "       'New cases of confirmed among health workers',\n",
       "       'Total case of confirmed among health workers',\n",
       "       'New deaths registered among health workers',\n",
       "       'Total deaths registered among health workers',\n",
       "       'New admits to CTE so far',\n",
       "       'Total number of hospitalized cases in CTE',\n",
       "       'Total number of exits from CTE',\n",
       "       'Total number of admissions to CTE',\n",
       "       'New contacts registered so far',\n",
       "       'Number of contacts followed today',\n",
       "       'Number of contacts out of the track 21 days',\n",
       "       'Number of contacts lost to follow up',\n",
       "       'Number of samples collected today', 'Number of samples under test',\n",
       "       'Total samples tested'], dtype=object)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "guinea['Description'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "interesting_vars = ['Total cases of confirmed','Total deaths (confirmed + probables + suspects)']\n",
    "guinea_relevant = guinea[guinea['Description'].isin(interesting_vars)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Beyla</th>\n",
       "      <th>Boffa</th>\n",
       "      <th>Conakry</th>\n",
       "      <th>Coyah</th>\n",
       "      <th>Dabola</th>\n",
       "      <th>Dalaba</th>\n",
       "      <th>Dinguiraye</th>\n",
       "      <th>Dubreka</th>\n",
       "      <th>Forecariah</th>\n",
       "      <th>Gueckedou</th>\n",
       "      <th>...</th>\n",
       "      <th>Kouroussa</th>\n",
       "      <th>Lola</th>\n",
       "      <th>Macenta</th>\n",
       "      <th>Mzerekore</th>\n",
       "      <th>Nzerekore</th>\n",
       "      <th>Pita</th>\n",
       "      <th>Siguiri</th>\n",
       "      <th>Telimele</th>\n",
       "      <th>Totals</th>\n",
       "      <th>Yomou</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Date</th>\n",
       "      <th>Description</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-04</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>18</td>\n",
       "      <td>78</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>188</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>351</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>247</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>363</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-26</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>207</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>490</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>34</td>\n",
       "      <td>243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>459</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>128</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>958</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-27</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>207</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23</td>\n",
       "      <td>499</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>268</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>438</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-30</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>211</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>533</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>55</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>271</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>468</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-31</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>214</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>137</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>563</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>55</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>279</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>489</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-02</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>110</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "      <td>219</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>591</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>283</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>501</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-04</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>111</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "      <td>223</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>169</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>621</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>57</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>284</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>116</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>522</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-07</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>113</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>230</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>209</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>678</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>57</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>289</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>557</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-08</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>113</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>230</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>209</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>678</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>57</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>289</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>142</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>557</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-09</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>113</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>230</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>213</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>683</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>57</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>291</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>565</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-11</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>117</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>232</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>232</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>716</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>57</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>292</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>148</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>568</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-14</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>120</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>235</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>245</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>743</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>59</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>292</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>156</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>595</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-16</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>129</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>235</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>771</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>61</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>293</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>165</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>608</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-17</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>129</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>235</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>258</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>771</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>61</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>293</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>165</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>608</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-19</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>131</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>236</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>272</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>788</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>68</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>295</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>171</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>630</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-21</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>133</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>237</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>296</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>818</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>68</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>295</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>172</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>632</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-22</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>136</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>239</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>303</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>832</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>68</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>295</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>175</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>635</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-23</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>139</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>241</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>317</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>854</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>71</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>295</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>180</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>643</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-24</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>149</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>243</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>325</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>876</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>72</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>296</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>182</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>648</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-26</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>160</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>15</td>\n",
       "      <td>244</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>335</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>902</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>73</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>297</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>193</td>\n",
       "      <td>26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>668</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-30</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>166</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>245</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>377</td>\n",
       "      <td>26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>961</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>76</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>300</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>229</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>714</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-10-01</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>170</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>248</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>378</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>977</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>78</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>301</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>246</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>739</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           Beyla Boffa  \\\n",
       "Date       Description                                                   \n",
       "2014-08-04 Total cases of confirmed                          NaN    18   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-08-26 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    34   \n",
       "2014-08-27 Total cases of confirmed                          NaN   NaN   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN   NaN   \n",
       "2014-08-30 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-08-31 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-02 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-04 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-07 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-08 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-09 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-11 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-14 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-16 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-17 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-19 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-21 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-22 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-23 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-24 Total cases of confirmed                          NaN    19   \n",
       "           Total deaths (confirmed + probables + suspects)   NaN    16   \n",
       "2014-09-26 Total cases of confirmed                            1    19   \n",
       "           Total deaths (confirmed + probables + suspects)     0    16   \n",
       "2014-09-30 Total cases of confirmed                            2    19   \n",
       "           Total deaths (confirmed + probables + suspects)     0    16   \n",
       "2014-10-01 Total cases of confirmed                            2    19   \n",
       "           Total deaths (confirmed + probables + suspects)     0    16   \n",
       "\n",
       "                                                           Conakry Coyah  \\\n",
       "Date       Description                                                     \n",
       "2014-08-04 Total cases of confirmed                             78   NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      42   NaN   \n",
       "2014-08-26 Total cases of confirmed                             98   NaN   \n",
       "           Total deaths (confirmed + probables + suspects)     243   NaN   \n",
       "2014-08-27 Total cases of confirmed                            102   NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      49   NaN   \n",
       "2014-08-30 Total cases of confirmed                            109   NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      55   NaN   \n",
       "2014-08-31 Total cases of confirmed                            110   NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      55   NaN   \n",
       "2014-09-02 Total cases of confirmed                            110     3   \n",
       "           Total deaths (confirmed + probables + suspects)      56     0   \n",
       "2014-09-04 Total cases of confirmed                            111     4   \n",
       "           Total deaths (confirmed + probables + suspects)      57     0   \n",
       "2014-09-07 Total cases of confirmed                            113     5   \n",
       "           Total deaths (confirmed + probables + suspects)      57     2   \n",
       "2014-09-08 Total cases of confirmed                            113     5   \n",
       "           Total deaths (confirmed + probables + suspects)      57     2   \n",
       "2014-09-09 Total cases of confirmed                            113     5   \n",
       "           Total deaths (confirmed + probables + suspects)      57     2   \n",
       "2014-09-11 Total cases of confirmed                            117     6   \n",
       "           Total deaths (confirmed + probables + suspects)      57     2   \n",
       "2014-09-14 Total cases of confirmed                            120     7   \n",
       "           Total deaths (confirmed + probables + suspects)      59     4   \n",
       "2014-09-16 Total cases of confirmed                            129     7   \n",
       "           Total deaths (confirmed + probables + suspects)      61     4   \n",
       "2014-09-17 Total cases of confirmed                            129     7   \n",
       "           Total deaths (confirmed + probables + suspects)      61     4   \n",
       "2014-09-19 Total cases of confirmed                            131     7   \n",
       "           Total deaths (confirmed + probables + suspects)      68     5   \n",
       "2014-09-21 Total cases of confirmed                            133     7   \n",
       "           Total deaths (confirmed + probables + suspects)      68     5   \n",
       "2014-09-22 Total cases of confirmed                            136     9   \n",
       "           Total deaths (confirmed + probables + suspects)      68     5   \n",
       "2014-09-23 Total cases of confirmed                            139     9   \n",
       "           Total deaths (confirmed + probables + suspects)      71     5   \n",
       "2014-09-24 Total cases of confirmed                            149     9   \n",
       "           Total deaths (confirmed + probables + suspects)      72     5   \n",
       "2014-09-26 Total cases of confirmed                            160     9   \n",
       "           Total deaths (confirmed + probables + suspects)      73     5   \n",
       "2014-09-30 Total cases of confirmed                            166    12   \n",
       "           Total deaths (confirmed + probables + suspects)      76     5   \n",
       "2014-10-01 Total cases of confirmed                            170    12   \n",
       "           Total deaths (confirmed + probables + suspects)      78     5   \n",
       "\n",
       "                                                           Dabola Dalaba  \\\n",
       "Date       Description                                                     \n",
       "2014-08-04 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-08-26 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      5    NaN   \n",
       "2014-08-27 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-08-30 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-08-31 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-02 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-04 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-07 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-08 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-09 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-11 Total cases of confirmed                             1    NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      4    NaN   \n",
       "2014-09-14 Total cases of confirmed                             1      1   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-16 Total cases of confirmed                             1      2   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-17 Total cases of confirmed                             1      2   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-19 Total cases of confirmed                             1      2   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-21 Total cases of confirmed                             1      2   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-22 Total cases of confirmed                             1      2   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-23 Total cases of confirmed                             1      2   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-24 Total cases of confirmed                             1      3   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-26 Total cases of confirmed                             1      5   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-09-30 Total cases of confirmed                             1      6   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "2014-10-01 Total cases of confirmed                             1      6   \n",
       "           Total deaths (confirmed + probables + suspects)      4      0   \n",
       "\n",
       "                                                           Dinguiraye Dubreka  \\\n",
       "Date       Description                                                          \n",
       "2014-08-04 Total cases of confirmed                                 0       0   \n",
       "           Total deaths (confirmed + probables + suspects)          1       0   \n",
       "2014-08-26 Total cases of confirmed                                 0       8   \n",
       "           Total deaths (confirmed + probables + suspects)          1       3   \n",
       "2014-08-27 Total cases of confirmed                                 0     NaN   \n",
       "           Total deaths (confirmed + probables + suspects)          1     NaN   \n",
       "2014-08-30 Total cases of confirmed                                 0      13   \n",
       "           Total deaths (confirmed + probables + suspects)          1       2   \n",
       "2014-08-31 Total cases of confirmed                                 0      18   \n",
       "           Total deaths (confirmed + probables + suspects)          1       3   \n",
       "2014-09-02 Total cases of confirmed                                 0      20   \n",
       "           Total deaths (confirmed + probables + suspects)          1       3   \n",
       "2014-09-04 Total cases of confirmed                                 0      20   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-07 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-08 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-09 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-11 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-14 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-16 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-17 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       4   \n",
       "2014-09-19 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-09-21 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-09-22 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-09-23 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-09-24 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-09-26 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-09-30 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "2014-10-01 Total cases of confirmed                                 0      21   \n",
       "           Total deaths (confirmed + probables + suspects)          1       7   \n",
       "\n",
       "                                                           Forecariah  \\\n",
       "Date       Description                                                  \n",
       "2014-08-04 Total cases of confirmed                               NaN   \n",
       "           Total deaths (confirmed + probables + suspects)        NaN   \n",
       "2014-08-26 Total cases of confirmed                               NaN   \n",
       "           Total deaths (confirmed + probables + suspects)        NaN   \n",
       "2014-08-27 Total cases of confirmed                               NaN   \n",
       "           Total deaths (confirmed + probables + suspects)        NaN   \n",
       "2014-08-30 Total cases of confirmed                                 5   \n",
       "           Total deaths (confirmed + probables + suspects)          3   \n",
       "2014-08-31 Total cases of confirmed                                 5   \n",
       "           Total deaths (confirmed + probables + suspects)          3   \n",
       "2014-09-02 Total cases of confirmed                                 5   \n",
       "           Total deaths (confirmed + probables + suspects)          3   \n",
       "2014-09-04 Total cases of confirmed                                 5   \n",
       "           Total deaths (confirmed + probables + suspects)          4   \n",
       "2014-09-07 Total cases of confirmed                                 6   \n",
       "           Total deaths (confirmed + probables + suspects)          4   \n",
       "2014-09-08 Total cases of confirmed                                 6   \n",
       "           Total deaths (confirmed + probables + suspects)          4   \n",
       "2014-09-09 Total cases of confirmed                                 6   \n",
       "           Total deaths (confirmed + probables + suspects)          4   \n",
       "2014-09-11 Total cases of confirmed                                 7   \n",
       "           Total deaths (confirmed + probables + suspects)          4   \n",
       "2014-09-14 Total cases of confirmed                                 8   \n",
       "           Total deaths (confirmed + probables + suspects)          5   \n",
       "2014-09-16 Total cases of confirmed                                 9   \n",
       "           Total deaths (confirmed + probables + suspects)          5   \n",
       "2014-09-17 Total cases of confirmed                                 9   \n",
       "           Total deaths (confirmed + probables + suspects)          5   \n",
       "2014-09-19 Total cases of confirmed                                 9   \n",
       "           Total deaths (confirmed + probables + suspects)          7   \n",
       "2014-09-21 Total cases of confirmed                                 9   \n",
       "           Total deaths (confirmed + probables + suspects)          7   \n",
       "2014-09-22 Total cases of confirmed                                 9   \n",
       "           Total deaths (confirmed + probables + suspects)          7   \n",
       "2014-09-23 Total cases of confirmed                                11   \n",
       "           Total deaths (confirmed + probables + suspects)          7   \n",
       "2014-09-24 Total cases of confirmed                                11   \n",
       "           Total deaths (confirmed + probables + suspects)          7   \n",
       "2014-09-26 Total cases of confirmed                                15   \n",
       "           Total deaths (confirmed + probables + suspects)          9   \n",
       "2014-09-30 Total cases of confirmed                                11   \n",
       "           Total deaths (confirmed + probables + suspects)          9   \n",
       "2014-10-01 Total cases of confirmed                                11   \n",
       "           Total deaths (confirmed + probables + suspects)          9   \n",
       "\n",
       "                                                           Gueckedou  ...   \\\n",
       "Date       Description                                                ...    \n",
       "2014-08-04 Total cases of confirmed                              188  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       247  ...    \n",
       "2014-08-26 Total cases of confirmed                              207  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       459  ...    \n",
       "2014-08-27 Total cases of confirmed                              207  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       268  ...    \n",
       "2014-08-30 Total cases of confirmed                              211  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       271  ...    \n",
       "2014-08-31 Total cases of confirmed                              214  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       279  ...    \n",
       "2014-09-02 Total cases of confirmed                              219  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       283  ...    \n",
       "2014-09-04 Total cases of confirmed                              223  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       284  ...    \n",
       "2014-09-07 Total cases of confirmed                              230  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       289  ...    \n",
       "2014-09-08 Total cases of confirmed                              230  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       289  ...    \n",
       "2014-09-09 Total cases of confirmed                              230  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       291  ...    \n",
       "2014-09-11 Total cases of confirmed                              232  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       292  ...    \n",
       "2014-09-14 Total cases of confirmed                              235  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       292  ...    \n",
       "2014-09-16 Total cases of confirmed                              235  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       293  ...    \n",
       "2014-09-17 Total cases of confirmed                              235  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       293  ...    \n",
       "2014-09-19 Total cases of confirmed                              236  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       295  ...    \n",
       "2014-09-21 Total cases of confirmed                              237  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       295  ...    \n",
       "2014-09-22 Total cases of confirmed                              239  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       295  ...    \n",
       "2014-09-23 Total cases of confirmed                              241  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       295  ...    \n",
       "2014-09-24 Total cases of confirmed                              243  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       296  ...    \n",
       "2014-09-26 Total cases of confirmed                              244  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       297  ...    \n",
       "2014-09-30 Total cases of confirmed                              245  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       300  ...    \n",
       "2014-10-01 Total cases of confirmed                              248  ...    \n",
       "           Total deaths (confirmed + probables + suspects)       301  ...    \n",
       "\n",
       "                                                           Kouroussa Lola  \\\n",
       "Date       Description                                                      \n",
       "2014-08-04 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         3  NaN   \n",
       "2014-08-26 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         5  NaN   \n",
       "2014-08-27 Total cases of confirmed                              NaN  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN  NaN   \n",
       "2014-08-30 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-08-31 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-02 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-04 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-07 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-08 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-09 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-11 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-14 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-16 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-17 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-19 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-21 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-22 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-23 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-24 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-26 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-09-30 Total cases of confirmed                                2  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)         4  NaN   \n",
       "2014-10-01 Total cases of confirmed                                2    2   \n",
       "           Total deaths (confirmed + probables + suspects)         4    0   \n",
       "\n",
       "                                                           Macenta Mzerekore  \\\n",
       "Date       Description                                                         \n",
       "2014-08-04 Total cases of confirmed                             28       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      29       NaN   \n",
       "2014-08-26 Total cases of confirmed                             99        10   \n",
       "           Total deaths (confirmed + probables + suspects)     128        12   \n",
       "2014-08-27 Total cases of confirmed                            103       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      64       NaN   \n",
       "2014-08-30 Total cases of confirmed                            117       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      80       NaN   \n",
       "2014-08-31 Total cases of confirmed                            137       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)      92       NaN   \n",
       "2014-09-02 Total cases of confirmed                            146        10   \n",
       "           Total deaths (confirmed + probables + suspects)      99         6   \n",
       "2014-09-04 Total cases of confirmed                            169        10   \n",
       "           Total deaths (confirmed + probables + suspects)     116         6   \n",
       "2014-09-07 Total cases of confirmed                            209       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)     142       NaN   \n",
       "2014-09-08 Total cases of confirmed                            209        13   \n",
       "           Total deaths (confirmed + probables + suspects)     142         6   \n",
       "2014-09-09 Total cases of confirmed                            213        14   \n",
       "           Total deaths (confirmed + probables + suspects)     146         8   \n",
       "2014-09-11 Total cases of confirmed                            232        14   \n",
       "           Total deaths (confirmed + probables + suspects)     148         8   \n",
       "2014-09-14 Total cases of confirmed                            245        17   \n",
       "           Total deaths (confirmed + probables + suspects)     156        21   \n",
       "2014-09-16 Total cases of confirmed                            258       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)     165       NaN   \n",
       "2014-09-17 Total cases of confirmed                            258        17   \n",
       "           Total deaths (confirmed + probables + suspects)     165        20   \n",
       "2014-09-19 Total cases of confirmed                            272        17   \n",
       "           Total deaths (confirmed + probables + suspects)     171        20   \n",
       "2014-09-21 Total cases of confirmed                            296        19   \n",
       "           Total deaths (confirmed + probables + suspects)     172        20   \n",
       "2014-09-22 Total cases of confirmed                            303        19   \n",
       "           Total deaths (confirmed + probables + suspects)     175        20   \n",
       "2014-09-23 Total cases of confirmed                            317        20   \n",
       "           Total deaths (confirmed + probables + suspects)     180        20   \n",
       "2014-09-24 Total cases of confirmed                            325        21   \n",
       "           Total deaths (confirmed + probables + suspects)     182        21   \n",
       "2014-09-26 Total cases of confirmed                            335        21   \n",
       "           Total deaths (confirmed + probables + suspects)     193        26   \n",
       "2014-09-30 Total cases of confirmed                            377        26   \n",
       "           Total deaths (confirmed + probables + suspects)     229        29   \n",
       "2014-10-01 Total cases of confirmed                            378       NaN   \n",
       "           Total deaths (confirmed + probables + suspects)     246       NaN   \n",
       "\n",
       "                                                           Nzerekore Pita  \\\n",
       "Date       Description                                                      \n",
       "2014-08-04 Total cases of confirmed                                4    1   \n",
       "           Total deaths (confirmed + probables + suspects)         1    2   \n",
       "2014-08-26 Total cases of confirmed                              NaN    5   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN   10   \n",
       "2014-08-27 Total cases of confirmed                              NaN  NaN   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN  NaN   \n",
       "2014-08-30 Total cases of confirmed                               10    5   \n",
       "           Total deaths (confirmed + probables + suspects)         6    4   \n",
       "2014-08-31 Total cases of confirmed                               10    6   \n",
       "           Total deaths (confirmed + probables + suspects)         6    4   \n",
       "2014-09-02 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-04 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-07 Total cases of confirmed                               13    6   \n",
       "           Total deaths (confirmed + probables + suspects)         6    3   \n",
       "2014-09-08 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-09 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-11 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-14 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-16 Total cases of confirmed                               17    6   \n",
       "           Total deaths (confirmed + probables + suspects)        20    3   \n",
       "2014-09-17 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-19 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-21 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-22 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-23 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-24 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-26 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-09-30 Total cases of confirmed                              NaN    6   \n",
       "           Total deaths (confirmed + probables + suspects)       NaN    3   \n",
       "2014-10-01 Total cases of confirmed                               31    6   \n",
       "           Total deaths (confirmed + probables + suspects)        32    3   \n",
       "\n",
       "                                                           Siguiri Telimele  \\\n",
       "Date       Description                                                        \n",
       "2014-08-04 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-08-26 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)      10       28   \n",
       "2014-08-27 Total cases of confirmed                            NaN       23   \n",
       "           Total deaths (confirmed + probables + suspects)     NaN       10   \n",
       "2014-08-30 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-08-31 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-02 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-04 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-07 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-08 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-09 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-11 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-14 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-16 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-17 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-19 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-21 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-22 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-23 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-24 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-26 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-09-30 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "2014-10-01 Total cases of confirmed                              6       23   \n",
       "           Total deaths (confirmed + probables + suspects)       3       10   \n",
       "\n",
       "                                                           Totals Yomou  \n",
       "Date       Description                                                   \n",
       "2014-08-04 Total cases of confirmed                           351   NaN  \n",
       "           Total deaths (confirmed + probables + suspects)    363   NaN  \n",
       "2014-08-26 Total cases of confirmed                           490    10  \n",
       "           Total deaths (confirmed + probables + suspects)    958     7  \n",
       "2014-08-27 Total cases of confirmed                           499   NaN  \n",
       "           Total deaths (confirmed + probables + suspects)    438   NaN  \n",
       "2014-08-30 Total cases of confirmed                           533    10  \n",
       "           Total deaths (confirmed + probables + suspects)    468     3  \n",
       "2014-08-31 Total cases of confirmed                           563    10  \n",
       "           Total deaths (confirmed + probables + suspects)    489     3  \n",
       "2014-09-02 Total cases of confirmed                           591    10  \n",
       "           Total deaths (confirmed + probables + suspects)    501     4  \n",
       "2014-09-04 Total cases of confirmed                           621    10  \n",
       "           Total deaths (confirmed + probables + suspects)    522     4  \n",
       "2014-09-07 Total cases of confirmed                           678    11  \n",
       "           Total deaths (confirmed + probables + suspects)    557     5  \n",
       "2014-09-08 Total cases of confirmed                           678    11  \n",
       "           Total deaths (confirmed + probables + suspects)    557     5  \n",
       "2014-09-09 Total cases of confirmed                           683    11  \n",
       "           Total deaths (confirmed + probables + suspects)    565     5  \n",
       "2014-09-11 Total cases of confirmed                           716    11  \n",
       "           Total deaths (confirmed + probables + suspects)    568     5  \n",
       "2014-09-14 Total cases of confirmed                           743    11  \n",
       "           Total deaths (confirmed + probables + suspects)    595     5  \n",
       "2014-09-16 Total cases of confirmed                           771    11  \n",
       "           Total deaths (confirmed + probables + suspects)    608     5  \n",
       "2014-09-17 Total cases of confirmed                           771    11  \n",
       "           Total deaths (confirmed + probables + suspects)    608     5  \n",
       "2014-09-19 Total cases of confirmed                           788    11  \n",
       "           Total deaths (confirmed + probables + suspects)    630     5  \n",
       "2014-09-21 Total cases of confirmed                           818    11  \n",
       "           Total deaths (confirmed + probables + suspects)    632     5  \n",
       "2014-09-22 Total cases of confirmed                           832    11  \n",
       "           Total deaths (confirmed + probables + suspects)    635     5  \n",
       "2014-09-23 Total cases of confirmed                           854    11  \n",
       "           Total deaths (confirmed + probables + suspects)    643     5  \n",
       "2014-09-24 Total cases of confirmed                           876    11  \n",
       "           Total deaths (confirmed + probables + suspects)    648     5  \n",
       "2014-09-26 Total cases of confirmed                           902    11  \n",
       "           Total deaths (confirmed + probables + suspects)    668     5  \n",
       "2014-09-30 Total cases of confirmed                           961    11  \n",
       "           Total deaths (confirmed + probables + suspects)    714     5  \n",
       "2014-10-01 Total cases of confirmed                           977    11  \n",
       "           Total deaths (confirmed + probables + suspects)    739     5  \n",
       "\n",
       "[44 rows x 23 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guinea_relevant.set_index(['Date', 'Description'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6      351.0\n",
       "15     363.0\n",
       "48     490.0\n",
       "55     958.0\n",
       "80     438.0\n",
       "87     402.0\n",
       "112    533.0\n",
       "119    468.0\n",
       "144    563.0\n",
       "151    489.0\n",
       "176    591.0\n",
       "183    501.0\n",
       "208    621.0\n",
       "215    522.0\n",
       "240    678.0\n",
       "247    557.0\n",
       "272    678.0\n",
       "279    557.0\n",
       "304    683.0\n",
       "311    565.0\n",
       "336    716.0\n",
       "343    568.0\n",
       "368    743.0\n",
       "375    595.0\n",
       "400    771.0\n",
       "407    608.0\n",
       "432    771.0\n",
       "439    608.0\n",
       "464    788.0\n",
       "471    630.0\n",
       "496    818.0\n",
       "503    632.0\n",
       "528    832.0\n",
       "535    635.0\n",
       "560    854.0\n",
       "567    643.0\n",
       "592    876.0\n",
       "599    648.0\n",
       "624    906.0\n",
       "631    668.0\n",
       "656    961.0\n",
       "663    713.0\n",
       "688    977.0\n",
       "695    736.0\n",
       "Name: total, dtype: float64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename the data\n",
    "guinea_clean = guinea_relevant\n",
    "\n",
    "#remove cols that have only NaN\n",
    "guinea_clean = guinea_clean.dropna(axis=1, how='all')\n",
    "\n",
    "#remove rows that have only NaN\n",
    "guinea_clean = guinea_clean.dropna(axis=0, thresh=3)\n",
    "\n",
    "# replace all NaNs with 0\n",
    "guinea_clean.fillna(value=0, inplace=True)\n",
    "\n",
    "# then make all types objects\n",
    "def change_to_numeric(data):\n",
    "    col_list = list(data.columns)\n",
    "    col_list.remove('Date')\n",
    "    col_list.remove('Description')\n",
    "    data[col_list] = data[col_list].apply(pd.to_numeric)\n",
    "    \n",
    "change_to_numeric(guinea_clean)\n",
    "\n",
    "\n",
    "#  create my own total col\n",
    "col_list = list(guinea_clean.columns)\n",
    "col_list.remove('Date')\n",
    "col_list.remove('Description')\n",
    "col_list.remove('Totals')\n",
    "guinea_clean['total'] = guinea_clean[col_list].sum(axis=1)\n",
    "guinea_clean['total']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6       0.0\n",
       "15      0.0\n",
       "48      0.0\n",
       "55      0.0\n",
       "80    -61.0\n",
       "87    -36.0\n",
       "112     0.0\n",
       "119     0.0\n",
       "144     0.0\n",
       "151     0.0\n",
       "176     0.0\n",
       "183     0.0\n",
       "208     0.0\n",
       "215     0.0\n",
       "240     0.0\n",
       "247     0.0\n",
       "272     0.0\n",
       "279     0.0\n",
       "304     0.0\n",
       "311     0.0\n",
       "336     0.0\n",
       "343     0.0\n",
       "368     0.0\n",
       "375     0.0\n",
       "400     0.0\n",
       "407     0.0\n",
       "432     0.0\n",
       "439     0.0\n",
       "464     0.0\n",
       "471     0.0\n",
       "496     0.0\n",
       "503     0.0\n",
       "528     0.0\n",
       "535     0.0\n",
       "560     0.0\n",
       "567     0.0\n",
       "592     0.0\n",
       "599     0.0\n",
       "624     4.0\n",
       "631     0.0\n",
       "656     0.0\n",
       "663    -1.0\n",
       "688     0.0\n",
       "695    -3.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check my total with National\n",
    "guinea_clean['total'] - guinea_clean['Totals']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lukas/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Beyla</th>\n",
       "      <th>Boffa</th>\n",
       "      <th>Conakry</th>\n",
       "      <th>Coyah</th>\n",
       "      <th>Dabola</th>\n",
       "      <th>Dalaba</th>\n",
       "      <th>Date</th>\n",
       "      <th>Description</th>\n",
       "      <th>Dinguiraye</th>\n",
       "      <th>Dubreka</th>\n",
       "      <th>...</th>\n",
       "      <th>Lola</th>\n",
       "      <th>Macenta</th>\n",
       "      <th>Mzerekore</th>\n",
       "      <th>Nzerekore</th>\n",
       "      <th>Pita</th>\n",
       "      <th>Siguiri</th>\n",
       "      <th>Telimele</th>\n",
       "      <th>Totals</th>\n",
       "      <th>Yomou</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-08-27</td>\n",
       "      <td>Total cases of confirmed</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>499</td>\n",
       "      <td>0</td>\n",
       "      <td>438.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-08-27</td>\n",
       "      <td>Total deaths (confirmed + probables + suspects)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>438</td>\n",
       "      <td>0</td>\n",
       "      <td>402.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>624</th>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "      <td>2014-09-26</td>\n",
       "      <td>Total cases of confirmed</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>335.0</td>\n",
       "      <td>21</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>902</td>\n",
       "      <td>11</td>\n",
       "      <td>906.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>663</th>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-09-30</td>\n",
       "      <td>Total deaths (confirmed + probables + suspects)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>29</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>714</td>\n",
       "      <td>5</td>\n",
       "      <td>713.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014-10-01</td>\n",
       "      <td>Total deaths (confirmed + probables + suspects)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>739</td>\n",
       "      <td>5</td>\n",
       "      <td>736.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Beyla  Boffa  Conakry  Coyah  Dabola  Dalaba        Date  \\\n",
       "80     0.0    0.0    102.0      0     1.0       0  2014-08-27   \n",
       "87     0.0    0.0     49.0      0     4.0       0  2014-08-27   \n",
       "624    1.0   19.0    160.0      9     1.0       5  2014-09-26   \n",
       "663    0.0   16.0     76.0      5     4.0       0  2014-09-30   \n",
       "695    0.0   16.0     78.0      5     4.0       0  2014-10-01   \n",
       "\n",
       "                                         Description  Dinguiraye  Dubreka  \\\n",
       "80                          Total cases of confirmed         0.0      0.0   \n",
       "87   Total deaths (confirmed + probables + suspects)         1.0      0.0   \n",
       "624                         Total cases of confirmed         0.0     21.0   \n",
       "663  Total deaths (confirmed + probables + suspects)         1.0      7.0   \n",
       "695  Total deaths (confirmed + probables + suspects)         1.0      7.0   \n",
       "\n",
       "     ...    Lola  Macenta  Mzerekore  Nzerekore  Pita  Siguiri  Telimele  \\\n",
       "80   ...       0    103.0          0        0.0   0.0      0.0      23.0   \n",
       "87   ...       0     64.0          0        0.0   0.0      0.0      10.0   \n",
       "624  ...       0    335.0         21        0.0   6.0      6.0      23.0   \n",
       "663  ...       0    229.0         29        0.0   3.0      3.0      10.0   \n",
       "695  ...       0    246.0          0       32.0   3.0      3.0      10.0   \n",
       "\n",
       "     Totals  Yomou  total  \n",
       "80      499      0  438.0  \n",
       "87      438      0  402.0  \n",
       "624     902     11  906.0  \n",
       "663     714      5  713.0  \n",
       "695     739      5  736.0  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are several not equal rows. Where the Totals is 0 i can take the total:\n",
    "guinea_clean[guinea_clean['total'] != guinea_clean['Totals']][guinea_clean['Totals'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Date</th>\n",
       "      <th>Description</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-04</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>351.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>363.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-26</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>490.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>958.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-27</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>438.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>402.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-30</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>533.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>468.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-08-31</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>563.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>489.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-02</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>591.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>501.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-04</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>621.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>522.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-07</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>678.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>557.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-08</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>678.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>557.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-09</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>683.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>565.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-11</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>716.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>568.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-14</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>743.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>595.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-16</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>771.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>608.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-17</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>771.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>608.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-19</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>788.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>630.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-21</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>818.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>632.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-22</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>832.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>635.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-23</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>854.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>643.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-24</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>876.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>648.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-26</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>906.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>668.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-09-30</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>961.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>713.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">2014-10-01</th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <td>977.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total deaths (confirmed + probables + suspects)</th>\n",
       "      <td>736.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            total\n",
       "Date       Description                                           \n",
       "2014-08-04 Total cases of confirmed                         351.0\n",
       "           Total deaths (confirmed + probables + suspects)  363.0\n",
       "2014-08-26 Total cases of confirmed                         490.0\n",
       "           Total deaths (confirmed + probables + suspects)  958.0\n",
       "2014-08-27 Total cases of confirmed                         438.0\n",
       "           Total deaths (confirmed + probables + suspects)  402.0\n",
       "2014-08-30 Total cases of confirmed                         533.0\n",
       "           Total deaths (confirmed + probables + suspects)  468.0\n",
       "2014-08-31 Total cases of confirmed                         563.0\n",
       "           Total deaths (confirmed + probables + suspects)  489.0\n",
       "2014-09-02 Total cases of confirmed                         591.0\n",
       "           Total deaths (confirmed + probables + suspects)  501.0\n",
       "2014-09-04 Total cases of confirmed                         621.0\n",
       "           Total deaths (confirmed + probables + suspects)  522.0\n",
       "2014-09-07 Total cases of confirmed                         678.0\n",
       "           Total deaths (confirmed + probables + suspects)  557.0\n",
       "2014-09-08 Total cases of confirmed                         678.0\n",
       "           Total deaths (confirmed + probables + suspects)  557.0\n",
       "2014-09-09 Total cases of confirmed                         683.0\n",
       "           Total deaths (confirmed + probables + suspects)  565.0\n",
       "2014-09-11 Total cases of confirmed                         716.0\n",
       "           Total deaths (confirmed + probables + suspects)  568.0\n",
       "2014-09-14 Total cases of confirmed                         743.0\n",
       "           Total deaths (confirmed + probables + suspects)  595.0\n",
       "2014-09-16 Total cases of confirmed                         771.0\n",
       "           Total deaths (confirmed + probables + suspects)  608.0\n",
       "2014-09-17 Total cases of confirmed                         771.0\n",
       "           Total deaths (confirmed + probables + suspects)  608.0\n",
       "2014-09-19 Total cases of confirmed                         788.0\n",
       "           Total deaths (confirmed + probables + suspects)  630.0\n",
       "2014-09-21 Total cases of confirmed                         818.0\n",
       "           Total deaths (confirmed + probables + suspects)  632.0\n",
       "2014-09-22 Total cases of confirmed                         832.0\n",
       "           Total deaths (confirmed + probables + suspects)  635.0\n",
       "2014-09-23 Total cases of confirmed                         854.0\n",
       "           Total deaths (confirmed + probables + suspects)  643.0\n",
       "2014-09-24 Total cases of confirmed                         876.0\n",
       "           Total deaths (confirmed + probables + suspects)  648.0\n",
       "2014-09-26 Total cases of confirmed                         906.0\n",
       "           Total deaths (confirmed + probables + suspects)  668.0\n",
       "2014-09-30 Total cases of confirmed                         961.0\n",
       "           Total deaths (confirmed + probables + suspects)  713.0\n",
       "2014-10-01 Total cases of confirmed                         977.0\n",
       "           Total deaths (confirmed + probables + suspects)  736.0"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# They are never far off, to I can taky my total\n",
    "guinea_final = guinea_clean[['Date', 'Description', 'total']]\n",
    "guinea_final.set_index(['Date', 'Description'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### and for the last"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Bomi County', 'Bong County', 'Date', 'Gbarpolu County', 'Grand Bassa',\n",
       "       'Grand Cape Mount', 'Grand Gedeh', 'Grand Kru', 'Lofa County',\n",
       "       'Margibi County', 'Maryland County', 'Montserrado County', 'National',\n",
       "       'Nimba County', 'River Gee County', 'RiverCess County', 'Sinoe County',\n",
       "       'Unnamed: 18', 'Variable'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load\n",
    "files=glob.glob('Data/ebola/liberia_data/*.csv')\n",
    "\n",
    "#\n",
    "liberia_data_array=[]\n",
    "for file in files:\n",
    "    read = pd.read_csv(file)\n",
    "    liberia_data_array.append(read)\n",
    "\n",
    "#\n",
    "liberia_data_array[0].append(liberia_data_array[1], ignore_index=True)\n",
    "liberia = pd.concat(liberia_data_array, ignore_index=True)\n",
    "liberia.head()\n",
    "\n",
    "#\n",
    "liberia.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Specimens collected', 'Specimens pending for testing',\n",
       "       'Total specimens tested', 'Newly reported deaths',\n",
       "       'Total death/s in confirmed cases',\n",
       "       'Total death/s in probable cases',\n",
       "       'Total death/s in suspected cases',\n",
       "       'Total death/s in confirmed, probable, suspected cases',\n",
       "       'Case Fatality Rate (CFR) - Confirmed & Probable Cases',\n",
       "       'Newly reported contacts', 'Total contacts listed',\n",
       "       'Currently under follow-up', 'Contacts seen',\n",
       "       'Contacts who completed 21 day follow-up',\n",
       "       'Contacts lost to follow-up', 'New admissions',\n",
       "       'Total no. currently in Treatment Units', 'Total discharges',\n",
       "       'Cumulative admission/isolation', 'Newly Reported Cases in HCW',\n",
       "       'Cumulative cases among HCW', 'Newly Reported deaths in HCW',\n",
       "       'Cumulative deaths among HCW', 'New Case/s (Suspected)',\n",
       "       'New Case/s (Probable)', 'New case/s (confirmed)',\n",
       "       'Total suspected cases', 'Total probable cases',\n",
       "       'Total confirmed cases',\n",
       "       'Total Number of Confirmed Cases of Sierra Leonean Nationality',\n",
       "       'Total Number of Confirmed Cases of Guinean Nationality',\n",
       "       'Cumulative (confirmed + probable + suspected)',\n",
       "       'Total Case/s (Suspected)', 'Total Case/s (Probable)',\n",
       "       'Total case/s (confirmed)', 'Cumulative CFR',\n",
       "       'Cumulative (confirmed + probable + suspects)',\n",
       "       'Total death/s in confirmed,  probable, suspected cases',\n",
       "       'Cumulative confirmed, probable and suspected cases',\n",
       "       'Total death/s in confirmed, \\n probable, suspected cases',\n",
       "       'Case Fatality Rate (CFR) - \\n Confirmed & Probable Cases',\n",
       "       'Contacts who completed 21 day \\n follow-up',\n",
       "       'Total no. currently in Treatment \\n Units',\n",
       "       'Total Number of Confirmed Cases \\n of Sierra Leonean Nationality',\n",
       "       'Total Number of Confirmed Cases \\n of Guinean Nationality'], dtype=object)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "liberia['Variable'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Bomi County</th>\n",
       "      <th>Bong County</th>\n",
       "      <th>Gbarpolu County</th>\n",
       "      <th>Grand Bassa</th>\n",
       "      <th>Grand Cape Mount</th>\n",
       "      <th>Grand Gedeh</th>\n",
       "      <th>Grand Kru</th>\n",
       "      <th>Lofa County</th>\n",
       "      <th>Margibi County</th>\n",
       "      <th>Maryland County</th>\n",
       "      <th>Montserrado County</th>\n",
       "      <th>National</th>\n",
       "      <th>Nimba County</th>\n",
       "      <th>River Gee County</th>\n",
       "      <th>RiverCess County</th>\n",
       "      <th>Sinoe County</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Date</th>\n",
       "      <th>Variable</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/16/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/17/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/22/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/24/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/25/14</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/28/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/29/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/1/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/2/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/3/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/7/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/8/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/10/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/13/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/17/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">10/29/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>84.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>504.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1408.0</td>\n",
       "      <td>2609.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">10/30/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>84.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1419.0</td>\n",
       "      <td>2636.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10/31/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">11/2/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>85.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>346.0</td>\n",
       "      <td>516.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1465.0</td>\n",
       "      <td>2700.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/4/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/8/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/14/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/15/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/19/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/20/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/21/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/23/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/24/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/26/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/27/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/28/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/29/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/30/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/1/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/2/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/3/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/4/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1630.0</td>\n",
       "      <td>2867.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/5/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1630.0</td>\n",
       "      <td>2867.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/6/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1630.0</td>\n",
       "      <td>2869.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/7/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1630.0</td>\n",
       "      <td>2869.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/8/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>325.0</td>\n",
       "      <td>379.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1667.0</td>\n",
       "      <td>2927.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/9/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>137.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1676.0</td>\n",
       "      <td>2946.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>176 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               Bomi County  \\\n",
       "Date       Variable                                                          \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...          1.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...          1.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...          2.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "...                                                                    ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...         84.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...         84.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/31/2014 New case/s (confirmed)                                      NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...         85.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "11/4/2014  New case/s (confirmed)                                      NaN   \n",
       "11/8/2014  New case/s (confirmed)                                      NaN   \n",
       "11/14/2014 New case/s (confirmed)                                      NaN   \n",
       "11/15/2014 New case/s (confirmed)                                      NaN   \n",
       "11/19/2014 New case/s (confirmed)                                      NaN   \n",
       "11/20/2014 New case/s (confirmed)                                      NaN   \n",
       "11/21/2014 New case/s (confirmed)                                      NaN   \n",
       "11/23/2014 New case/s (confirmed)                                      NaN   \n",
       "11/24/2014 New case/s (confirmed)                                      NaN   \n",
       "11/26/2014 New case/s (confirmed)                                      NaN   \n",
       "11/27/2014 New case/s (confirmed)                                      0.0   \n",
       "11/28/2014 New case/s (confirmed)                                      0.0   \n",
       "11/29/2014 New case/s (confirmed)                                      0.0   \n",
       "11/30/2014 New case/s (confirmed)                                      0.0   \n",
       "12/1/2014  New case/s (confirmed)                                      0.0   \n",
       "12/2/2014  New case/s (confirmed)                                      0.0   \n",
       "12/3/2014  New case/s (confirmed)                                      NaN   \n",
       "12/4/2014  New case/s (confirmed)                                    135.0   \n",
       "12/5/2014  New case/s (confirmed)                                    135.0   \n",
       "12/6/2014  New case/s (confirmed)                                    135.0   \n",
       "12/7/2014  New case/s (confirmed)                                    135.0   \n",
       "12/8/2014  New case/s (confirmed)                                    135.0   \n",
       "12/9/2014  New case/s (confirmed)                                    137.0   \n",
       "\n",
       "                                                               Bong County  \\\n",
       "Date       Variable                                                          \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...          0.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "...                                                                    ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...         77.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...         76.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/31/2014 New case/s (confirmed)                                      NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...         77.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "11/4/2014  New case/s (confirmed)                                      NaN   \n",
       "11/8/2014  New case/s (confirmed)                                      NaN   \n",
       "11/14/2014 New case/s (confirmed)                                      NaN   \n",
       "11/15/2014 New case/s (confirmed)                                      NaN   \n",
       "11/19/2014 New case/s (confirmed)                                      NaN   \n",
       "11/20/2014 New case/s (confirmed)                                      NaN   \n",
       "11/21/2014 New case/s (confirmed)                                      NaN   \n",
       "11/23/2014 New case/s (confirmed)                                      NaN   \n",
       "11/24/2014 New case/s (confirmed)                                      NaN   \n",
       "11/26/2014 New case/s (confirmed)                                      NaN   \n",
       "11/27/2014 New case/s (confirmed)                                      0.0   \n",
       "11/28/2014 New case/s (confirmed)                                      0.0   \n",
       "11/29/2014 New case/s (confirmed)                                      0.0   \n",
       "11/30/2014 New case/s (confirmed)                                      0.0   \n",
       "12/1/2014  New case/s (confirmed)                                      0.0   \n",
       "12/2/2014  New case/s (confirmed)                                      0.0   \n",
       "12/3/2014  New case/s (confirmed)                                      NaN   \n",
       "12/4/2014  New case/s (confirmed)                                    135.0   \n",
       "12/5/2014  New case/s (confirmed)                                    135.0   \n",
       "12/6/2014  New case/s (confirmed)                                    135.0   \n",
       "12/7/2014  New case/s (confirmed)                                    135.0   \n",
       "12/8/2014  New case/s (confirmed)                                    135.0   \n",
       "12/9/2014  New case/s (confirmed)                                    135.0   \n",
       "\n",
       "                                                               Gbarpolu County  \\\n",
       "Date       Variable                                                              \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "...                                                                        ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...              3.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...              3.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "10/31/2014 New case/s (confirmed)                                          NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...              3.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "11/4/2014  New case/s (confirmed)                                          NaN   \n",
       "11/8/2014  New case/s (confirmed)                                          NaN   \n",
       "11/14/2014 New case/s (confirmed)                                          NaN   \n",
       "11/15/2014 New case/s (confirmed)                                          NaN   \n",
       "11/19/2014 New case/s (confirmed)                                          NaN   \n",
       "11/20/2014 New case/s (confirmed)                                          NaN   \n",
       "11/21/2014 New case/s (confirmed)                                          NaN   \n",
       "11/23/2014 New case/s (confirmed)                                          NaN   \n",
       "11/24/2014 New case/s (confirmed)                                          NaN   \n",
       "11/26/2014 New case/s (confirmed)                                          NaN   \n",
       "11/27/2014 New case/s (confirmed)                                          0.0   \n",
       "11/28/2014 New case/s (confirmed)                                          0.0   \n",
       "11/29/2014 New case/s (confirmed)                                          0.0   \n",
       "11/30/2014 New case/s (confirmed)                                          0.0   \n",
       "12/1/2014  New case/s (confirmed)                                          0.0   \n",
       "12/2/2014  New case/s (confirmed)                                          0.0   \n",
       "12/3/2014  New case/s (confirmed)                                          NaN   \n",
       "12/4/2014  New case/s (confirmed)                                         10.0   \n",
       "12/5/2014  New case/s (confirmed)                                         10.0   \n",
       "12/6/2014  New case/s (confirmed)                                         10.0   \n",
       "12/7/2014  New case/s (confirmed)                                         10.0   \n",
       "12/8/2014  New case/s (confirmed)                                         10.0   \n",
       "12/9/2014  New case/s (confirmed)                                         10.0   \n",
       "\n",
       "                                                               Grand Bassa  \\\n",
       "Date       Variable                                                          \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "...                                                                    ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...         50.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...         50.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/31/2014 New case/s (confirmed)                                      NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...         50.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "11/4/2014  New case/s (confirmed)                                      NaN   \n",
       "11/8/2014  New case/s (confirmed)                                      NaN   \n",
       "11/14/2014 New case/s (confirmed)                                      NaN   \n",
       "11/15/2014 New case/s (confirmed)                                      NaN   \n",
       "11/19/2014 New case/s (confirmed)                                      NaN   \n",
       "11/20/2014 New case/s (confirmed)                                      NaN   \n",
       "11/21/2014 New case/s (confirmed)                                      NaN   \n",
       "11/23/2014 New case/s (confirmed)                                      NaN   \n",
       "11/24/2014 New case/s (confirmed)                                      NaN   \n",
       "11/26/2014 New case/s (confirmed)                                      NaN   \n",
       "11/27/2014 New case/s (confirmed)                                      0.0   \n",
       "11/28/2014 New case/s (confirmed)                                      3.0   \n",
       "11/29/2014 New case/s (confirmed)                                      1.0   \n",
       "11/30/2014 New case/s (confirmed)                                      0.0   \n",
       "12/1/2014  New case/s (confirmed)                                      0.0   \n",
       "12/2/2014  New case/s (confirmed)                                      0.0   \n",
       "12/3/2014  New case/s (confirmed)                                      NaN   \n",
       "12/4/2014  New case/s (confirmed)                                     40.0   \n",
       "12/5/2014  New case/s (confirmed)                                     40.0   \n",
       "12/6/2014  New case/s (confirmed)                                     61.0   \n",
       "12/7/2014  New case/s (confirmed)                                     61.0   \n",
       "12/8/2014  New case/s (confirmed)                                     41.0   \n",
       "12/9/2014  New case/s (confirmed)                                     43.0   \n",
       "\n",
       "                                                               Grand Cape Mount  \\\n",
       "Date       Variable                                                               \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "...                                                                         ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...              31.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...              43.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "10/31/2014 New case/s (confirmed)                                           NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...              43.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "11/4/2014  New case/s (confirmed)                                           NaN   \n",
       "11/8/2014  New case/s (confirmed)                                           NaN   \n",
       "11/14/2014 New case/s (confirmed)                                           NaN   \n",
       "11/15/2014 New case/s (confirmed)                                           NaN   \n",
       "11/19/2014 New case/s (confirmed)                                           NaN   \n",
       "11/20/2014 New case/s (confirmed)                                           NaN   \n",
       "11/21/2014 New case/s (confirmed)                                           NaN   \n",
       "11/23/2014 New case/s (confirmed)                                           NaN   \n",
       "11/24/2014 New case/s (confirmed)                                           NaN   \n",
       "11/26/2014 New case/s (confirmed)                                           NaN   \n",
       "11/27/2014 New case/s (confirmed)                                           2.0   \n",
       "11/28/2014 New case/s (confirmed)                                           0.0   \n",
       "11/29/2014 New case/s (confirmed)                                           0.0   \n",
       "11/30/2014 New case/s (confirmed)                                           0.0   \n",
       "12/1/2014  New case/s (confirmed)                                           0.0   \n",
       "12/2/2014  New case/s (confirmed)                                           0.0   \n",
       "12/3/2014  New case/s (confirmed)                                           NaN   \n",
       "12/4/2014  New case/s (confirmed)                                          59.0   \n",
       "12/5/2014  New case/s (confirmed)                                          59.0   \n",
       "12/6/2014  New case/s (confirmed)                                          59.0   \n",
       "12/7/2014  New case/s (confirmed)                                          59.0   \n",
       "12/8/2014  New case/s (confirmed)                                          67.0   \n",
       "12/9/2014  New case/s (confirmed)                                          68.0   \n",
       "\n",
       "                                                               Grand Gedeh  \\\n",
       "Date       Variable                                                          \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...          NaN   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "...                                                                    ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...          3.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...          3.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/31/2014 New case/s (confirmed)                                      NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...          3.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "11/4/2014  New case/s (confirmed)                                      NaN   \n",
       "11/8/2014  New case/s (confirmed)                                      NaN   \n",
       "11/14/2014 New case/s (confirmed)                                      NaN   \n",
       "11/15/2014 New case/s (confirmed)                                      NaN   \n",
       "11/19/2014 New case/s (confirmed)                                      NaN   \n",
       "11/20/2014 New case/s (confirmed)                                      NaN   \n",
       "11/21/2014 New case/s (confirmed)                                      NaN   \n",
       "11/23/2014 New case/s (confirmed)                                      NaN   \n",
       "11/24/2014 New case/s (confirmed)                                      NaN   \n",
       "11/26/2014 New case/s (confirmed)                                      NaN   \n",
       "11/27/2014 New case/s (confirmed)                                      0.0   \n",
       "11/28/2014 New case/s (confirmed)                                      0.0   \n",
       "11/29/2014 New case/s (confirmed)                                      0.0   \n",
       "11/30/2014 New case/s (confirmed)                                      0.0   \n",
       "12/1/2014  New case/s (confirmed)                                      NaN   \n",
       "12/2/2014  New case/s (confirmed)                                      NaN   \n",
       "12/3/2014  New case/s (confirmed)                                      NaN   \n",
       "12/4/2014  New case/s (confirmed)                                      2.0   \n",
       "12/5/2014  New case/s (confirmed)                                      2.0   \n",
       "12/6/2014  New case/s (confirmed)                                      2.0   \n",
       "12/7/2014  New case/s (confirmed)                                      2.0   \n",
       "12/8/2014  New case/s (confirmed)                                      2.0   \n",
       "12/9/2014  New case/s (confirmed)                                      2.0   \n",
       "\n",
       "                                                               Grand Kru  \\\n",
       "Date       Variable                                                        \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...        NaN   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...        NaN   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...        NaN   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...        NaN   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...        NaN   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...        0.0   \n",
       "           New case/s (confirmed)                                    0.0   \n",
       "...                                                                  ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...       22.0   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...       22.0   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "10/31/2014 New case/s (confirmed)                                    NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...       22.0   \n",
       "           New case/s (confirmed)                                    NaN   \n",
       "11/4/2014  New case/s (confirmed)                                    NaN   \n",
       "11/8/2014  New case/s (confirmed)                                    NaN   \n",
       "11/14/2014 New case/s (confirmed)                                    NaN   \n",
       "11/15/2014 New case/s (confirmed)                                    NaN   \n",
       "11/19/2014 New case/s (confirmed)                                    NaN   \n",
       "11/20/2014 New case/s (confirmed)                                    NaN   \n",
       "11/21/2014 New case/s (confirmed)                                    NaN   \n",
       "11/23/2014 New case/s (confirmed)                                    NaN   \n",
       "11/24/2014 New case/s (confirmed)                                    NaN   \n",
       "11/26/2014 New case/s (confirmed)                                    NaN   \n",
       "11/27/2014 New case/s (confirmed)                                    0.0   \n",
       "11/28/2014 New case/s (confirmed)                                    0.0   \n",
       "11/29/2014 New case/s (confirmed)                                    0.0   \n",
       "11/30/2014 New case/s (confirmed)                                    0.0   \n",
       "12/1/2014  New case/s (confirmed)                                    0.0   \n",
       "12/2/2014  New case/s (confirmed)                                    0.0   \n",
       "12/3/2014  New case/s (confirmed)                                    NaN   \n",
       "12/4/2014  New case/s (confirmed)                                    4.0   \n",
       "12/5/2014  New case/s (confirmed)                                    4.0   \n",
       "12/6/2014  New case/s (confirmed)                                    4.0   \n",
       "12/7/2014  New case/s (confirmed)                                    4.0   \n",
       "12/8/2014  New case/s (confirmed)                                    4.0   \n",
       "12/9/2014  New case/s (confirmed)                                    4.0   \n",
       "\n",
       "                                                               Lofa County  \\\n",
       "Date       Variable                                                          \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...          8.0   \n",
       "           New case/s (confirmed)                                      1.0   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...          8.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...         14.0   \n",
       "           New case/s (confirmed)                                      3.0   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...         17.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...         19.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...         27.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...         27.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...         32.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...         35.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...         35.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...         41.0   \n",
       "           New case/s (confirmed)                                      4.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...         42.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...         46.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...         49.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...         53.0   \n",
       "           New case/s (confirmed)                                      0.0   \n",
       "...                                                                    ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...        345.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...        345.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "10/31/2014 New case/s (confirmed)                                      NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...        346.0   \n",
       "           New case/s (confirmed)                                      NaN   \n",
       "11/4/2014  New case/s (confirmed)                                      NaN   \n",
       "11/8/2014  New case/s (confirmed)                                      NaN   \n",
       "11/14/2014 New case/s (confirmed)                                      NaN   \n",
       "11/15/2014 New case/s (confirmed)                                      NaN   \n",
       "11/19/2014 New case/s (confirmed)                                      NaN   \n",
       "11/20/2014 New case/s (confirmed)                                      NaN   \n",
       "11/21/2014 New case/s (confirmed)                                      NaN   \n",
       "11/23/2014 New case/s (confirmed)                                      NaN   \n",
       "11/24/2014 New case/s (confirmed)                                      NaN   \n",
       "11/26/2014 New case/s (confirmed)                                      NaN   \n",
       "11/27/2014 New case/s (confirmed)                                      0.0   \n",
       "11/28/2014 New case/s (confirmed)                                      0.0   \n",
       "11/29/2014 New case/s (confirmed)                                      0.0   \n",
       "11/30/2014 New case/s (confirmed)                                      0.0   \n",
       "12/1/2014  New case/s (confirmed)                                      0.0   \n",
       "12/2/2014  New case/s (confirmed)                                      0.0   \n",
       "12/3/2014  New case/s (confirmed)                                      NaN   \n",
       "12/4/2014  New case/s (confirmed)                                    326.0   \n",
       "12/5/2014  New case/s (confirmed)                                    326.0   \n",
       "12/6/2014  New case/s (confirmed)                                    326.0   \n",
       "12/7/2014  New case/s (confirmed)                                    326.0   \n",
       "12/8/2014  New case/s (confirmed)                                    325.0   \n",
       "12/9/2014  New case/s (confirmed)                                    326.0   \n",
       "\n",
       "                                                               Margibi County  \\\n",
       "Date       Variable                                                             \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...             NaN   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...             NaN   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...             NaN   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...             NaN   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...             0.0   \n",
       "           New case/s (confirmed)                                         1.0   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...             2.0   \n",
       "           New case/s (confirmed)                                         0.0   \n",
       "...                                                                       ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...           504.0   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...           508.0   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "10/31/2014 New case/s (confirmed)                                         NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...           516.0   \n",
       "           New case/s (confirmed)                                         NaN   \n",
       "11/4/2014  New case/s (confirmed)                                         NaN   \n",
       "11/8/2014  New case/s (confirmed)                                         NaN   \n",
       "11/14/2014 New case/s (confirmed)                                         NaN   \n",
       "11/15/2014 New case/s (confirmed)                                         NaN   \n",
       "11/19/2014 New case/s (confirmed)                                         NaN   \n",
       "11/20/2014 New case/s (confirmed)                                         NaN   \n",
       "11/21/2014 New case/s (confirmed)                                         NaN   \n",
       "11/23/2014 New case/s (confirmed)                                         NaN   \n",
       "11/24/2014 New case/s (confirmed)                                         NaN   \n",
       "11/26/2014 New case/s (confirmed)                                         NaN   \n",
       "11/27/2014 New case/s (confirmed)                                         0.0   \n",
       "11/28/2014 New case/s (confirmed)                                         0.0   \n",
       "11/29/2014 New case/s (confirmed)                                         0.0   \n",
       "11/30/2014 New case/s (confirmed)                                         0.0   \n",
       "12/1/2014  New case/s (confirmed)                                         0.0   \n",
       "12/2/2014  New case/s (confirmed)                                         0.0   \n",
       "12/3/2014  New case/s (confirmed)                                         NaN   \n",
       "12/4/2014  New case/s (confirmed)                                       365.0   \n",
       "12/5/2014  New case/s (confirmed)                                       365.0   \n",
       "12/6/2014  New case/s (confirmed)                                       365.0   \n",
       "12/7/2014  New case/s (confirmed)                                       365.0   \n",
       "12/8/2014  New case/s (confirmed)                                       379.0   \n",
       "12/9/2014  New case/s (confirmed)                                       383.0   \n",
       "\n",
       "                                                               Maryland County  \\\n",
       "Date       Variable                                                              \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...              NaN   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...              0.0   \n",
       "           New case/s (confirmed)                                          0.0   \n",
       "...                                                                        ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...             14.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...             14.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "10/31/2014 New case/s (confirmed)                                          NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...             14.0   \n",
       "           New case/s (confirmed)                                          NaN   \n",
       "11/4/2014  New case/s (confirmed)                                          NaN   \n",
       "11/8/2014  New case/s (confirmed)                                          NaN   \n",
       "11/14/2014 New case/s (confirmed)                                          NaN   \n",
       "11/15/2014 New case/s (confirmed)                                          NaN   \n",
       "11/19/2014 New case/s (confirmed)                                          NaN   \n",
       "11/20/2014 New case/s (confirmed)                                          NaN   \n",
       "11/21/2014 New case/s (confirmed)                                          NaN   \n",
       "11/23/2014 New case/s (confirmed)                                          NaN   \n",
       "11/24/2014 New case/s (confirmed)                                          NaN   \n",
       "11/26/2014 New case/s (confirmed)                                          NaN   \n",
       "11/27/2014 New case/s (confirmed)                                          0.0   \n",
       "11/28/2014 New case/s (confirmed)                                          0.0   \n",
       "11/29/2014 New case/s (confirmed)                                          0.0   \n",
       "11/30/2014 New case/s (confirmed)                                          0.0   \n",
       "12/1/2014  New case/s (confirmed)                                          0.0   \n",
       "12/2/2014  New case/s (confirmed)                                          0.0   \n",
       "12/3/2014  New case/s (confirmed)                                          NaN   \n",
       "12/4/2014  New case/s (confirmed)                                          4.0   \n",
       "12/5/2014  New case/s (confirmed)                                          4.0   \n",
       "12/6/2014  New case/s (confirmed)                                          4.0   \n",
       "12/7/2014  New case/s (confirmed)                                          4.0   \n",
       "12/8/2014  New case/s (confirmed)                                          4.0   \n",
       "12/9/2014  New case/s (confirmed)                                          4.0   \n",
       "\n",
       "                                                               Montserrado County  \\\n",
       "Date       Variable                                                                 \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...                 0.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...                 8.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...                11.0   \n",
       "           New case/s (confirmed)                                             2.0   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...                15.0   \n",
       "           New case/s (confirmed)                                             4.0   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...                18.0   \n",
       "           New case/s (confirmed)                                             1.0   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...                20.0   \n",
       "           New case/s (confirmed)                                             1.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...                20.0   \n",
       "           New case/s (confirmed)                                             2.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...                27.0   \n",
       "           New case/s (confirmed)                                             2.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...                29.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...                33.0   \n",
       "           New case/s (confirmed)                                             1.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...                35.0   \n",
       "           New case/s (confirmed)                                             1.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...                36.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...                36.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...                42.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...                51.0   \n",
       "           New case/s (confirmed)                                             0.0   \n",
       "...                                                                           ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...              1408.0   \n",
       "           New case/s (confirmed)                                             NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...              1419.0   \n",
       "           New case/s (confirmed)                                             NaN   \n",
       "10/31/2014 New case/s (confirmed)                                             NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...              1465.0   \n",
       "           New case/s (confirmed)                                             NaN   \n",
       "11/4/2014  New case/s (confirmed)                                             NaN   \n",
       "11/8/2014  New case/s (confirmed)                                             NaN   \n",
       "11/14/2014 New case/s (confirmed)                                             NaN   \n",
       "11/15/2014 New case/s (confirmed)                                             NaN   \n",
       "11/19/2014 New case/s (confirmed)                                             NaN   \n",
       "11/20/2014 New case/s (confirmed)                                             NaN   \n",
       "11/21/2014 New case/s (confirmed)                                             NaN   \n",
       "11/23/2014 New case/s (confirmed)                                             NaN   \n",
       "11/24/2014 New case/s (confirmed)                                             NaN   \n",
       "11/26/2014 New case/s (confirmed)                                             NaN   \n",
       "11/27/2014 New case/s (confirmed)                                            10.0   \n",
       "11/28/2014 New case/s (confirmed)                                             4.0   \n",
       "11/29/2014 New case/s (confirmed)                                             8.0   \n",
       "11/30/2014 New case/s (confirmed)                                            10.0   \n",
       "12/1/2014  New case/s (confirmed)                                             0.0   \n",
       "12/2/2014  New case/s (confirmed)                                             0.0   \n",
       "12/3/2014  New case/s (confirmed)                                             NaN   \n",
       "12/4/2014  New case/s (confirmed)                                          1630.0   \n",
       "12/5/2014  New case/s (confirmed)                                          1630.0   \n",
       "12/6/2014  New case/s (confirmed)                                          1630.0   \n",
       "12/7/2014  New case/s (confirmed)                                          1630.0   \n",
       "12/8/2014  New case/s (confirmed)                                          1667.0   \n",
       "12/9/2014  New case/s (confirmed)                                          1676.0   \n",
       "\n",
       "                                                               National  \\\n",
       "Date       Variable                                                       \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...      16.0   \n",
       "           New case/s (confirmed)                                   1.0   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...      16.0   \n",
       "           New case/s (confirmed)                                   0.0   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...      25.0   \n",
       "           New case/s (confirmed)                                   5.0   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...      32.0   \n",
       "           New case/s (confirmed)                                   4.0   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...      37.0   \n",
       "           New case/s (confirmed)                                   2.0   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...      49.0   \n",
       "           New case/s (confirmed)                                   1.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...      49.0   \n",
       "           New case/s (confirmed)                                   2.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...      61.0   \n",
       "           New case/s (confirmed)                                   2.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...      66.0   \n",
       "           New case/s (confirmed)                                   0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...      70.0   \n",
       "           New case/s (confirmed)                                   1.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...      78.0   \n",
       "           New case/s (confirmed)                                   5.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...      80.0   \n",
       "           New case/s (confirmed)                                   0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...      85.0   \n",
       "           New case/s (confirmed)                                   0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...      94.0   \n",
       "           New case/s (confirmed)                                   0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...     108.0   \n",
       "           New case/s (confirmed)                                   0.0   \n",
       "...                                                                 ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...    2609.0   \n",
       "           New case/s (confirmed)                                   NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...    2636.0   \n",
       "           New case/s (confirmed)                                   NaN   \n",
       "10/31/2014 New case/s (confirmed)                                   NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...    2700.0   \n",
       "           New case/s (confirmed)                                   NaN   \n",
       "11/4/2014  New case/s (confirmed)                                   NaN   \n",
       "11/8/2014  New case/s (confirmed)                                   NaN   \n",
       "11/14/2014 New case/s (confirmed)                                   0.0   \n",
       "11/15/2014 New case/s (confirmed)                                   NaN   \n",
       "11/19/2014 New case/s (confirmed)                                   NaN   \n",
       "11/20/2014 New case/s (confirmed)                                   NaN   \n",
       "11/21/2014 New case/s (confirmed)                                   NaN   \n",
       "11/23/2014 New case/s (confirmed)                                   NaN   \n",
       "11/24/2014 New case/s (confirmed)                                   NaN   \n",
       "11/26/2014 New case/s (confirmed)                                   NaN   \n",
       "11/27/2014 New case/s (confirmed)                                  12.0   \n",
       "11/28/2014 New case/s (confirmed)                                   7.0   \n",
       "11/29/2014 New case/s (confirmed)                                  10.0   \n",
       "11/30/2014 New case/s (confirmed)                                  10.0   \n",
       "12/1/2014  New case/s (confirmed)                                   1.0   \n",
       "12/2/2014  New case/s (confirmed)                                   9.0   \n",
       "12/3/2014  New case/s (confirmed)                                   NaN   \n",
       "12/4/2014  New case/s (confirmed)                                2867.0   \n",
       "12/5/2014  New case/s (confirmed)                                2867.0   \n",
       "12/6/2014  New case/s (confirmed)                                2869.0   \n",
       "12/7/2014  New case/s (confirmed)                                2869.0   \n",
       "12/8/2014  New case/s (confirmed)                                2927.0   \n",
       "12/9/2014  New case/s (confirmed)                                2946.0   \n",
       "\n",
       "                                                               Nimba County  \\\n",
       "Date       Variable                                                           \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "...                                                                     ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...          39.0   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...          39.0   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "10/31/2014 New case/s (confirmed)                                       NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...          44.0   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "11/4/2014  New case/s (confirmed)                                       NaN   \n",
       "11/8/2014  New case/s (confirmed)                                       NaN   \n",
       "11/14/2014 New case/s (confirmed)                                       NaN   \n",
       "11/15/2014 New case/s (confirmed)                                       NaN   \n",
       "11/19/2014 New case/s (confirmed)                                       NaN   \n",
       "11/20/2014 New case/s (confirmed)                                       NaN   \n",
       "11/21/2014 New case/s (confirmed)                                       NaN   \n",
       "11/23/2014 New case/s (confirmed)                                       NaN   \n",
       "11/24/2014 New case/s (confirmed)                                       NaN   \n",
       "11/26/2014 New case/s (confirmed)                                       NaN   \n",
       "11/27/2014 New case/s (confirmed)                                       0.0   \n",
       "11/28/2014 New case/s (confirmed)                                       0.0   \n",
       "11/29/2014 New case/s (confirmed)                                       0.0   \n",
       "11/30/2014 New case/s (confirmed)                                       0.0   \n",
       "12/1/2014  New case/s (confirmed)                                       1.0   \n",
       "12/2/2014  New case/s (confirmed)                                       9.0   \n",
       "12/3/2014  New case/s (confirmed)                                       NaN   \n",
       "12/4/2014  New case/s (confirmed)                                     114.0   \n",
       "12/5/2014  New case/s (confirmed)                                     114.0   \n",
       "12/6/2014  New case/s (confirmed)                                     114.0   \n",
       "12/7/2014  New case/s (confirmed)                                     114.0   \n",
       "12/8/2014  New case/s (confirmed)                                     114.0   \n",
       "12/9/2014  New case/s (confirmed)                                     114.0   \n",
       "\n",
       "                                                               River Gee County  \\\n",
       "Date       Variable                                                               \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "...                                                                         ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...               8.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...               8.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "10/31/2014 New case/s (confirmed)                                           NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...               8.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "11/4/2014  New case/s (confirmed)                                           NaN   \n",
       "11/8/2014  New case/s (confirmed)                                           NaN   \n",
       "11/14/2014 New case/s (confirmed)                                           NaN   \n",
       "11/15/2014 New case/s (confirmed)                                           NaN   \n",
       "11/19/2014 New case/s (confirmed)                                           NaN   \n",
       "11/20/2014 New case/s (confirmed)                                           NaN   \n",
       "11/21/2014 New case/s (confirmed)                                           NaN   \n",
       "11/23/2014 New case/s (confirmed)                                           NaN   \n",
       "11/24/2014 New case/s (confirmed)                                           NaN   \n",
       "11/26/2014 New case/s (confirmed)                                           NaN   \n",
       "11/27/2014 New case/s (confirmed)                                           0.0   \n",
       "11/28/2014 New case/s (confirmed)                                           0.0   \n",
       "11/29/2014 New case/s (confirmed)                                           0.0   \n",
       "11/30/2014 New case/s (confirmed)                                           0.0   \n",
       "12/1/2014  New case/s (confirmed)                                           0.0   \n",
       "12/2/2014  New case/s (confirmed)                                           0.0   \n",
       "12/3/2014  New case/s (confirmed)                                           NaN   \n",
       "12/4/2014  New case/s (confirmed)                                           7.0   \n",
       "12/5/2014  New case/s (confirmed)                                           7.0   \n",
       "12/6/2014  New case/s (confirmed)                                           7.0   \n",
       "12/7/2014  New case/s (confirmed)                                           7.0   \n",
       "12/8/2014  New case/s (confirmed)                                           7.0   \n",
       "12/9/2014  New case/s (confirmed)                                           7.0   \n",
       "\n",
       "                                                               RiverCess County  \\\n",
       "Date       Variable                                                               \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...               NaN   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...               0.0   \n",
       "           New case/s (confirmed)                                           0.0   \n",
       "...                                                                         ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...              11.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...              12.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "10/31/2014 New case/s (confirmed)                                           NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...              14.0   \n",
       "           New case/s (confirmed)                                           NaN   \n",
       "11/4/2014  New case/s (confirmed)                                           NaN   \n",
       "11/8/2014  New case/s (confirmed)                                           NaN   \n",
       "11/14/2014 New case/s (confirmed)                                           NaN   \n",
       "11/15/2014 New case/s (confirmed)                                           NaN   \n",
       "11/19/2014 New case/s (confirmed)                                           NaN   \n",
       "11/20/2014 New case/s (confirmed)                                           NaN   \n",
       "11/21/2014 New case/s (confirmed)                                           NaN   \n",
       "11/23/2014 New case/s (confirmed)                                           NaN   \n",
       "11/24/2014 New case/s (confirmed)                                           NaN   \n",
       "11/26/2014 New case/s (confirmed)                                           NaN   \n",
       "11/27/2014 New case/s (confirmed)                                           0.0   \n",
       "11/28/2014 New case/s (confirmed)                                           0.0   \n",
       "11/29/2014 New case/s (confirmed)                                           0.0   \n",
       "11/30/2014 New case/s (confirmed)                                           0.0   \n",
       "12/1/2014  New case/s (confirmed)                                           0.0   \n",
       "12/2/2014  New case/s (confirmed)                                           0.0   \n",
       "12/3/2014  New case/s (confirmed)                                           NaN   \n",
       "12/4/2014  New case/s (confirmed)                                          20.0   \n",
       "12/5/2014  New case/s (confirmed)                                          20.0   \n",
       "12/6/2014  New case/s (confirmed)                                          20.0   \n",
       "12/7/2014  New case/s (confirmed)                                          20.0   \n",
       "12/8/2014  New case/s (confirmed)                                          20.0   \n",
       "12/9/2014  New case/s (confirmed)                                          20.0   \n",
       "\n",
       "                                                               Sinoe County  \\\n",
       "Date       Variable                                                           \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...           NaN   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...           0.0   \n",
       "           New case/s (confirmed)                                       0.0   \n",
       "...                                                                     ...   \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...          10.0   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...          10.0   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "10/31/2014 New case/s (confirmed)                                       NaN   \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...          10.0   \n",
       "           New case/s (confirmed)                                       NaN   \n",
       "11/4/2014  New case/s (confirmed)                                       NaN   \n",
       "11/8/2014  New case/s (confirmed)                                       NaN   \n",
       "11/14/2014 New case/s (confirmed)                                       NaN   \n",
       "11/15/2014 New case/s (confirmed)                                       NaN   \n",
       "11/19/2014 New case/s (confirmed)                                       NaN   \n",
       "11/20/2014 New case/s (confirmed)                                       NaN   \n",
       "11/21/2014 New case/s (confirmed)                                       NaN   \n",
       "11/23/2014 New case/s (confirmed)                                       NaN   \n",
       "11/24/2014 New case/s (confirmed)                                       NaN   \n",
       "11/26/2014 New case/s (confirmed)                                       NaN   \n",
       "11/27/2014 New case/s (confirmed)                                       0.0   \n",
       "11/28/2014 New case/s (confirmed)                                       0.0   \n",
       "11/29/2014 New case/s (confirmed)                                       0.0   \n",
       "11/30/2014 New case/s (confirmed)                                       0.0   \n",
       "12/1/2014  New case/s (confirmed)                                       0.0   \n",
       "12/2/2014  New case/s (confirmed)                                       0.0   \n",
       "12/3/2014  New case/s (confirmed)                                       NaN   \n",
       "12/4/2014  New case/s (confirmed)                                      16.0   \n",
       "12/5/2014  New case/s (confirmed)                                      16.0   \n",
       "12/6/2014  New case/s (confirmed)                                      16.0   \n",
       "12/7/2014  New case/s (confirmed)                                      16.0   \n",
       "12/8/2014  New case/s (confirmed)                                      16.0   \n",
       "12/9/2014  New case/s (confirmed)                                      16.0   \n",
       "\n",
       "                                                               Unnamed: 18  \n",
       "Date       Variable                                                         \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "6/25/14    Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "...                                                                    ...  \n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "10/31/2014 New case/s (confirmed)                                      NaN  \n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...          NaN  \n",
       "           New case/s (confirmed)                                      NaN  \n",
       "11/4/2014  New case/s (confirmed)                                      NaN  \n",
       "11/8/2014  New case/s (confirmed)                                      NaN  \n",
       "11/14/2014 New case/s (confirmed)                                      NaN  \n",
       "11/15/2014 New case/s (confirmed)                                      NaN  \n",
       "11/19/2014 New case/s (confirmed)                                      NaN  \n",
       "11/20/2014 New case/s (confirmed)                                      NaN  \n",
       "11/21/2014 New case/s (confirmed)                                      NaN  \n",
       "11/23/2014 New case/s (confirmed)                                      NaN  \n",
       "11/24/2014 New case/s (confirmed)                                      NaN  \n",
       "11/26/2014 New case/s (confirmed)                                      NaN  \n",
       "11/27/2014 New case/s (confirmed)                                      NaN  \n",
       "11/28/2014 New case/s (confirmed)                                      NaN  \n",
       "11/29/2014 New case/s (confirmed)                                      NaN  \n",
       "11/30/2014 New case/s (confirmed)                                      NaN  \n",
       "12/1/2014  New case/s (confirmed)                                      0.0  \n",
       "12/2/2014  New case/s (confirmed)                                      NaN  \n",
       "12/3/2014  New case/s (confirmed)                                      NaN  \n",
       "12/4/2014  New case/s (confirmed)                                      NaN  \n",
       "12/5/2014  New case/s (confirmed)                                      NaN  \n",
       "12/6/2014  New case/s (confirmed)                                      NaN  \n",
       "12/7/2014  New case/s (confirmed)                                      NaN  \n",
       "12/8/2014  New case/s (confirmed)                                      NaN  \n",
       "12/9/2014  New case/s (confirmed)                                      NaN  \n",
       "\n",
       "[176 rows x 17 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interesting_vars = ['Total death/s in confirmed, probable, suspected cases','New case/s (confirmed)']\n",
    "liberia_relevant = liberia[liberia['Variable'].isin(interesting_vars)]\n",
    "\n",
    "liberia_relevant.set_index(['Date', 'Variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7       -8.0\n",
       "25       0.0\n",
       "38       0.0\n",
       "56       0.0\n",
       "69       0.0\n",
       "87       0.0\n",
       "100      0.0\n",
       "118      0.0\n",
       "131      0.0\n",
       "149      0.0\n",
       "162      0.0\n",
       "180      0.0\n",
       "193      0.0\n",
       "211      0.0\n",
       "224      0.0\n",
       "242      0.0\n",
       "255      0.0\n",
       "273      0.0\n",
       "286      0.0\n",
       "304      0.0\n",
       "317      0.0\n",
       "335      0.0\n",
       "348      0.0\n",
       "366      0.0\n",
       "379      0.0\n",
       "397      0.0\n",
       "410      0.0\n",
       "428      0.0\n",
       "441      0.0\n",
       "459      0.0\n",
       "        ... \n",
       "2060     0.0\n",
       "2074     0.0\n",
       "2092     0.0\n",
       "2105     0.0\n",
       "2123     0.0\n",
       "2136     0.0\n",
       "2154     0.0\n",
       "2167     0.0\n",
       "2185    -4.0\n",
       "2199     0.0\n",
       "2217     0.0\n",
       "2231     0.0\n",
       "2249     0.0\n",
       "2263   -68.0\n",
       "2295     0.0\n",
       "2327     0.0\n",
       "2391     0.0\n",
       "2505     0.0\n",
       "2761     0.0\n",
       "2793     0.0\n",
       "2825    -1.0\n",
       "2857     0.0\n",
       "2889     0.0\n",
       "2921     0.0\n",
       "2985     0.0\n",
       "3017     0.0\n",
       "3049    19.0\n",
       "3081    19.0\n",
       "3113    -1.0\n",
       "3145    -1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename the data\n",
    "liberia_clean = liberia_relevant\n",
    "\n",
    "#remove cols that have only NaN\n",
    "liberia_clean = liberia_clean.dropna(axis=1, how='all')\n",
    "\n",
    "#remove rows that have only NaN\n",
    "liberia_clean = liberia_clean.dropna(axis=0, thresh=3)\n",
    "\n",
    "# replace all NaNs with 0\n",
    "liberia_clean.fillna(value=0, inplace=True)\n",
    "\n",
    "# then make all types objects\n",
    "def change_to_numeric(data):\n",
    "    col_list = list(data.columns)\n",
    "    col_list.remove('Date')\n",
    "    col_list.remove('Variable')\n",
    "    data[col_list] = data[col_list].apply(pd.to_numeric)\n",
    "    \n",
    "change_to_numeric(liberia_clean)\n",
    "\n",
    "\n",
    "#  create my own total col\n",
    "col_list = list(liberia_clean.columns)\n",
    "col_list.remove('Date')\n",
    "col_list.remove('Variable')\n",
    "col_list.remove('National')\n",
    "liberia_clean['total'] = liberia_clean[col_list].sum(axis=1)\n",
    "liberia_clean['total']\n",
    "\n",
    "# check my total with National\n",
    "liberia_clean['total'] - liberia_clean['National']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19.0"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(liberia_clean['total'] - liberia_clean['National']).max(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lukas/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bomi County</th>\n",
       "      <th>Bong County</th>\n",
       "      <th>Date</th>\n",
       "      <th>Gbarpolu County</th>\n",
       "      <th>Grand Bassa</th>\n",
       "      <th>Grand Cape Mount</th>\n",
       "      <th>Grand Gedeh</th>\n",
       "      <th>Grand Kru</th>\n",
       "      <th>Lofa County</th>\n",
       "      <th>Margibi County</th>\n",
       "      <th>Maryland County</th>\n",
       "      <th>Montserrado County</th>\n",
       "      <th>National</th>\n",
       "      <th>Nimba County</th>\n",
       "      <th>River Gee County</th>\n",
       "      <th>RiverCess County</th>\n",
       "      <th>Sinoe County</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "      <th>Variable</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6/16/2014</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Total death/s in confirmed, probable, suspecte...</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1570</th>\n",
       "      <td>68.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>10/4/14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>437.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2199.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Total death/s in confirmed, probable, suspecte...</td>\n",
       "      <td>1119.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1658</th>\n",
       "      <td>69.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>10/8/2014</td>\n",
       "      <td>2.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>439.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2344.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Total death/s in confirmed, probable, suspecte...</td>\n",
       "      <td>1174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1690</th>\n",
       "      <td>69.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>10/9/2014</td>\n",
       "      <td>2.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>441.0</td>\n",
       "      <td>268.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1207.0</td>\n",
       "      <td>2389.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Total death/s in confirmed, probable, suspecte...</td>\n",
       "      <td>2378.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2185</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10/23/2014</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2263</th>\n",
       "      <td>79.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10/28/2014</td>\n",
       "      <td>50.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>344.0</td>\n",
       "      <td>489.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1308.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>2471.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Total death/s in confirmed, probable, suspecte...</td>\n",
       "      <td>2403.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2825</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11/29/2014</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3049</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>12/6/2014</td>\n",
       "      <td>10.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1630.0</td>\n",
       "      <td>2869.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>2888.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3081</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>12/7/2014</td>\n",
       "      <td>10.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1630.0</td>\n",
       "      <td>2869.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>2888.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3113</th>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>12/8/2014</td>\n",
       "      <td>10.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>325.0</td>\n",
       "      <td>379.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1667.0</td>\n",
       "      <td>2927.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>2926.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3145</th>\n",
       "      <td>137.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>12/9/2014</td>\n",
       "      <td>10.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1676.0</td>\n",
       "      <td>2946.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>2945.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Bomi County  Bong County        Date  Gbarpolu County  Grand Bassa  \\\n",
       "7             0.0          0.0   6/16/2014              0.0          0.0   \n",
       "1570         68.0        131.0     10/4/14              0.0         37.0   \n",
       "1658         69.0        153.0   10/8/2014              2.0         40.0   \n",
       "1690         69.0        158.0   10/9/2014              2.0         40.0   \n",
       "2185          0.0          0.0  10/23/2014              0.0          0.0   \n",
       "2263         79.0          3.0  10/28/2014             50.0         28.0   \n",
       "2825          0.0          0.0  11/29/2014              0.0          1.0   \n",
       "3049        135.0        135.0   12/6/2014             10.0         61.0   \n",
       "3081        135.0        135.0   12/7/2014             10.0         61.0   \n",
       "3113        135.0        135.0   12/8/2014             10.0         41.0   \n",
       "3145        137.0        135.0   12/9/2014             10.0         43.0   \n",
       "\n",
       "      Grand Cape Mount  Grand Gedeh  Grand Kru  Lofa County  Margibi County  \\\n",
       "7                  0.0          0.0        0.0          8.0             0.0   \n",
       "1570              18.0          2.0       13.0        437.0           250.0   \n",
       "1658              22.0          2.0       17.0        439.0           261.0   \n",
       "1690              22.0          2.0       17.0        441.0           268.0   \n",
       "2185               0.0          0.0        0.0          0.0             2.0   \n",
       "2263               3.0         22.0      344.0        489.0            11.0   \n",
       "2825               0.0          0.0        0.0          0.0             0.0   \n",
       "3049              59.0          2.0        4.0        326.0           365.0   \n",
       "3081              59.0          2.0        4.0        326.0           365.0   \n",
       "3113              67.0          2.0        4.0        325.0           379.0   \n",
       "3145              68.0          2.0        4.0        326.0           383.0   \n",
       "\n",
       "      Maryland County  Montserrado County  National  Nimba County  \\\n",
       "7                 0.0                 0.0      16.0           0.0   \n",
       "1570              9.0                 1.0    2199.0         137.0   \n",
       "1658              9.0                 7.0    2344.0         137.0   \n",
       "1690              9.0              1207.0    2389.0         137.0   \n",
       "2185              0.0                 0.0       6.0           0.0   \n",
       "2263           1308.0                37.0    2471.0           3.0   \n",
       "2825              0.0                 8.0      10.0           0.0   \n",
       "3049              4.0              1630.0    2869.0         114.0   \n",
       "3081              4.0              1630.0    2869.0         114.0   \n",
       "3113              4.0              1667.0    2927.0         114.0   \n",
       "3145              4.0              1676.0    2946.0         114.0   \n",
       "\n",
       "      River Gee County  RiverCess County  Sinoe County  Unnamed: 18  \\\n",
       "7                  0.0               0.0           0.0          0.0   \n",
       "1570               6.0               8.0           2.0          0.0   \n",
       "1658               6.0               8.0           2.0          0.0   \n",
       "1690               6.0               0.0           0.0          0.0   \n",
       "2185               0.0               0.0           0.0          0.0   \n",
       "2263               8.0               8.0          10.0          0.0   \n",
       "2825               0.0               0.0           0.0          0.0   \n",
       "3049               7.0              20.0          16.0          0.0   \n",
       "3081               7.0              20.0          16.0          0.0   \n",
       "3113               7.0              20.0          16.0          0.0   \n",
       "3145               7.0              20.0          16.0          0.0   \n",
       "\n",
       "                                               Variable   total  \n",
       "7     Total death/s in confirmed, probable, suspecte...     8.0  \n",
       "1570  Total death/s in confirmed, probable, suspecte...  1119.0  \n",
       "1658  Total death/s in confirmed, probable, suspecte...  1174.0  \n",
       "1690  Total death/s in confirmed, probable, suspecte...  2378.0  \n",
       "2185                             New case/s (confirmed)     2.0  \n",
       "2263  Total death/s in confirmed, probable, suspecte...  2403.0  \n",
       "2825                             New case/s (confirmed)     9.0  \n",
       "3049                             New case/s (confirmed)  2888.0  \n",
       "3081                             New case/s (confirmed)  2888.0  \n",
       "3113                             New case/s (confirmed)  2926.0  \n",
       "3145                             New case/s (confirmed)  2945.0  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are several not equal rows. Where the Totals is 0 i can take the total:\n",
    "liberia_clean[liberia_clean['total'] != liberia_clean['National']][liberia_clean['National'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Date</th>\n",
       "      <th>Variable</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/16/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/17/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/22/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/24/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/25/14</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/28/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">6/29/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/1/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/2/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>66.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/3/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/7/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>78.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/8/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>80.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/10/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>85.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/13/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>94.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/17/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>108.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10/22/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">7/26/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>145.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">9/10/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>1296.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">10/3/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2154.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">10/23/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2104.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">10/24/14</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2104.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">10/25/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2106.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10/28/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2403.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10/29/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2609.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10/30/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2636.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/2/2014</th>\n",
       "      <th>Total death/s in confirmed, probable, suspected cases</th>\n",
       "      <td>2700.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/14/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/27/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/28/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/29/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11/30/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/1/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/2/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/4/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2867.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/5/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2867.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/6/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2888.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/7/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2888.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/8/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2926.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12/9/2014</th>\n",
       "      <th>New case/s (confirmed)</th>\n",
       "      <td>2945.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>159 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                total\n",
       "Date       Variable                                                  \n",
       "6/16/2014  Total death/s in confirmed, probable, suspected...     8.0\n",
       "           New case/s (confirmed)                                 1.0\n",
       "6/17/2014  Total death/s in confirmed, probable, suspected...    16.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "6/22/2014  Total death/s in confirmed, probable, suspected...    25.0\n",
       "           New case/s (confirmed)                                 5.0\n",
       "6/24/2014  Total death/s in confirmed, probable, suspected...    32.0\n",
       "           New case/s (confirmed)                                 4.0\n",
       "6/25/14    Total death/s in confirmed, probable, suspected...    37.0\n",
       "           New case/s (confirmed)                                 2.0\n",
       "6/28/2014  Total death/s in confirmed, probable, suspected...    49.0\n",
       "           New case/s (confirmed)                                 1.0\n",
       "6/29/2014  Total death/s in confirmed, probable, suspected...    49.0\n",
       "           New case/s (confirmed)                                 2.0\n",
       "7/1/2014   Total death/s in confirmed, probable, suspected...    61.0\n",
       "           New case/s (confirmed)                                 2.0\n",
       "7/2/2014   Total death/s in confirmed, probable, suspected...    66.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "7/3/2014   Total death/s in confirmed, probable, suspected...    70.0\n",
       "           New case/s (confirmed)                                 1.0\n",
       "7/7/2014   Total death/s in confirmed, probable, suspected...    78.0\n",
       "           New case/s (confirmed)                                 5.0\n",
       "7/8/2014   Total death/s in confirmed, probable, suspected...    80.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "7/10/2014  Total death/s in confirmed, probable, suspected...    85.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "7/13/2014  Total death/s in confirmed, probable, suspected...    94.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "7/17/2014  Total death/s in confirmed, probable, suspected...   108.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "...                                                               ...\n",
       "10/22/2014 New case/s (confirmed)                                 0.0\n",
       "7/26/2014  Total death/s in confirmed, probable, suspected...   145.0\n",
       "           New case/s (confirmed)                                 6.0\n",
       "9/10/2014  Total death/s in confirmed, probable, suspected...  1296.0\n",
       "           New case/s (confirmed)                                 2.0\n",
       "10/3/2014  Total death/s in confirmed, probable, suspected...  2154.0\n",
       "           New case/s (confirmed)                                 2.0\n",
       "10/23/2014 Total death/s in confirmed, probable, suspected...  2104.0\n",
       "           New case/s (confirmed)                                 2.0\n",
       "10/24/14   Total death/s in confirmed, probable, suspected...  2104.0\n",
       "           New case/s (confirmed)                                 0.0\n",
       "10/25/2014 Total death/s in confirmed, probable, suspected...  2106.0\n",
       "           New case/s (confirmed)                                 1.0\n",
       "10/28/2014 Total death/s in confirmed, probable, suspected...  2403.0\n",
       "10/29/2014 Total death/s in confirmed, probable, suspected...  2609.0\n",
       "10/30/2014 Total death/s in confirmed, probable, suspected...  2636.0\n",
       "11/2/2014  Total death/s in confirmed, probable, suspected...  2700.0\n",
       "11/14/2014 New case/s (confirmed)                                 0.0\n",
       "11/27/2014 New case/s (confirmed)                                12.0\n",
       "11/28/2014 New case/s (confirmed)                                 7.0\n",
       "11/29/2014 New case/s (confirmed)                                 9.0\n",
       "11/30/2014 New case/s (confirmed)                                10.0\n",
       "12/1/2014  New case/s (confirmed)                                 1.0\n",
       "12/2/2014  New case/s (confirmed)                                 9.0\n",
       "12/4/2014  New case/s (confirmed)                              2867.0\n",
       "12/5/2014  New case/s (confirmed)                              2867.0\n",
       "12/6/2014  New case/s (confirmed)                              2888.0\n",
       "12/7/2014  New case/s (confirmed)                              2888.0\n",
       "12/8/2014  New case/s (confirmed)                              2926.0\n",
       "12/9/2014  New case/s (confirmed)                              2945.0\n",
       "\n",
       "[159 rows x 1 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# They are never far off, to I can taky my total\n",
    "liberia_final = liberia_clean[['Date', 'Variable', 'total']]\n",
    "liberia_final.set_index(['Date', 'Variable'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Merge the 3 countries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# rename variables\n",
    "new_infected = 'new_infected'\n",
    "new_death = 'new_death'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_death</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>264.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>2014-08-13</td>\n",
       "      <td>new_death</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2014-08-13</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>273.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>2014-08-14</td>\n",
       "      <td>new_death</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          date      variable  total\n",
       "4   2014-08-12     new_death   11.0\n",
       "11  2014-08-12  new_infected  264.0\n",
       "33  2014-08-13     new_death   15.0\n",
       "40  2014-08-13  new_infected  273.0\n",
       "62  2014-08-14     new_death   13.0"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_final = sl_data_final\n",
    "# we take the variable names 'new_confirmed' and 'death_confirmed'\n",
    "sl_final['variable'][sl_final['variable'] == 'new_confirmed'] = new_death\n",
    "sl_final['variable'][sl_final['variable'] == 'death_confirmed'] = new_infected\n",
    "sl_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Date</th>\n",
       "      <th>Description</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>new_death</td>\n",
       "      <td>351.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>363.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>2014-08-26</td>\n",
       "      <td>new_death</td>\n",
       "      <td>490.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>2014-08-26</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>958.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>2014-08-27</td>\n",
       "      <td>new_death</td>\n",
       "      <td>438.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Date   Description  total\n",
       "6   2014-08-04     new_death  351.0\n",
       "15  2014-08-04  new_infected  363.0\n",
       "48  2014-08-26     new_death  490.0\n",
       "55  2014-08-26  new_infected  958.0\n",
       "80  2014-08-27     new_death  438.0"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guinea_final['Description'][guinea_final['Description'] == 'Total cases of confirmed'] = new_death\n",
    "guinea_final['Description'][guinea_final['Description'] == 'Total deaths (confirmed + probables + suspects)'] = new_infected\n",
    "guinea_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Date</th>\n",
       "      <th>Variable</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6/16/2014</td>\n",
       "      <td>new_death</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>6/16/2014</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>6/17/2014</td>\n",
       "      <td>new_death</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>6/17/2014</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>6/22/2014</td>\n",
       "      <td>new_death</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Date      Variable  total\n",
       "7   6/16/2014     new_death    8.0\n",
       "25  6/16/2014  new_infected    1.0\n",
       "38  6/17/2014     new_death   16.0\n",
       "56  6/17/2014  new_infected    0.0\n",
       "69  6/22/2014     new_death   25.0"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "liberia_final['Variable'][liberia_final['Variable'] == 'Total death/s in confirmed, probable, suspected cases'] = new_death\n",
    "liberia_final['Variable'][liberia_final['Variable'] == 'New case/s (confirmed)'] = new_infected\n",
    "liberia_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Date</th>\n",
       "      <th>Variable</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6-16-2014</td>\n",
       "      <td>new_death</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>6-16-2014</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>6-17-2014</td>\n",
       "      <td>new_death</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>6-17-2014</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>6-22-2014</td>\n",
       "      <td>new_death</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Date      Variable  total\n",
       "7   6-16-2014     new_death    8.0\n",
       "25  6-16-2014  new_infected    1.0\n",
       "38  6-17-2014     new_death   16.0\n",
       "56  6-17-2014  new_infected    0.0\n",
       "69  6-22-2014     new_death   25.0"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make date of liberia the same as the other two\n",
    "liberia_final['Date'] = liberia_final['Date'].apply(lambda x: x.replace('/', '-')) # replace '/' by '-'\n",
    "liberia_final['Date'].apply(lambda x: x.split('-')[2] + '-' + x.split('-')[1] + '-' + x.split('-')[0])\n",
    "liberia_final.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th>total</th>\n",
       "      <th>country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_death</td>\n",
       "      <td>11.0</td>\n",
       "      <td>sl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>264.0</td>\n",
       "      <td>sl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>2014-08-13</td>\n",
       "      <td>new_death</td>\n",
       "      <td>15.0</td>\n",
       "      <td>sl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2014-08-13</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>273.0</td>\n",
       "      <td>sl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>2014-08-14</td>\n",
       "      <td>new_death</td>\n",
       "      <td>13.0</td>\n",
       "      <td>sl</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          date      variable  total country\n",
       "4   2014-08-12     new_death   11.0      sl\n",
       "11  2014-08-12  new_infected  264.0      sl\n",
       "33  2014-08-13     new_death   15.0      sl\n",
       "40  2014-08-13  new_infected  273.0      sl\n",
       "62  2014-08-14     new_death   13.0      sl"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# put in a country column\n",
    "sl_final['country'] = 'sl'\n",
    "guinea_final['country'] = 'guinea'\n",
    "liberia_final['country'] = 'liberia'\n",
    "sl_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>variable</th>\n",
       "      <th>total</th>\n",
       "      <th>country</th>\n",
       "      <th>new_infected</th>\n",
       "      <th>new_death</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_death</td>\n",
       "      <td>11.0</td>\n",
       "      <td>sl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2014-08-12</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>264.0</td>\n",
       "      <td>sl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>2014-08-13</td>\n",
       "      <td>new_death</td>\n",
       "      <td>15.0</td>\n",
       "      <td>sl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2014-08-13</td>\n",
       "      <td>new_infected</td>\n",
       "      <td>273.0</td>\n",
       "      <td>sl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>2014-08-14</td>\n",
       "      <td>new_death</td>\n",
       "      <td>13.0</td>\n",
       "      <td>sl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          date      variable  total country  new_infected  new_death\n",
       "4   2014-08-12     new_death   11.0      sl           NaN        NaN\n",
       "11  2014-08-12  new_infected  264.0      sl           NaN        NaN\n",
       "33  2014-08-13     new_death   15.0      sl           NaN        NaN\n",
       "40  2014-08-13  new_infected  273.0      sl           NaN        NaN\n",
       "62  2014-08-14     new_death   13.0      sl           NaN        NaN"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# put the new_death and new_infected as column\n",
    "sl_final[new_infected] = np.nan\n",
    "sl_final[new_death] = np.nan\n",
    "#sl_final = sl_final.set_index(['country', 'date', 'variable'])\n",
    "sl_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method DataFrame.iterrows of             date      variable   total country  new_infected  new_death\n",
       "4     2014-08-12     new_death    11.0      sl           NaN        NaN\n",
       "11    2014-08-12  new_infected   264.0      sl           NaN        NaN\n",
       "33    2014-08-13     new_death    15.0      sl           NaN        NaN\n",
       "40    2014-08-13  new_infected   273.0      sl           NaN        NaN\n",
       "62    2014-08-14     new_death    13.0      sl           NaN        NaN\n",
       "69    2014-08-14  new_infected   280.0      sl           NaN        NaN\n",
       "91    2014-08-15     new_death    10.0      sl           NaN        NaN\n",
       "98    2014-08-15  new_infected   287.0      sl           NaN        NaN\n",
       "120   2014-08-16     new_death    18.0      sl           NaN        NaN\n",
       "127   2014-08-16  new_infected   297.0      sl           NaN        NaN\n",
       "149   2014-08-17     new_death     2.0      sl           NaN        NaN\n",
       "156   2014-08-17  new_infected   305.0      sl           NaN        NaN\n",
       "178   2014-08-18     new_death     5.0      sl           NaN        NaN\n",
       "185   2014-08-18  new_infected   312.0      sl           NaN        NaN\n",
       "207   2014-08-19     new_death     9.0      sl           NaN        NaN\n",
       "214   2014-08-19  new_infected   320.0      sl           NaN        NaN\n",
       "236   2014-08-20     new_death     4.0      sl           NaN        NaN\n",
       "243   2014-08-20  new_infected   322.0      sl           NaN        NaN\n",
       "265   2014-08-21     new_death     9.0      sl           NaN        NaN\n",
       "272   2014-08-21  new_infected   329.0      sl           NaN        NaN\n",
       "294   2014-08-22     new_death    56.0      sl           NaN        NaN\n",
       "301   2014-08-22  new_infected   333.0      sl           NaN        NaN\n",
       "323   2014-08-23     new_death    23.0      sl           NaN        NaN\n",
       "330   2014-08-23  new_infected   336.0      sl           NaN        NaN\n",
       "352   2014-08-24     new_death    31.0      sl           NaN        NaN\n",
       "359   2014-08-24  new_infected   341.0      sl           NaN        NaN\n",
       "381   2014-08-25     new_death    20.0      sl           NaN        NaN\n",
       "388   2014-08-25  new_infected   355.0      sl           NaN        NaN\n",
       "410   2014-08-26     new_death     6.0      sl           NaN        NaN\n",
       "417   2014-08-26  new_infected   363.0      sl           NaN        NaN\n",
       "...          ...           ...     ...     ...           ...        ...\n",
       "2607  2014-11-13  new_infected     0.0      sl           NaN        NaN\n",
       "2637  2014-11-15     new_death    54.0      sl           NaN        NaN\n",
       "2644  2014-11-15  new_infected     0.0      sl           NaN        NaN\n",
       "2674  2014-11-16     new_death    89.0      sl           NaN        NaN\n",
       "2681  2014-11-16  new_infected     0.0      sl           NaN        NaN\n",
       "2711  2014-11-17     new_death    53.0      sl           NaN        NaN\n",
       "2718  2014-11-17  new_infected     0.0      sl           NaN        NaN\n",
       "2748  2014-11-18     new_death    43.0      sl           NaN        NaN\n",
       "2755  2014-11-18  new_infected     0.0      sl           NaN        NaN\n",
       "2822  2014-11-20     new_death    94.0      sl           NaN        NaN\n",
       "2829  2014-11-20  new_infected  1282.0      sl           NaN        NaN\n",
       "2859  2014-11-21     new_death    51.0      sl           NaN        NaN\n",
       "2866  2014-11-21  new_infected  1303.0      sl           NaN        NaN\n",
       "2896  2014-11-22     new_death    47.0      sl           NaN        NaN\n",
       "2903  2014-11-22  new_infected  1333.0      sl           NaN        NaN\n",
       "2933  2014-11-23     new_death    39.0      sl           NaN        NaN\n",
       "2940  2014-11-23  new_infected  1364.0      sl           NaN        NaN\n",
       "2970  2014-11-24     new_death    83.0      sl           NaN        NaN\n",
       "2977  2014-11-24  new_infected  1397.0      sl           NaN        NaN\n",
       "3007  2014-11-28     new_death    84.0      sl           NaN        NaN\n",
       "3014  2014-11-28  new_infected  2962.0      sl           NaN        NaN\n",
       "3044  2014-11-29     new_death   134.0      sl           NaN        NaN\n",
       "3051  2014-11-29  new_infected  1496.0      sl           NaN        NaN\n",
       "3081  2014-12-01     new_death    72.0      sl           NaN        NaN\n",
       "3088  2014-12-01  new_infected  1549.0      sl           NaN        NaN\n",
       "3118  2014-12-04     new_death    37.0      sl           NaN        NaN\n",
       "3125  2014-12-04  new_infected  1648.0      sl           NaN        NaN\n",
       "3155  2014-12-05     new_death    54.0      sl           NaN        NaN\n",
       "3162  2014-12-05  new_infected  1669.0      sl           NaN        NaN\n",
       "3199  2014-12-06  new_infected  1708.0      sl           NaN        NaN\n",
       "\n",
       "[199 rows x 6 columns]>"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_final.iterrows"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## References\n",
    "\n",
    "[Python for Data Analysis](http://shop.oreilly.com/product/0636920023784.do) Wes McKinney"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  },
  "nav_menu": {},
  "toc": {
   "navigate_menu": true,
   "number_sections": false,
   "sideBar": false,
   "threshold": "3",
   "toc_cell": true,
   "toc_section_display": "none",
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
